---
title: "Count features"
author: "Charles Plessy"
date: "07/07/2023"
output: 
  html_document: 
    keep_md: yes
params:
  alnFile: "/absolute/path/to/your/file"
  matchType: "match_part or match depending on GFF file format"
  prefix: "PrefixForOutputFileNames"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(progress = TRUE, verbose = TRUE, root.dir = system2("pwd", stdout = TRUE))
```

```{r load_libraries}
library("GenomicBreaks") |> suppressPackageStartupMessages()
library("ggplot2")       |> suppressPackageStartupMessages()
```

How to use this Rmarkdown template:

It is essential to set `R_LIBS_USER` to an empty value, otherwise `R` will
attempt to load packages from the user's library, which were compiled in a
different environment than the GenomicBreaks Singularity image.

Example command

```
R_LIBS_USER='' R -e 'rmarkdown::render("thisTemplate.Rmd", output_file = "./outFile.html", params = list(alnFile = "/path/do/dir/with/alignment/files"))'
```

Load data
---------

```{r defaults}
if (params$alnFile == "/absolute/path/to/your/file") {
  alnFile <- system.file("extdata/contigs.genome.maf.gz", package = "GenomicBreaks")
  matchType <- "match_part"
} else {
  alnFile <- params$alnFile
  matchType <- params$matchType
}
```

```{r load_data}
# Display parameters
params

# Load the alignment in a GBreaks object
gb <-load_genomic_breaks(alnFile, type = params$matchType)
gb_p <- keepLongestPair(gb, drop = TRUE)
```

Coalesce contigs
----------------

```{r coalesce_contigs}
coa   <- coalesce_contigs(gb)
coa_p <- coalesce_contigs(gb_p)
makeOxfordPlots(coa, col = 'strand') + ggtitle(params$alnFile)
```

## Classify regions

We divide the genome into four categories: _isolated alignments_, _breakpoint
regions_, _colinear alignments_ and _bridge regions_.

_Colinear alignments_ are defined by the colinearity relationship computed
in `?flagColinearAlignments`.  _Bridge regions_ separate alignments that are
colinear to each other.  _Isolated alignments_ have no colinear counterparts
and _breakpoint regions_ are the remaining intervals.
   
```{r clean_gaps}
wgo    <- wholeGenomeClassification(gb,   coa)
wgol   <- split(wgo, wgo$type)
wgo_p  <- wholeGenomeClassification(gb_p, coa_p)
wgol_p <- split(wgo_p, wgo_p$type)
```

Flag all structural variants that we can detect
-----------------------------------------------

```{r study_inversions}
coa   <- coa   |> flagColinearAlignments() |> flagInversions() |> flagDoubleInversions() |> flagTranslocations() |> flagAll()
coa$flag[is.na(coa$flag)] <- "" # for easier subsetting later.
coa_p <- coa_p |> flagColinearAlignments() |> flagInversions() |> flagDoubleInversions() |> flagTranslocations() |> flagAll()
coa_p$flag[is.na(coa_p$flag)] <- ""

```

### Flip the simple inversions and coalesce

```{r study_inversions2}
# flipped <- flipInversions(coa) |> coalesce_contigs()
# 
# coa <- flagInversions(coa)
# sum(coa$inv)
# inv <- filterInversions(coa)
# head(inv, 11)
```


Width plots
-----------

```{r plot_widths, fig.height=12, dev=c('svg', 'png')}
# Need to wrap in a function to handle empty GRanges objects...
width2df <- function(what, gr) {
  if (length(gr) == 0) {
    data.frame(what = what, width = 0)
  } else { 
    data.frame(what = what, width = width(gr))
  }
}
plotRegionWidths <- function(wgol, coa) {
  rbind(
    width2df(what = "isol_aln", gr = wgol$`isolated alignment`),
    width2df(what = "breakpoint_regions",   gr = wgol$`breakpoint region`),
    width2df(what = "colinear_aln", gr = wgol$`collinear alignment`),
    width2df(what = "colinear_region", gr = coa),
    width2df(what = "bridge", gr = wgol$`bridge region`),
    width2df(what = "translocations", gr = coa[coa$flag == "Tra"]),
    width2df(what = "inversions", gr = coa[coa$flag == "Inv"])
  ) |> ggplot() +
    aes(width) +
    geom_histogram() +
    scale_x_log10() +
    facet_wrap(~what, ncol = 1, scales = "free_y")
}
plotRegionWidths(wgol, coa) + ggtitle("Whole object")
if (length(wgol_p > 0) & length(coa_p) > 0) {
  plotRegionWidths(wgol_p, coa_p) + ggtitle("Longest 2 matching sequences")
}
```


Calculate numbers and prepare them for export in a YAML file
------------------------------------------------------------

```{r count_features}
customSummary <- function(x, pasteToNames=NULL, suffix = NULL) {
  if (length(x) == 0) {
    s <- c(Min = NA, Q1 = NA, Median = NA, Mean = NA, Q3 = NA, Max = NA, L50 = NA, Total = NA, N = NA)
  } else {
    s <- summary(x)
    names(s) <- c("Min", "Q1", "Median", "Mean", "Q3", "Max")
    s["L50"]     <- weighted.mean(x, as.numeric(x)) # as.num to avoid integer overflow
    s["Total"]   <- sum(x)
    s["N"]       <- length(x)
  }
  s <- as.list(s)
  if (!is.null(suffix)) pasteToNames <- paste(pasteToNames, suffix, sep = "_")
  names(s) <- paste(pasteToNames, names(s), sep = "_")
  s
}

summaryWidth <- function(gb, pasteToNames=NULL) {
  w <- if(length(gb) == 0) {
      0
    } else {
      width(gb)
    }
  customSummary(w, pasteToNames)
}

pasteif <- function(x, suffix) {
  if (! is.null(suffix))
    names(x) <- paste(names(x), suffix, sep = "_")
  x
}

makeRreport <- function(gb, coa, wgol, suffix = NULL) {
  gb$mismatches <- (width(gb) + width(gb$query) - gb$aLength - gb$matches)
  list() |>
  c(customSummary(gb$aLength,                            "aligned_length"            , suffix)) |>
  c(customSummary(score(gb),                             "aligned_score"             , suffix)) |>
  c(customSummary(gb$matches,                            "aligned_matches"           , suffix)) |>
  c(customSummary(gb$mismatches,                         "aligned_mismatches"        , suffix)) |>
  c(customSummary(gb$aLength - width(gb),                "aligned_gaps_target"       , suffix)) |>
  c(customSummary(gb$aLength - width(gb$query),          "aligned_gaps_query"        , suffix)) |>
  c(customSummary(100 * gb$matches / gb$aLength,         "matching_aligned"          , suffix)) |>
  c(customSummary(100 * gb$matches / width(gb),          "matching_target"           , suffix)) |>
  c(customSummary(100 * gb$matches / width(gb$query),    "matching_query"            , suffix)) |>
  c(customSummary(100 * gb$mismatches / gb$aLength,      "mismatching_aligned"       , suffix)) |>
  c(customSummary(100 * gb$mismatches / width(gb),       "mismatching_target"        , suffix)) |>
  c(customSummary(100 * gb$mismatches / width(gb$query), "mismatching_query"         , suffix)) |>
  c(customSummary(width(gb),                             "aligned_width_target"      , suffix)) |>
  c(customSummary(width(gb$query),                       "aligned_width_query"       , suffix)) |>
  c(customSummary(width(coa),                            "chain_width_target"        , suffix)) |>
  c(customSummary(width(coa$query),                      "chain_width_query"         , suffix)) |>
  c(customSummary(width(wgol$`collinear alignment`),     "collinear_width_target"    , suffix)) |>
  c(customSummary(width(wgol$`isolated alignment`),      "isolated_width_target"     , suffix)) |>
  c(customSummary(width(wgol$`bridge region`),           "bridge_width_target"       , suffix)) |>
  c(customSummary(width(wgol$`breakpoint region`),       "breakpoint_width_target"   , suffix)) |>
  c(customSummary(width(coa[coa$flag == "Inv"]),         "inverted_width_target"     , suffix)) |>
  c(customSummary(width(coa[coa$flag == "Tra"]),         "translocated_width_target" , suffix)) |>
  c(customSummary(guessSeqLengths(gb),                   "guessed_target_length"     , suffix)) |>
  c(customSummary(guessSeqLengths(gb$query),             "guessed_query_length"      , suffix)) |>
  c(list(index_synteny_target      = synteny_index(gb)                    ) |> pasteif(suffix)) |>
  c(list(index_synteny_query       = synteny_index(swap(gb))              ) |> pasteif(suffix)) |>
  c(list(index_correlation_target  = correlation_index(gb)                ) |> pasteif(suffix)) |>
  c(list(index_correlation_query   = correlation_index(swap(gb))          ) |> pasteif(suffix)) |>
  c(list(index_GOCvicinity4_target = GOC(gb, vicinity = 4)                ) |> pasteif(suffix)) |> 
  c(list(index_GOCvicinity4_query  = GOC(swap(gb), vicinity = 4)          ) |> pasteif(suffix)) |>
  c(list(index_strandRand_target   = strand_randomisation_index(gb)       ) |> pasteif(suffix)) |>
  c(list(index_strandRand_query    = strand_randomisation_index(swap(gb)) ) |> pasteif(suffix))
}
report <- c(
  makeRreport(gb,   coa,   wgol),
  makeRreport(gb_p, coa_p, wgol_p, suffix = "bestpair"))
report[sapply(report, length) == 0] <- NA
```

 - `aligned_length`: length of the alignments (including gaps on each genome).
 - `aligned_score`: score of the alignments (as computed by the aligner).
 - `aligned_matches`: number of identical bases in the alignment.
 - `aligned_mismatches`: number of bases aligned to each other but mismatching.
 - `aligned_gaps_target`: number of alignment gaps on the _target_ side.
 - `aligned_gaps_query`: number of alignment gaps on the _query_ side.
 - `matching_aligned`: number of identical bases divided by alignment length (%).
 - `matching_target`: number of identical bases divided by _target_ sequence length (%).
 - `matching_query`: number of identical bases divided by _query_ sequence length (%).
 - `mismatching_aligned`: number of mismatching bases divided by alignment length (%).
 - `mismatching_target`: number of mismatching bases divided by _target_ sequence length (%).
 - `mismatching_query`: number of mismatching bases divided by _query_ sequence length (%).
 - `aligned_width_target`: width of the aligned sequence on the _target_ genome (excluding gaps).
 - `aligned_width_query`: width of the aligned sequence on the _query_ genome (excluding gaps).
 - `chained_width_target`: width of the coalesced regions on the _target_ genome.
 - `chained_width_query`: width of the coalesced regions on the _query_ genome.
 - `isolated_width_target`: width of the isolated aligned regions on the _target_ genome.
 - `collinear_width_target`: width of the collinear aligned regions on the _target_ genome.
 - `bridge_width_target`: width of the bridge regions on the _target_ genome.
 - `breakpoint_width_target`: width of the breakpoint regions on the _target_ genome.
 - `inverted_width_target`: width of the inverted regions on the _target_ genome.
 - `translocated_width_target`: width of the translocated regions on the _target_ genome.
 - `guessed_target_length`: guessed width of the sequence features on the _target_ genome.
 - `guessed_query_length`: guessed width of the sequence features on the _query_ genome.
 - `index_synteny_target`: synteny index.
 - `index_synteny_query`: synteny index after flipping _target_ and _query_.
 - `index_correlation_target`: correlation index.
 - `index_correlation_query`: correlation index after flipping _target_ and _query_.
 - `index_GOCvicinity4_target`: GOC index (vicinity = 4).
 - `index_GOCvicinity4_query`: GOC index (vicinity = 4) after flipping _target_ and _query_.
 - `index_strandRand_target`: strand randomisation index.
 - `index_strandRand_query`: strand randomisation index after flipping _target_ and _query_.

Export the results to a YAML file.

```{r export_results}
yaml <- yaml::as.yaml(report)
yaml::write_yaml(yaml, paste0(params$prefix, ".yaml"))
cat(yaml)
```
