<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Load pairwise genome alignments into objects, coalesce colinear
    alignments to discover syntenic regions, flag inversions, and much more.">
<title>Pairwise Alignment Breakpoint Analysis • GenomicBreaks</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Pairwise Alignment Breakpoint Analysis">
<meta property="og:description" content="Load pairwise genome alignments into objects, coalesce colinear
    alignments to discover syntenic regions, flag inversions, and much more.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.14.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/GenomicBreaks.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/StructuralVariants.html">Structural Variants</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="genomicbreaks">GenomicBreaks<a class="anchor" aria-label="anchor" href="#genomicbreaks"></a>
</h1></div>
<div class="section level2">
<h2 id="purpose">Purpose<a class="anchor" aria-label="anchor" href="#purpose"></a>
</h2>
<p><em>GenomicBreaks</em> is a <code>R</code> package using <em><a href="https://bioconductor.org/" class="external-link">Bioconductor</a></em> libraries to analyse pairwise alignments of whole genomes in which the gene order has been scrambled by evolution, like in the picture below that represents the comparison of homologous chromosomes in two distantly related molds, <em>N. crassa</em> (chrIII) and <em>P. comata</em> (chr7).</p>
<center>
<img src="reference/figures/plotApairOfChrs_Neu-2.png" alt="Comparison between Neurospora crassa chrIII / Podospora comata chr7 (rev-complemented)" width="40%">
</center>
<p>This package is especially designed to parse and process the alignment files produced by the our <a href="https://github.com/oist/plessy_pairwiseGenomeComparison" class="external-link">pairwise genome alignment pipeline</a>, but should be capable to import output of other pipelines as well.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="install-the-package">Install the package.<a class="anchor" aria-label="anchor" href="#install-the-package"></a>
</h3>
<p>If you have <a href="https://www.bioconductor.org/install/" class="external-link">Bioconductor</a> and the <a href="https://remotes.r-lib.org/#installation" class="external-link">remotes</a> already installed, the following should work.</p>
<pre><code>Rscript -e 'remotes::install_github("oist/GenomicBreaks", repos=BiocManager::repositories())'</code></pre>
<p>Add <code>dependencies=TRUE</code> if you would like to install the packages needed to build the vignettes.</p>
</div>
<div class="section level3">
<h3 id="how-to-create-a-singularity-container-with-genomicbreaks-installed">How to create a Singularity container with GenomicBreaks installed.<a class="anchor" aria-label="anchor" href="#how-to-create-a-singularity-container-with-genomicbreaks-installed"></a>
</h3>
<p>See the <a href="./Singularity.def">Singularity reciepe file</a></p>
<pre><code>sudo singularity build GenomicBreaks.sif Singularity.def</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="genomicbreaks-in-brief">GenomicBreaks in brief:<a class="anchor" aria-label="anchor" href="#genomicbreaks-in-brief"></a>
</h2>
<p>A pairwise alignment of two genomes is loaded in <code>GBreaks</code> objects wrapping the <code>GRanges</code> class. Here is an example:</p>
<pre><code>GBreaks object with 505 ranges and 2 metadata columns:
        seqnames      ranges strand |     score                   query
           &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;               &lt;GRanges&gt;
    [1]     chrI  5860-10010      + |     10609  NC_047487.1:8723-12716
    [2]     chrI 11157-11782      + |      1253 NC_047487.1:13035-13680
    [3]     chrI 25371-26528      + |      3651 NC_047487.1:15380-16537
    [4]     chrI 26849-29699      + |      6130 NC_047487.1:17263-20175
    [5]     chrI 29937-30607      + |       767 NC_047487.1:20203-20866
    ...      ...         ...    ... .       ...                     ...
  [501]     chrM 67039-67870      + |      1441   NC_018044.1:7758-8439
  [502]     chrM 68205-68580      + |       830   NC_018044.1:8783-9180
  [503]     chrM 69178-76168      + |     14528  NC_018044.1:9650-16261
  [504]     chrM 77356-80022      + |      8066 NC_018044.1:53206-55865
  [505]     chrM 80919-85779      + |      6712 NC_018044.1:57885-61592
  -------
  seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
<p>See “<em>Get started</em>” on <a href="https://oist.github.io/GenomicBreaks" class="uri">https://oist.github.io/GenomicBreaks</a> for further details.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/oist/GenomicBreaks/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/BSD-3-Clause" class="external-link">BSD_3_clause</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing GenomicBreaks</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Charlotte West <br><small class="roles"> Author </small>  </li>
<li>Charles Plessy <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-7410-6295" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
