<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Changelog • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.19.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-news">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Changelog</h1>
      <small>Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.19.0" id="genomicbreaks-0190">GenomicBreaks 0.19.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0190"></a></h2>
<ul><li>Add a <code><a href="../reference/GCequilibrium.html">GCequilibrium()</a></code>, <code><a href="../reference/GCpressure.html">GCpressure()</a></code> and <code><a href="../reference/GCproportion.html">GCproportion()</a></code> function.</li>
<li>Add a <code><a href="../reference/P_distance.html">P_error()</a></code> function to compute the binomial standard error on <code><a href="../reference/P_distance.html">P_distance()</a></code>
</li>
<li>Add lazy-loaded example data, <code>exampleProbabilityMatrix</code> and <code>exampleSubstitutionMatrix</code> for easy testing of the functions above.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.18.0" id="genomicbreaks-0180">GenomicBreaks 0.18.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0180"></a></h2>
<ul><li>New <code><a href="../reference/tau_index.html">tau_index()</a></code> function. Closes <a href="https://github.com/oist/GenomicBreaks/issues/16" class="external-link">#16</a></li>
<li>Reform the nucleotide distance functions to also take count matrices as input.</li>
<li>Implement the 4 variants of percent nucleotide difference in <code><a href="../reference/P_distance.html">P_distance()</a></code> as defined by <a href="https://doi.org/10.1016/j.str.2004.04.001" class="external-link">May (2004)</a>. Closes <a href="https://github.com/oist/GenomicBreaks/issues/26" class="external-link">#26</a></li>
<li>Add a <code><a href="../reference/gapProportion.html">gapProportion()</a></code> function.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.17.0" id="genomicbreaks-0170">GenomicBreaks 0.17.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0170"></a></h2>
<ul><li>Switch to semantic versioning.</li>
<li>New <code><a href="../reference/P_distance.html">P_distance()</a></code> (ungapped percent difference) distance function.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.16.3" id="genomicbreaks-0163">GenomicBreaks 0.16.3<a class="anchor" aria-label="anchor" href="#genomicbreaks-0163"></a></h2>
<ul><li>New <code><a href="../reference/F81_distance.html">F81_distance()</a></code> (Felsenstein 1981) distance function.</li>
<li>Documentation and namespace fixes.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.16.2" id="genomicbreaks-0162">GenomicBreaks 0.16.2<a class="anchor" aria-label="anchor" href="#genomicbreaks-0162"></a></h2>
<ul><li>New <code><a href="../reference/matchPairs.html">matchPairs()</a></code> function to remove non-syntenic regions.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.16.1" id="genomicbreaks-0161">GenomicBreaks 0.16.1<a class="anchor" aria-label="anchor" href="#genomicbreaks-0161"></a></h2>
<ul><li>Fix a bug where <code><a href="../reference/removeInversions.html">removeInversions()</a></code>, <code><a href="../reference/removeDoubleInversions.html">removeDoubleInversions()</a></code>, and <code><a href="../reference/removeTranslocations.html">removeTranslocations()</a></code> would discard all the ranges if no inversion, double inversion or translocation (respectively) were found.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.16.0" id="genomicbreaks-0160">GenomicBreaks 0.16.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0160"></a></h2>
<ul><li>New functions to consolidate alignments by removing structural variants and re-coalescing the objects: <code><a href="../reference/removeInversions.html">removeInversions()</a></code>, <code><a href="../reference/removeDoubleInversions.html">removeDoubleInversions()</a></code>, and <code><a href="../reference/removeTranslocations.html">removeTranslocations()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.15.0" id="genomicbreaks-0150">GenomicBreaks 0.15.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0150"></a></h2>
<ul><li>New <code>detect</code> option in the show and filter functions for inversions and translocations, to re-compute flags automatically and simplify interactive use. Set to <code>FALSE</code> and flag manually when performance is an issue.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.10" id="genomicbreaks-01410">GenomicBreaks 0.14.10<a class="anchor" aria-label="anchor" href="#genomicbreaks-01410"></a></h2>
<ul><li>New <code><a href="../reference/slidingWindow.html">slidingWindow()</a></code> function to compute similarity indices on windowed intervals.</li>
<li>Experimental <code>tile</code> argument in <code><a href="../reference/strand_randomisation_index.html">strand_randomisation_index()</a></code>. It will be removed once <code><a href="../reference/slidingWindow.html">slidingWindow()</a></code> can be applied to <code>strand_randomisation_index</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.9" id="genomicbreaks-0149">GenomicBreaks 0.14.9<a class="anchor" aria-label="anchor" href="#genomicbreaks-0149"></a></h2>
<ul><li>New distance functions written by Zikun Yang.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.8" id="genomicbreaks-0148">GenomicBreaks 0.14.8<a class="anchor" aria-label="anchor" href="#genomicbreaks-0148"></a></h2>
<ul><li>Update for Bioc 3.21 and R 4.5.</li>
<li>New experimental inversion analysis functions written by Bruna Fistarol.</li>
<li>Namespace and documentation fixes.</li>
<li>Guard against empty <code>GBreaks</code> objects in <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code>.</li>
</ul><div class="section level3">
<h3 id="dropped-features-0-14-8">Dropped features<a class="anchor" aria-label="anchor" href="#dropped-features-0-14-8"></a></h3>
<ul><li>Removed <code>bp_coverage</code>, not used anymore.</li>
<li>Removed support for Axt files as CNEr is broken for the moment.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.7" id="genomicbreaks-0147">GenomicBreaks 0.14.7<a class="anchor" aria-label="anchor" href="#genomicbreaks-0147"></a></h2>
<ul><li>Correct a bug in <code><a href="../reference/load_genomic_breaks.html">load_genomic_breaks()</a></code> where sequence lengths would be too long of 1 nucleotide when loaded from MAF files.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.6" id="genomicbreaks-0146">GenomicBreaks 0.14.6<a class="anchor" aria-label="anchor" href="#genomicbreaks-0146"></a></h2>
<ul><li>Experimental support for loading sequence information from FASTA files.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.5" id="genomicbreaks-0145">GenomicBreaks 0.14.5<a class="anchor" aria-label="anchor" href="#genomicbreaks-0145"></a></h2>
<ul><li>Support loading MAF files where alignment sign is on the query.</li>
<li>Correct example data to make it really one-to-one.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.4" id="genomicbreaks-0144">GenomicBreaks 0.14.4<a class="anchor" aria-label="anchor" href="#genomicbreaks-0144"></a></h2>
<ul><li>Remove dependency on BOOST thanks to C++ code taken from LAST (M. Frith / GPLv3+).</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.3" id="genomicbreaks-0143">GenomicBreaks 0.14.3<a class="anchor" aria-label="anchor" href="#genomicbreaks-0143"></a></h2>
<div class="section level3">
<h3 id="new-functionalities-0-14-3">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-14-3"></a></h3>
<ul><li>New <code><a href="../reference/scaffoldByFlipAndMerge.html">scaffoldByFlipAndMerge()</a></code> and <code><a href="../reference/splitSeqLevel.html">splitSeqLevel()</a></code> functions for trivial scaffolding by hand.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.2" id="genomicbreaks-0142">GenomicBreaks 0.14.2<a class="anchor" aria-label="anchor" href="#genomicbreaks-0142"></a></h2>
<div class="section level3">
<h3 id="new-functionalities-0-14-2">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-14-2"></a></h3>
<ul><li>New <code><a href="../reference/GBreaksToMatrix.html">GBreaksToMatrix()</a></code> function to apply image processing approaches on scrambling patterns.</li>
<li>New <code><a href="../reference/keepLongestPair.html">keepLongestPair()</a></code> function to focus on the longest sequence feature in each genome.</li>
<li>New <code><a href="../reference/wholeGenomeClassification.html">wholeGenomeClassification()</a></code> function to classify <em>isolated alignments</em>, <em>collinear alignments</em>, <em>bridge regions</em> and <em>breakpoint regions</em>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.1" id="genomicbreaks-0141">GenomicBreaks 0.14.1<a class="anchor" aria-label="anchor" href="#genomicbreaks-0141"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-0-14-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-14-1"></a></h3>
<ul><li>Correct a 1-nt shift in minus-strand coordinates of the <em>query</em> genome of <em>GBreaks</em> objects loaded from MAF files. This shift could cause overlapping ranges that crashed the coalescing algorithm.</li>
<li>
<code><a href="../reference/bridgeRegions.html">bridgeRegions()</a></code> now returns empty <em>GBreaks</em> objects if no bridge region was found. Closes issue <a href="https://github.com/oist/GenomicBreaks/issues/20" class="external-link">#20</a>.</li>
<li>Allow <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code> to work on genomes larger than the maximum 32-bit integer. Closes issue <a href="https://github.com/oist/GenomicBreaks/issues/24" class="external-link">#24</a>.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functionalities-0-14-1">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-14-1"></a></h3>
<ul><li>New <code><a href="../reference/flipInversions.html">flipInversions()</a></code> function to remove detected inversions before searching for nested ones.</li>
<li>New <code><a href="../reference/filterDoubleInversions.html">filterDoubleInversions()</a></code> function for extracting double inversions.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.14.0" id="genomicbreaks-0140">GenomicBreaks 0.14.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0140"></a></h2>
<ul><li>The <code><a href="../reference/load_genomic_breaks.html">load_genomic_breaks()</a></code> function gains support for files in the <em>Multiple Alignment Format</em> (MAF).</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.13.1" id="genomicbreaks-0131">GenomicBreaks 0.13.1<a class="anchor" aria-label="anchor" href="#genomicbreaks-0131"></a></h2>
<div class="section level3">
<h3 id="new-functionalities-0-13-1">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-13-1"></a></h3>
<ul><li>The <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code> function gains a <code>col</code> argument for choosing to color by strand or by score.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-13-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-13-1"></a></h3>
<ul><li>In <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code>, the sequence names are now centered on their region.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.13.0" id="genomicbreaks-0130">GenomicBreaks 0.13.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0130"></a></h2>
<div class="section level3">
<h3 id="important-bug-fixes-0-13-0">Important bug fixes<a class="anchor" aria-label="anchor" href="#important-bug-fixes-0-13-0"></a></h3>
<ul><li>The <code><a href="../reference/flagColinearAlignments.html">flagColinearAlignments()</a></code> and <code><a href="../reference/coalesce_contigs.html">coalesce_contigs()</a></code> functions were corrected so that strandless ranges are called <em>colinear</em> when they satisfy the conditions on any strand. Before, they would be treated like plus-strand ranges, causing failure to detect colinearity on the minus strand.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functionalities-0-13-0">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-13-0"></a></h3>
<ul><li>New <code><a href="../reference/getSeq.html">getSeq()</a></code> method for <code>GRanges</code> and <code>GBreaks</code> objects that have proper <code>seqinfo</code> for an installed <code>BSgenome</code> package.</li>
<li>New <code><a href="../reference/pairwiseAlignment.html">pairwiseAlignment()</a></code> method for <code>GRanges</code> and <code>GBreaks</code> objects, that will call <code><a href="../reference/getSeq.html">getSeq()</a></code> automatically.</li>
<li>New “mid” option to the <code>direction</code> argument of <code><a href="../reference/get_bps.html">get_bps()</a></code>, to output breakpoints at mid-distance between ranges.</li>
<li>New <code><a href="../reference/bridgeRegions.html">bridgeRegions()</a></code> function that maps unaligned regions of the <em>target</em> to the <em>query</em> genome when they are flanked by colinear regions.</li>
<li>
<code><a href="../reference/filterTranslocations.html">filterTranslocations()</a></code> gains a <code>remove</code> flag and <code>flagTranslocations</code> gains a <code>both</code> flag.</li>
<li>
<code>swap</code> gains a <code>sort</code> flag, so that in interactive sessions, <code>gb |&gt; swap() |&gt; sort(i=T)</code> can be replaced by the shorter <code>gb |&gt; swap(s = T)</code>.</li>
</ul></div>
<div class="section level3">
<h3 id="backwards-incompatible-changes-0-13-0">Backwards-incompatible changes<a class="anchor" aria-label="anchor" href="#backwards-incompatible-changes-0-13-0"></a></h3>
<ul><li>The function <code>bp_heatmap</code> now expects to genome sequences to be reachable via <code><a href="https://rdrr.io/pkg/BSgenome/man/available.genomes.html" class="external-link">BSgenome::getBSgenome</a></code>.</li>
<li>The function <code>dist2next</code> now operates on <code>GRanges</code> and <code>GBreaks</code>, and its main argument is renamed to <code>x</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.12.2" id="genomicbreaks-0122">GenomicBreaks 0.12.2<a class="anchor" aria-label="anchor" href="#genomicbreaks-0122"></a></h2>
<div class="section level3">
<h3 id="new-functionalities-0-12-2">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-12-2"></a></h3>
<ul><li>The function <code>forceSeqLevels()</code> now operates on both the <em>target</em> and the <em>query</em> ranges of <code>GBreaks</code> objects.</li>
<li>New example data set: <em>Neurospora crassa</em> chromosome III aligned to its homologue in <em>Podospora comata</em>: chromosome 7.</li>
<li>New <code><a href="../reference/strand_randomisation_index.html">strand_randomisation_index()</a></code>.</li>
<li>
<code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code> includes strand information in the returned object.</li>
<li>New <code>type = 'none'</code> option for <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code>, for better customisation.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.12.1" id="genomicbreaks-0121">GenomicBreaks 0.12.1<a class="anchor" aria-label="anchor" href="#genomicbreaks-0121"></a></h2>
<ul><li>The <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> function now plots the name of the selected sequence levels instead of <code>target</code> and <code>query</code>. The labels can be overriden with the <code>dna_seg_labels</code> parameters. Closes issue <a href="https://github.com/oist/GenomicBreaks/issues/6" class="external-link">#6</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.12.0" id="genomicbreaks-0120">GenomicBreaks 0.12.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0120"></a></h2>
<div class="section level3">
<h3 id="backwards-incompatible-changes-0-12-0">Backwards-incompatible changes<a class="anchor" aria-label="anchor" href="#backwards-incompatible-changes-0-12-0"></a></h3>
<ul><li>Removed <code>selChroms</code> parameter in <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code>.</li>
<li>New defaults for <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code>: fixed-ratio line plot with round ends and size of 1. Axis labels are now “target” and “query” by default. In any case, Seqname is displayed instead of genome name if there is only one sequence in the ranges object.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functionalities-0-12-0">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-12-0"></a></h3>
<ul><li>New function <code><a href="../reference/orderQuerySeqLevels.html">orderQuerySeqLevels()</a></code> to better diagonalise dot plots.</li>
<li>New <code>diag</code> parameter in <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code> to diagonalise the plots by re-ordering <em>query</em> sequence levels.</li>
<li>New <code>size</code> parameter in <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code>
</li>
<li>New <code><a href="../reference/guessSeqLengths.html">guessSeqLengths()</a></code> and <code><a href="../reference/forceSeqLengths.html">forceSeqLengths()</a></code> functions to use as last ressort when sequence lengths are not known.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.11.0" id="genomicbreaks-0110">GenomicBreaks 0.11.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0110"></a></h2>
<div class="section level3">
<h3 id="backwards-incompatible-changes-0-11-0">Backwards-incompatible changes<a class="anchor" aria-label="anchor" href="#backwards-incompatible-changes-0-11-0"></a></h3>
<ul><li>Remove project-specific <code>load_one_to_ones</code> function.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.10.0" id="genomicbreaks-0100">GenomicBreaks 0.10.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-0100"></a></h2>
<div class="section level3">
<h3 id="backwards-incompatible-changes-0-10-0">Backwards-incompatible changes<a class="anchor" aria-label="anchor" href="#backwards-incompatible-changes-0-10-0"></a></h3>
<ul><li>Removed <code>flagInsersions</code>, redundant with <code><a href="../reference/flagTranslocations.html">flagTranslocations()</a></code> and buggy on minus strands.</li>
<li>
<code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> now plots coordinates and a scale bar by default.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functionalities-0-10-0">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-10-0"></a></h3>
<ul><li>New <code><a href="../reference/flagPairs.html">flagPairs()</a></code> function to exhaustively flag windows of 2 successive <code>GBreaks</code> entries.</li>
<li>New <code><a href="../reference/gb2xlim.html">gb2xlim()</a></code> function to narrow focus on plots made with <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code>.</li>
<li>New <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code> function to create macrosynteny plots. Thanks to Aleksandra Bliznina.</li>
<li>In <code><a href="../reference/mergeSeqLevels.html">mergeSeqLevels()</a></code>, use maximal end value for each <code>seqlevel</code> if no if no <code>seqlengths</code> were provided.</li>
<li>A new set of example 5-uplets, to illustrate double inversions.</li>
<li>Flag double inversions with <code><a href="../reference/flagDoubleInversions.html">flagDoubleInversions()</a></code> and <code><a href="../reference/flagAll.html">flagAll()</a></code>, test them with <code>exampleDoubleInversion</code>.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-10-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-10-0"></a></h3>
<ul><li>Add strand check on <code><a href="../reference/flagTranslocations.html">flagTranslocations()</a></code> so that it does not return a false positive on <code>exampleDoubleInversion</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.9.1" id="genomicbreaks-091">GenomicBreaks 0.9.1<a class="anchor" aria-label="anchor" href="#genomicbreaks-091"></a></h2>
<div class="section level3">
<h3 id="new-functionalities-0-9-1">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-9-1"></a></h3>
<ul><li>
<code><a href="../reference/flagAll.html">flagAll()</a></code> now ensures that flags do not collide, and reports translocations.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.9.0" id="genomicbreaks-090">GenomicBreaks 0.9.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-090"></a></h2>
<div class="section level3">
<h3 id="new-functionalities-0-9-0">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-9-0"></a></h3>
<ul><li>New functions <code><a href="../reference/flagTranslocations.html">flagTranslocations()</a></code>, <code><a href="../reference/showTranslocations.html">showTranslocations()</a></code> a <code><a href="../reference/filterTranslocations.html">filterTranslocations()</a></code>.</li>
<li>New <code><a href="../reference/filterColinearRegions.html">filterColinearRegions()</a></code> function.</li>
<li>New <code><a href="../reference/cleanGaps.html">cleanGaps()</a></code> function that is guaranteed to return only the intervals between the input ranges.</li>
<li>The <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> constructor now accepts three new parameters, <code>target</code>, <code>query</code> and <code>strand</code>, to construct <code>GBreaks</code> objects directly from <code>GRanges</code> objects.</li>
<li>New <code><a href="https://rdrr.io/r/base/range.html" class="external-link">range()</a></code> methods for <code>GBreaks</code> objects, that runs <code>GRanges::range</code> on each pair of target/query <em>seqnames</em>, ignoring strand information.</li>
<li>New <code>subsetByOverlaps()</code> range that combines regions overlapping by their <em>target</em> or by their <em>query</em> ranges.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-9-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-9-0"></a></h3>
<ul><li>
<code><a href="../reference/flagColinearAlignments.html">flagColinearAlignments()</a></code> now handles properly strandless ranges. (Issue <a href="https://github.com/oist/GenomicBreaks/issues/1" class="external-link">#1</a>)</li>
<li>
<code><a href="../reference/swap.html">swap()</a></code> now returns proper <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> objects.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.8.0" id="genomicbreaks-080">GenomicBreaks 0.8.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-080"></a></h2>
<div class="section level3">
<h3 id="backwards-incompatible-changes-0-8-0">Backwards-incompatible changes<a class="anchor" aria-label="anchor" href="#backwards-incompatible-changes-0-8-0"></a></h3>
<ul><li>Remove <code>flagInterruptions</code>, which was nonsense.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functionalities-0-8-0">New functionalities<a class="anchor" aria-label="anchor" href="#new-functionalities-0-8-0"></a></h3>
<ul><li>New <code><a href="../reference/isSorted.html">isSorted()</a></code> method to properly test with <code>ignore.strand</code> set to <code>TRUE</code>.</li>
<li>Default <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> to the first <em>sequence level</em> of the <code>GBreaks</code> object.</li>
<li>
<code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> now renders regions as blocks and not arrows as the strand information is encoded in the color and shape of the comparison bands. It also plots in light blue the blocks that are not matched. Closes issues <a href="https://github.com/oist/GenomicBreaks/issues/3" class="external-link">#3</a> and <a href="https://github.com/oist/GenomicBreaks/issues/5" class="external-link">#5</a>.</li>
<li>Added example data, see ?StructuralVariants and `vignette(“Structural Variants”, package = “GenomicBreaks”) for details.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-8-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-8-0"></a></h3>
<ul><li>In <code><a href="../reference/flagInversions.html">flagInversions()</a></code>, check if input was sorted. Closes Issue <a href="https://github.com/oist/GenomicBreaks/issues/2" class="external-link">#2</a>.</li>
<li>
<code><a href="../reference/flagLongShort.html">flagLongShort()</a></code> now handles the case where some annotations do not match.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.7.1" id="genomicbreaks-071">GenomicBreaks 0.7.1<a class="anchor" aria-label="anchor" href="#genomicbreaks-071"></a></h2>
<ul><li>Fix namespace and variable bugs that were preventing <code><a href="../reference/reverse.html">reverse()</a></code> to run.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.7.0" id="genomicbreaks-070">GenomicBreaks 0.7.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-070"></a></h2>
<ul><li>New <code><a href="../reference/mergeSeqLevels.html">mergeSeqLevels()</a></code> function to concatenate sequence features, for instance before plotting.</li>
<li>New <code><a href="../reference/reverse.html">reverse()</a></code> method for <code>GBreaks</code> and <code>GRanges</code> objects.</li>
<li>Really return a <code>GBreaks</code> object in <code><a href="../reference/coalesce_contigs.html">coalesce_contigs()</a></code>.</li>
<li>Pass extra arguments from <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> to <code><a href="https://rdrr.io/pkg/genoPlotR/man/plot_gene_map.html" class="external-link">genoPlotR::plot_gene_map</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.6.0" id="genomicbreaks-060">GenomicBreaks 0.6.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-060"></a></h2>
<ul><li>Add pairwise chromosome plot functions using the <a href="http://genoplotr.r-forge.r-project.org" class="external-link"><code>genoPlotR</code></a> package, see in particular <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.5.3" id="genomicbreaks-053">GenomicBreaks 0.5.3<a class="anchor" aria-label="anchor" href="#genomicbreaks-053"></a></h2>
<ul><li>
<em>pkgdown</em> update.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.5.2" id="genomicbreaks-052">GenomicBreaks 0.5.2<a class="anchor" aria-label="anchor" href="#genomicbreaks-052"></a></h2>
<ul><li>Greatly speed up <code><a href="../reference/flagLongShort.html">flagLongShort()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.5.1" id="genomicbreaks-051">GenomicBreaks 0.5.1<a class="anchor" aria-label="anchor" href="#genomicbreaks-051"></a></h2>
<ul><li>Correct a variable name bug in <code><a href="../reference/flagLongShort.html">flagLongShort()</a></code>.</li>
<li>Export the <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> constructor.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.5.0" id="genomicbreaks-050">GenomicBreaks 0.5.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-050"></a></h2>
<ul><li>Add a <code><a href="../reference/flagLongShort.html">flagLongShort()</a></code> function to transfer annotations about chromosome arms.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.4.0" id="genomicbreaks-040">GenomicBreaks 0.4.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-040"></a></h2>
<ul><li>Use alignments of <em>Saccharomyces paradoxus</em> to <em>S. cerevisiae</em> as example data.</li>
<li>Addition of a <code><a href="../reference/GOC.html">GOC()</a></code> function to calculate Gene Order Conservation.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.3.0" id="genomicbreaks-030">GenomicBreaks 0.3.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-030"></a></h2>
<ul><li>Added the <code><a href="../reference/leftInversionGaps.html">leftInversionGaps()</a></code> function.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.0" id="genomicbreaks-020">GenomicBreaks 0.2.0<a class="anchor" aria-label="anchor" href="#genomicbreaks-020"></a></h2>
<ul><li>Document the package with <a href="https://pkgdown.r-lib.org/" class="external-link"><code>pkgdown</code></a>.</li>
</ul></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

