<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Changelog • GenomicBreaks</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Changelog" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicBreaks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomicBreaks.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/StructuralVariants.html">Structural Variants</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/oist/GenomicBreaks/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href='https://github.com/oist/GenomicBreaks/blob/master/NEWS.md'><code>NEWS.md</code></a></small>
    </div>

    <div id="genomicbreaks-0101" class="section level1">
<h1 class="page-header" data-toc-text="0.10.1">
<a href="#genomicbreaks-0101" class="anchor"></a>GenomicBreaks 0.10.1</h1>
<div id="backwards-incompatible-changes" class="section level2">
<h2 class="hasAnchor">
<a href="#backwards-incompatible-changes" class="anchor"></a>Backwards-incompatible changes</h2>
<ul>
<li>Removed <code>flagInsersions</code>, redundant with <code><a href="../reference/flagTranslocations.html">flagTranslocations()</a></code> and buggy on minus strands.</li>
<li>
<code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> now plots coordinates and a scale bar by default.</li>
</ul>
</div>
<div id="new-functionalities" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functionalities" class="anchor"></a>New functionalities</h2>
<ul>
<li>New <code><a href="../reference/gb2xlim.html">gb2xlim()</a></code> function to narrow focus on plots made with <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code>.</li>
<li>New <code><a href="../reference/makeOxfordPlots.html">makeOxfordPlots()</a></code> function to create macrosynteny plots. Thanks to Aleksandra Bliznina.</li>
<li>A new set of example 5-uplets, to illustrate double inversions.</li>
<li>Flag double inversions with <code>flagDoubleInversions()</code> and <code><a href="../reference/flagAll.html">flagAll()</a></code>, test them with <code>exampleDoubleInversion</code>.</li>
</ul>
</div>
<div id="bug-fixes" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-fixes" class="anchor"></a>Bug fixes</h2>
<ul>
<li>Add strand check on <code><a href="../reference/flagTranslocations.html">flagTranslocations()</a></code> so that it does not return a false positive on <code>exampleDoubleInversion</code>.</li>
</ul>
</div>
</div>
    <div id="genomicbreaks-091" class="section level1">
<h1 class="page-header" data-toc-text="0.9.1">
<a href="#genomicbreaks-091" class="anchor"></a>GenomicBreaks 0.9.1</h1>
<div id="new-functionalities-1" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functionalities-1" class="anchor"></a>New functionalities</h2>
<ul>
<li>
<code><a href="../reference/flagAll.html">flagAll()</a></code> now ensures that flags do not collide, and reports translocations.</li>
</ul>
</div>
</div>
    <div id="genomicbreaks-090" class="section level1">
<h1 class="page-header" data-toc-text="0.9.0">
<a href="#genomicbreaks-090" class="anchor"></a>GenomicBreaks 0.9.0</h1>
<div id="new-functionalities-2" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functionalities-2" class="anchor"></a>New functionalities</h2>
<ul>
<li>New functions <code><a href="../reference/flagTranslocations.html">flagTranslocations()</a></code>, <code><a href="../reference/showTranslocations.html">showTranslocations()</a></code> a <code><a href="../reference/filterTranslocations.html">filterTranslocations()</a></code>.</li>
<li>New <code><a href="../reference/filterColinearRegions.html">filterColinearRegions()</a></code> function.</li>
<li>New <code><a href="../reference/cleanGaps.html">cleanGaps()</a></code> function that is guaranteed to return only the intervals between the input ranges.</li>
<li>The <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> constructor now accepts three new parameters, <code>target</code>, <code>query</code> and <code>strand</code>, to construct <code>GBreaks</code> objects directly from <code>GRanges</code> objects.</li>
<li>New <code><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">range()</a></code> methods for <code>GBreaks</code> objects, that runs <code>GRanges::range</code> on each pair of target/query <em>seqnames</em>, ignoring strand information.</li>
<li>New <code><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">subsetByOverlaps()</a></code> range that combines regions overlapping by their <em>target</em> or by their <em>query</em> ranges.</li>
</ul>
</div>
<div id="bug-fixes-1" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-fixes-1" class="anchor"></a>Bug fixes</h2>
<ul>
<li>
<code><a href="../reference/flagColinearAlignments.html">flagColinearAlignments()</a></code> now handles properly strandless ranges. (Issue <a href='https://github.com/oist/GenomicBreaks/issues/1'>#1</a>)</li>
<li>
<code><a href="../reference/swap.html">swap()</a></code> now returns proper <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> objects.</li>
</ul>
</div>
</div>
    <div id="genomicbreaks-080" class="section level1">
<h1 class="page-header" data-toc-text="0.8.0">
<a href="#genomicbreaks-080" class="anchor"></a>GenomicBreaks 0.8.0</h1>
<div id="backwards-incompatible-changes-1" class="section level2">
<h2 class="hasAnchor">
<a href="#backwards-incompatible-changes-1" class="anchor"></a>Backwards-incompatible changes</h2>
<ul>
<li>Remove <code>flagInterruptions</code>, which was nonsense.</li>
</ul>
</div>
<div id="new-functionalities-3" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functionalities-3" class="anchor"></a>New functionalities</h2>
<ul>
<li>New <code><a href="../reference/isSorted.html">isSorted()</a></code> method to properly test with <code>ignore.strand</code> set to <code>TRUE</code>.</li>
<li>Default <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> to the first <em>sequence level</em> of the <code>GBreaks</code> object.</li>
<li>
<code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> now renders regions as blocks and not arrows as the strand information is encoded in the color and shape of the comparison bands. It also plots in light blue the blocks that are not matched. Closes issues <a href='https://github.com/oist/GenomicBreaks/issues/3'>#3</a> and <a href='https://github.com/oist/GenomicBreaks/issues/5'>#5</a>.</li>
<li>Added example data, see ?StructuralVariants and `vignette(“Structural Variants”, package = “GenomicBreaks”) for details.</li>
</ul>
</div>
<div id="bug-fixes-2" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-fixes-2" class="anchor"></a>Bug fixes</h2>
<ul>
<li>In <code><a href="../reference/flagInversions.html">flagInversions()</a></code>, check if input was sorted. Closes Issue <a href='https://github.com/oist/GenomicBreaks/issues/2'>#2</a>.</li>
<li>
<code><a href="../reference/flagLongShort.html">flagLongShort()</a></code> now handles the case where some annotations do not match.</li>
</ul>
</div>
</div>
    <div id="genomicbreaks-071" class="section level1">
<h1 class="page-header" data-toc-text="0.7.1">
<a href="#genomicbreaks-071" class="anchor"></a>GenomicBreaks 0.7.1</h1>
<ul>
<li>Fix namespace and variable bugs that were preventing <code><a href="../reference/reverse.html">reverse()</a></code> to run.</li>
</ul>
</div>
    <div id="genomicbreaks-070" class="section level1">
<h1 class="page-header" data-toc-text="0.7.0">
<a href="#genomicbreaks-070" class="anchor"></a>GenomicBreaks 0.7.0</h1>
<ul>
<li>New <code><a href="../reference/mergeSeqLevels.html">mergeSeqLevels()</a></code> function to concatenate sequence features, for instance before plotting.</li>
<li>New <code><a href="../reference/reverse.html">reverse()</a></code> method for <code>GBreaks</code> and <code>GRanges</code> objects.</li>
<li>Really return a <code>GBreaks</code> object in <code><a href="../reference/coalesce_contigs.html">coalesce_contigs()</a></code>.</li>
<li>Pass extra arguments from <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code> to <code><a href="https://rdrr.io/pkg/genoPlotR/man/plot_gene_map.html">genoPlotR::plot_gene_map</a></code>.</li>
</ul>
</div>
    <div id="genomicbreaks-060" class="section level1">
<h1 class="page-header" data-toc-text="0.6.0">
<a href="#genomicbreaks-060" class="anchor"></a>GenomicBreaks 0.6.0</h1>
<ul>
<li>Add pairwise chromosome plot functions using the <a href="http://genoplotr.r-forge.r-project.org"><code>genoPlotR</code></a> package, see in particular <code><a href="../reference/plotApairOfChrs.html">plotApairOfChrs()</a></code>.</li>
</ul>
</div>
    <div id="genomicbreaks-053" class="section level1">
<h1 class="page-header" data-toc-text="0.5.3">
<a href="#genomicbreaks-053" class="anchor"></a>GenomicBreaks 0.5.3</h1>
<ul>
<li>
<em>pkgdown</em> update.</li>
</ul>
</div>
    <div id="genomicbreaks-052" class="section level1">
<h1 class="page-header" data-toc-text="0.5.2">
<a href="#genomicbreaks-052" class="anchor"></a>GenomicBreaks 0.5.2</h1>
<ul>
<li>Greatly speed up <code><a href="../reference/flagLongShort.html">flagLongShort()</a></code>.</li>
</ul>
</div>
    <div id="genomicbreaks-051" class="section level1">
<h1 class="page-header" data-toc-text="0.5.1">
<a href="#genomicbreaks-051" class="anchor"></a>GenomicBreaks 0.5.1</h1>
<ul>
<li>Correct a variable name bug in <code><a href="../reference/flagLongShort.html">flagLongShort()</a></code>.</li>
<li>Export the <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> constructor.</li>
</ul>
</div>
    <div id="genomicbreaks-050" class="section level1">
<h1 class="page-header" data-toc-text="0.5.0">
<a href="#genomicbreaks-050" class="anchor"></a>GenomicBreaks 0.5.0</h1>
<ul>
<li>Add a <code><a href="../reference/flagLongShort.html">flagLongShort()</a></code> function to transfer annotations about chromosome arms.</li>
</ul>
</div>
    <div id="genomicbreaks-040" class="section level1">
<h1 class="page-header" data-toc-text="0.4.0">
<a href="#genomicbreaks-040" class="anchor"></a>GenomicBreaks 0.4.0</h1>
<ul>
<li>Use alignments of <em>Saccharomyces paradoxus</em> to <em>S. cerevisiae</em> as example data.</li>
<li>Addition of a <code><a href="../reference/GOC.html">GOC()</a></code> function to calculate Gene Order Conservation.</li>
</ul>
</div>
    <div id="genomicbreaks-030" class="section level1">
<h1 class="page-header" data-toc-text="0.3.0">
<a href="#genomicbreaks-030" class="anchor"></a>GenomicBreaks 0.3.0</h1>
<ul>
<li>Added the <code><a href="../reference/leftInversionGaps.html">leftInversionGaps()</a></code> function.</li>
</ul>
</div>
    <div id="genomicbreaks-020" class="section level1">
<h1 class="page-header" data-toc-text="0.2.0">
<a href="#genomicbreaks-020" class="anchor"></a>GenomicBreaks 0.2.0</h1>
<ul>
<li>Document the package with <a href="https://pkgdown.r-lib.org/"><code>pkgdown</code></a>.</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


