---
title: "Orthogroup syntenies"
output: 
  html_document: 
    toc: yes
    number_sections: yes
    keep_md: yes
    df_print: kable
editor_options: 
  chunk_output_type: console
---

Load packages and data
======================

R packages
----------

```{r load_packages}
library('GenomicBreaks')
suppressPackageStartupMessages({
  library('BSgenome')
  library('GenomicFeatures')
  library('GenomicRanges')
})
```

BSgenome packages
-----------------

Example command to install one _Oikopleura_ BSgenome package if you do not yet
have it:

```
install.packages("BSgenome.Odioica.local.OKI2018.I69", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_BSgenome_packages}
library("BSgenome.Odioica.local.OKI2018.I69")
library("BSgenome.Odioica.local.OSKA2016v1.9")
library("BSgenome.Odioica.local.Bar2.p4")
library("BSgenome.Odioica.local.Odioica.reference.v3.0")
library("BSgenome.Odioica.local.KUM.M3")

genomes <- SimpleList(
  # Chromosome assemblies
  Oki = OKI2018_I69, Osa = OSKA2016v1.9, Bar = Bar2_p4,
  # Less contiguous assemblies
  Kum = KUM_M3, Nor = OdB3)
```

Annotations
-----------

This assumes that `/bucket/LuscombeU/common/Breakpoints/Annotations` was copied
in the package's root directory.

```{r load_annotations}
annots <- SimpleList()

gff2txdb <- function(file, genome) {
  tx <- rtracklayer::import.gff(file)
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

gff2txdb_Norway <- function(file, genome) {
  tx <- rtracklayer::import.gff(file)
  # Remove "scaffoldA" objects in the OdB3 annotation
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx$ID <- NA_character_
  mcols(tx)[tx$type == "mRNA","ID"] <- paste0(tx[tx$type == "mRNA"]$mRNA, ".mRNA")
  mcols(tx)[tx$type == "CDS","ID"] <- paste0(tx[tx$type == "CDS"]$CDS, ".CDS")
  mcols(tx)[tx$type == "UTR","ID"] <- paste0(tx[tx$type == "UTR"]$UTR, ".UTR")
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

annots$Oki <- gff2txdb("../Annotations/OKI2018_I69.v2/OKI2018_I69.v2.gm.gff", OKI2018_I69)
annots$Osa <- gff2txdb("../Annotations/OSKA2016v1.9/OSKA2016v1.9.gm.gff",     OSKA2016v1.9)
annots$Bar <- gff2txdb("../Annotations/Bar2_p4.Flye/Bar2_p4.Flye.gm.gff", Bar2_p4)
annots$Kum <- gff2txdb("../Annotations/KUM-M3-7f/KUM-M3-7f.gm.gff", KUM_M3)
# annots$Aom <- gff2txdb("../Annotations/AOM-5-5f/AOM-5-5f.gm.gff", NULL)
annots$Nor <- gff2txdb_Norway("../Annotations/OdB3/Oikopleura_annot_v1.0.gff", OdB3)
```

Orthogroups
-----------

This assumes that `/bucket/LuscombeU/common/Breakpoints/Orthologues/OKI+BAR+OSA+Ciona+Bla_lp_msa_raxml_ng/OrthoFinder/Results_Jun28/Phylogenetic_Hierarchical_Orthogroups/N1.tsv`
was copied in the package's root directory.

```{r load_orthogroups}
hogs <- load_one_to_ones("../N1.tsv")
hogs
```

Test on Bar - OSKA
==================

Preapre a GBreaks object
------------------------

```{r Bar_Osk}
Bar_Osk <- load_one_to_ones("../N1.tsv", c("Bar2_p4.Flye.prot.lp","OSKA2016v1.9.prot.lp"))
Bar_Osk

IDs2GRanges <- function (IDs, annot, prefix = "") {
  IDs <- IDs |> sub(pat = prefix, rep = "")
  names(annot) <- annot$tx_name
  gr <- granges(annot[IDs])
  strand(gr) <- "*"
  gr
}

gb <- IDs2GRanges(Bar_Osk$Bar2_p4.Flye.prot.lp, transcripts(annots$Bar), prefix = "Bar2_p4.Flye.")
gb$query <- IDs2GRanges(Bar_Osk$OSKA2016v1.9.prot.lp, transcripts(annots$Osa), prefix = "OSKA2016v1.9.")
gb <- GenomicBreaks:::GBreaks(gb)
gb <- sort(gb)
```

Problem: some annotations overlap
---------------------------------

```{r fix_Bar_Osk}
transcripts(annots$Bar) |> length()
transcripts(annots$Bar) |> reduce(min = 0) |> length()

flagOverlaps <- function(gr) {
  # Find overlaps
  ov <- findOverlaps(gr)
  # Index of self hits in the gr object
  idx <- queryHits(ov[!isSelfHit(ov)])
  # Convert to Boolean indexing the gr object
  seq_along(gr) %in% idx
}

# Show examples
gb[flagOverlaps(gb)]
gb$query[flagOverlaps(gb$query)]

# Remove
gb <- gb[!flagOverlaps(gb)]
gb <- gb[!flagOverlaps(gb$query)]
```

Coalesce syntenic blocks and annotate them
------------------------------------------

```{r coalesce}
# Coalesce
coal <- coalesce_contigs(gb)

# Map annotations to the coalesced blocks
ov <- findOverlaps(coal, gb)

# Extract gene names mapped to each blocks
geneNames <- split(names(gb)[subjectHits(ov)], queryHits(ov))

# Concatenate them as single strings and graft them to the coal object
coal$geneNames <- sapply(geneNames, paste, collapse = ", ")

# Count genes per syntenic block
coal$geneNumber <- sapply(geneNames, length)
```

Quick overview of the results
-----------------------------

```{r quick_overview}
stem(coal$geneNumber)
table(coal$geneNumber)
coal[order(coal$geneNumber)] |> tail(11)
```
