<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Breakpoint Associated Heatmaps — bp_heatmap • GenomicBreaks</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Breakpoint Associated Heatmaps — bp_heatmap"><meta name="description" content="This function generates a heatmap of the specified pattern, over breakpoints
aligned at the centre of the plot. The result can be piped into
heatmaps::smoothHeatmap and then heatmaps::plotHeatmapList or
heatmaps::plotHeatmapMeta"><meta property="og:description" content="This function generates a heatmap of the specified pattern, over breakpoints
aligned at the centre of the plot. The result can be piped into
heatmaps::smoothHeatmap and then heatmaps::plotHeatmapList or
heatmaps::plotHeatmapMeta"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.14.7</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Breakpoint Associated Heatmaps</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/bp_heatmap.R" class="external-link"><code>R/bp_heatmap.R</code></a></small>
      <div class="d-none name"><code>bp_heatmap.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function generates a heatmap of the specified pattern, over breakpoints
aligned at the centre of the plot. The result can be piped into
<code><a href="https://rdrr.io/pkg/heatmaps/man/smoothHeatmap.html" class="external-link">heatmaps::smoothHeatmap</a></code> and then <code><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html" class="external-link">heatmaps::plotHeatmapList</a></code> or
<code><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">heatmaps::plotHeatmapMeta</a></code></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">bp_heatmap</span><span class="op">(</span><span class="va">gr</span>, <span class="va">window</span>, <span class="va">pattern</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gr">gr<a class="anchor" aria-label="anchor" href="#arg-gr"></a></dt>
<dd><p><code>GRanges</code> object containing pairwise alignment</p></dd>


<dt id="arg-window">window<a class="anchor" aria-label="anchor" href="#arg-window"></a></dt>
<dd><p>Range over which heatmap characteristic is plotted.
Breakpoints will be aligned at the center of this.</p></dd>


<dt id="arg-pattern">pattern<a class="anchor" aria-label="anchor" href="#arg-pattern"></a></dt>
<dd><p>Character string of desired pattern/characteristic to be
plotted on heatmap.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Pass other arguments to <code><a href="get_bps.html">get_bps</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a <code><a href="https://rdrr.io/pkg/heatmaps/man/Heatmap.html" class="external-link">heatmaps::Heatmap</a></code> object of <code>pattern</code> around centred breakpoints.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>The <code>GRanges</code> object is expected to have a <em>sequence information</em>
(see <code><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">GenomeInfoDb::seqinfo</a></code>) that allows the retrieval its corresponding <code>BSgenome</code>
object via the <code><a href="https://rdrr.io/pkg/BSgenome/man/available.genomes.html" class="external-link">BSgenome::getBSgenome</a></code> function.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other plot functions:
<code><a href="bp_pair_analysis.html">bp_pair_analysis</a>()</code>,
<code><a href="feature_coverage.html">feature_coverage</a>()</code>,
<code><a href="makeOxfordPlots.html">makeOxfordPlots</a>()</code>,
<code><a href="plotApairOfChrs.html">plotApairOfChrs</a>()</code></p>
<p>Other heatmap functions:
<code><a href="bp_pair_analysis.html">bp_pair_analysis</a>()</code>,
<code><a href="feature_coverage.html">feature_coverage</a>()</code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Charlotte West</p>
<p>Charles Plessy</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># The plot makes no sense, but that is the best example I have at the moment.</span></span></span>
<span class="r-in"><span><span class="va">exdata_Sac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/SacCer3__SacPar.gff3.gz"</span>, package <span class="op">=</span> <span class="st">"GenomicBreaks"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="load_genomic_breaks.html">load_genomic_breaks</a></span><span class="op">(</span><span class="va">exdata_Sac</span>, <span class="fu">BSgenome.Scerevisiae.UCSC.sacCer3</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Scerevisiae.UCSC.sacCer3/man/package.html" class="external-link">Scerevisiae</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">bp_heatmap</span><span class="op">(</span><span class="va">gb</span>, <span class="fl">200</span>, <span class="st">'GC'</span>, dir <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu">heatmaps</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/smoothHeatmap.html" class="external-link">smoothHeatmap</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">heatmaps</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html" class="external-link">plotHeatmapList</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘Biostrings’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:GenomicBreaks’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pairwiseAlignment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     strsplit</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating kernel density...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> plotting heatmap GC</span>
<span class="r-plt img"><img src="bp_heatmap-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

