<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Bridge regions — bridgeRegions • GenomicBreaks</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Bridge regions — bridgeRegions"><meta property="og:description" content="Maps unaligned regions of the target to the query genome when they are
flanked by colinear regions."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicBreaks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.13.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/GenomicBreaks.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/StructuralVariants.html">Structural Variants</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/oist/GenomicBreaks/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bridge regions</h1>
    <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/HEAD/R/bridgeRegions.R" class="external-link"><code>R/bridgeRegions.R</code></a></small>
    <div class="hidden name"><code>bridgeRegions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Maps unaligned regions of the <em>target</em> to the <em>query</em> genome when they are
flanked by colinear regions.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">bridgeRegions</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>gb</dt>
<dd><p>A <code><a href="GBreaks-class.html">GBreaks</a></code> object.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Returns a new <code>GBreaks</code> object of shorter length.</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>Because some aligned regions can be directly adjacent (no gaps), the
returned <code>GBreaks</code> object may contain ranges of width zero, where the <em>start</em>
coordinate is 1 nucleotide higher than the <em>end</em> coordinate.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Bridge regions have also been called “simultaneous gaps”
in the comparison of the mouse and human genomes by Kent WJ, Baertsch R,
Hinrichs A, Miller W, Haussler D. (<em>Evolution's cauldron: duplication,
deletion, and rearrangement in the mouse and human genomes.</em> Proc Natl Acad
Sci U S A. 2003;100(20):11484-11489. doi:10.1073/pnas.1932072100)</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other Colinearity functions: 
<code><a href="GOC.html">GOC</a>()</code>,
<code><a href="chain_contigs.html">chain_contigs</a>()</code>,
<code><a href="coalesce_contigs.html">coalesce_contigs</a>()</code>,
<code><a href="dist2next.html">dist2next</a>()</code>,
<code><a href="filterColinearRegions.html">filterColinearRegions</a>()</code>,
<code><a href="flagColinearAlignments.html">flagColinearAlignments</a>()</code>,
<code><a href="flagPairs.html">flagPairs</a>()</code></p>
<p>Other modifier functions: 
<code><a href="coalesce_contigs.html">coalesce_contigs</a>()</code>,
<code><a href="forceSeqLengths.html">forceSeqLengths</a>()</code>,
<code><a href="guessSeqLengths.html">guessSeqLengths</a>()</code>,
<code><a href="mergeSeqLevels.html">mergeSeqLevels</a>()</code>,
<code><a href="reverse.html">reverse</a>()</code>,
<code><a href="swap.html">swap</a>()</code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Charles Plessy</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">exampleColinear5</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-190      + | chrB:100-190</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   200-290      + | chrB:200-290</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   300-390      + | chrB:300-390</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chrA   400-490      + | chrB:400-490</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chrA   500-590      + | chrB:500-590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span class="fu">bridgeRegions</span><span class="op">(</span><span class="va">exampleColinear5</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 4 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   191-199      * | chrB:191-199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   291-299      * | chrB:291-299</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   391-399      * | chrB:391-399</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chrA   491-499      * | chrB:491-499</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Note the zero-width ranges when aligned regions are directly adjacent.</span></span>
<span class="r-in"><span class="va">exampleColinear3</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrB:201-300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span class="fu">bridgeRegions</span><span class="op">(</span><span class="va">exampleColinear3</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 2 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   201-200      * | chrB:201-200</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   301-300      * | chrB:301-300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

