<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Flag successive entries of a GBreaks object — flagPairs • GenomicBreaks</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Flag successive entries of a GBreaks object — flagPairs"><meta name="description" content="Scans a sorted GBreaks object and issues a flag describing the relation
between the current entry and the next entry."><meta property="og:description" content="Scans a sorted GBreaks object and issues a flag describing the relation
between the current entry and the next entry."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.14.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Flag successive entries of a <code>GBreaks</code> object</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/flagPairs.R" class="external-link"><code>R/flagPairs.R</code></a></small>
      <div class="d-none name"><code>flagPairs.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Scans a sorted <code><a href="GBreaks-class.html">GBreaks</a></code> object and issues a flag describing the relation
between the current entry and the next entry.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">flagPairs</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gb">gb<a class="anchor" aria-label="anchor" href="#arg-gb"></a></dt>
<dd><p>A <code><a href="GBreaks-class.html">GBreaks()</a></code> object.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns the <code>GBreaks</code> object with an extra <code>pairs</code> metadata column.
This <code>pairs</code> column is a factor of all flags, so that <code>table(gb$pairs)</code>
always returns a vector of the same length, reporting <code>0</code> when necessary.
This allows to easily aggregate results from multiple objects.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Because the object is sorted, the next entry on the <em>target</em> genome is by
definition following the current one unless the end of the sequence feature
(contig, scaffold, …) is reached.  The two <em>target</em> ranges can be on the
same strand or on opposite strands.  No difference is made between <code>+/-</code>
and <code>-/+</code> orientations because it will be arbitrary unless the sequence
features of both genomes are oriented, as in the case of comparing
chromosomal assemblies of related species.</p>
<p>Depending on whether the <em>query</em> ranges precede or follow each other, or
are not next to each other, and depending on the strand of the alignments,
a flag is issued on the current entry.  <code>Col</code> signals colinearity with the
next entry, <code>Inv</code> an inversion on either entry, <code>Flp</code> signals that the order
of the entries is as if one hand jumped over the other one.  When the <em>query</em>
ranges are not next to each other, a <code>Scr</code> flag, for <em>scrambled</em>, is issued.
Lastly, <code>Bnd</code> (for <em>boundary</em>) signals that there is no pair to analyse
because the current entry is the last one for the current sequence level on
the <em>target</em> genome.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Flagging functions:
<code><a href="flagAll.html">flagAll</a>()</code>,
<code><a href="flagColinearAlignments.html">flagColinearAlignments</a>()</code>,
<code><a href="flagDoubleInversions.html">flagDoubleInversions</a>()</code>,
<code><a href="flagInversions.html">flagInversions</a>()</code>,
<code><a href="flagLongShort.html">flagLongShort</a>()</code>,
<code><a href="flagTranslocations.html">flagTranslocations</a>()</code></p>
<p>Other Inversion functions:
<code><a href="filterDoubleInversions.html">filterDoubleInversions</a>()</code>,
<code><a href="filterInversions.html">filterInversions</a>()</code>,
<code><a href="flagDoubleInversions.html">flagDoubleInversions</a>()</code>,
<code><a href="flagInversions.html">flagInversions</a>()</code>,
<code><a href="flipInversions.html">flipInversions</a>()</code>,
<code><a href="leftInversionGaps.html">leftInversionGaps</a>()</code>,
<code><a href="showInversions.html">showInversions</a>()</code></p>
<p>Other Colinearity functions:
<code><a href="GOC.html">GOC</a>()</code>,
<code><a href="bridgeRegions.html">bridgeRegions</a>()</code>,
<code><a href="chain_contigs.html">chain_contigs</a>()</code>,
<code><a href="coalesce_contigs.html">coalesce_contigs</a>()</code>,
<code><a href="dist2next.html">dist2next</a>()</code>,
<code><a href="filterColinearRegions.html">filterColinearRegions</a>()</code>,
<code><a href="flagColinearAlignments.html">flagColinearAlignments</a>()</code></p>
<p>Other Structural variants:
<code><a href="StructuralVariants.html">StructuralVariants</a></code>,
<code><a href="flagDoubleInversions.html">flagDoubleInversions</a>()</code>,
<code><a href="flagInversions.html">flagInversions</a>()</code>,
<code><a href="flagTranslocations.html">flagTranslocations</a>()</code>,
<code><a href="plotApairOfChrs.html">plotApairOfChrs</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">flagPairs</span><span class="op">(</span><span class="va">exampleInversion</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query    pairs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-190      + | chrB:100-190      Inv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   210-291      - | chrB:210-291      Inv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400      Bnd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagPairs</span><span class="op">(</span><span class="va">exampleTranslocation</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query    pairs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200      Scr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrC:201-300      Scr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400      Bnd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># How the pair classes are defined:</span></span></span>
<span class="r-in"><span><span class="fu">GenomicBreaks</span><span class="fu">:::</span><span class="va">allPossiblePairClasses</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Qnext Strnd Snext pair     paste</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   next     +     +  Col  next + +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   prev     +     +  Flp  prev + +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  other     +     +  Scr other + +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   next     -     +  Inv  next - +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   prev     -     +  Inv  prev - +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  other     -     +  Scr other - +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7   next     +     -  Inv  next + -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8   prev     +     -  Inv  prev + -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  other     +     -  Scr other + -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10  next     -     -  Flp  next - -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11  prev     -     -  Col  prev - -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12 other     -     -  Scr other - -</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

