<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Flag translocations — flagTranslocations • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Flag translocations — flagTranslocations"><meta name="description" content="Flag ranges that start a triplet that would be colinear if the central pair
were colinear, and that are not an inversion."><meta property="og:description" content="Flag ranges that start a triplet that would be colinear if the central pair
were colinear, and that are not an inversion."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.21.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Flag translocations</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/Translocations.R" class="external-link"><code>R/Translocations.R</code></a></small>
      <div class="d-none name"><code>flagTranslocations.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Flag ranges that start a triplet that would be colinear if the central pair
were colinear, and that are not an inversion.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">gb</span>, tol <span class="op">=</span> <span class="cn">Inf</span>, both <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gb">gb<a class="anchor" aria-label="anchor" href="#arg-gb"></a></dt>
<dd><p>A <code><a href="GBreaks-class.html">GBreaks()</a></code> object.</p></dd>


<dt id="arg-tol">tol<a class="anchor" aria-label="anchor" href="#arg-tol"></a></dt>
<dd><p>Tolerance window for the distance between two ranges.</p></dd>


<dt id="arg-both">both<a class="anchor" aria-label="anchor" href="#arg-both"></a></dt>
<dd><p>Flag both the <em>target</em> and <em>query</em> ranges.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns the <code>GBreaks</code> object with an extra <code>tra</code> metadata column.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The result of the flagging differs if the <em>target</em> and <em>query</em> ranges are
swapped.  Therefore, by default this function will search for translocations
on both cases, and flag both ranges.</p>
<p>Here is a trivial example of a translocation:</p>
<p></p><div class="sourceCode"><pre><code><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>┌──────────────┬──────────────┬──────────────┐</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>│ chrA<span class="sc">:</span><span class="dv">101-200</span> │ chrA<span class="sc">:</span><span class="dv">201-300</span> │ chrA<span class="sc">:</span><span class="dv">301-400</span> │ (Target genome)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>└──────────────┴──────────────┴──────────────┘</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>       <span class="sc">+</span>               <span class="sc">+</span>             <span class="sc">+</span>         (Alignment direction)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>┌──────────────┬──────────────┬──────────────┐</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>│ chrB<span class="sc">:</span><span class="dv">101-200</span> │ chrC<span class="sc">:</span><span class="dv">201-300</span> │ chrB<span class="sc">:</span><span class="dv">301-400</span> │ (Query genome)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>└──────────────┴──────────────┴──────────────┘</span></code></pre><p></p></div>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Flagging functions:
<code><a href="flagAll.html">flagAll()</a></code>,
<code><a href="flagColinearAlignments.html">flagColinearAlignments()</a></code>,
<code><a href="flagDoubleInversions.html">flagDoubleInversions()</a></code>,
<code><a href="flagInversions.html">flagInversions()</a></code>,
<code><a href="flagLongShort.html">flagLongShort()</a></code>,
<code><a href="flagPairs.html">flagPairs()</a></code>,
<code><a href="flagTwinInversions.html">flagTwinInversions()</a></code></p>
<p>Other Translocation functions:
<code><a href="filterTranslocations.html">filterTranslocations()</a></code>,
<code><a href="removeTranslocations.html">removeTranslocations()</a></code>,
<code><a href="showTranslocations.html">showTranslocations()</a></code></p>
<p>Other Structural variants:
<code><a href="StructuralVariants.html">StructuralVariants</a></code>,
<code><a href="flagDoubleInversions.html">flagDoubleInversions()</a></code>,
<code><a href="flagInversions.html">flagInversions()</a></code>,
<code><a href="flagPairs.html">flagPairs()</a></code>,
<code><a href="flagTwinInversions.html">flagTwinInversions()</a></code>,
<code><a href="plotApairOfChrs.html">plotApairOfChrs()</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">exampleTranslocation</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query   tra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrC:201-300 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">exampleTranslocation2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query   tra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      - | chrC:201-300 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu"><a href="plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleTranslocation</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="flagTranslocations-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">exampleDeletion</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query   tra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrC:401-500 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:201-300 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">exampleInsertion</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query   tra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrB:301-400 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrC   401-500      + | chrB:201-300 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 2 sequences from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">exampleInsertion</span>  <span class="op">|&gt;</span> <span class="fu"><a href="swap.html">swap</a></span><span class="op">(</span>sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query   tra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrB   100-200      + | chrA:100-200  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrB   201-300      + | chrC:401-500 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrB   301-400      + | chrA:201-300 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">exampleInsertion</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="swap.html">swap</a></span><span class="op">(</span>sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |   tra        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;Rle&gt;    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrB   100-200      + |  TRUE chrA:100-200</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrB   201-300      + | FALSE chrC:401-500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrB   301-400      + | FALSE chrA:201-300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleDeletion</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query   tra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   201-300      - | chrB:201-300  TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   301-400      - | chrC:401-500 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   401-501      - | chrB:100-200 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">exampleInversion</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query   tra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-190      + | chrB:100-190 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   210-291      - | chrB:210-291 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagTranslocations</span><span class="op">(</span><span class="va">exampleColinear3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query   tra</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrB:201-300 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400 FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

