<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Flag colinear alignments — flagColinearAlignments • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Flag colinear alignments — flagColinearAlignments"><meta name="description" content="Flags alignments that are colinear with the next one in sequence order.
The flag is added to the first alignment.  Colinearity is defined by the fact
that the next alignment on the target genome is the same as the next
alignment on the query genome, with strand information being taken into
account."><meta property="og:description" content="Flags alignments that are colinear with the next one in sequence order.
The flag is added to the first alignment.  Colinearity is defined by the fact
that the next alignment on the target genome is the same as the next
alignment on the query genome, with strand information being taken into
account."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.16.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Flag colinear alignments</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/flagColinearAlignments.R" class="external-link"><code>R/flagColinearAlignments.R</code></a></small>
      <div class="d-none name"><code>flagColinearAlignments.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Flags alignments that are colinear with the next one in sequence order.
The flag is added to the first alignment.  Colinearity is defined by the fact
that the next alignment on the <em>target</em> genome is the same as the next
alignment on the <em>query</em> genome, with strand information being taken into
account.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">flagColinearAlignments</span><span class="op">(</span><span class="va">gb</span>, tol <span class="op">=</span> <span class="cn">Inf</span>, minwidth <span class="op">=</span> <span class="fl">0</span>, details <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gb">gb<a class="anchor" aria-label="anchor" href="#arg-gb"></a></dt>
<dd><p><code><a href="GBreaks-class.html">GBreaks</a></code> object of the pairwise alignment.</p></dd>


<dt id="arg-tol">tol<a class="anchor" aria-label="anchor" href="#arg-tol"></a></dt>
<dd><p>Unaligned regions larger than this <em>tolerance</em> threshold will
interrupt colinearity.</p></dd>


<dt id="arg-minwidth">minwidth<a class="anchor" aria-label="anchor" href="#arg-minwidth"></a></dt>
<dd><p>Remove the intervals whose width smaller than this value.</p></dd>


<dt id="arg-details">details<a class="anchor" aria-label="anchor" href="#arg-details"></a></dt>
<dd><p>Returns more metadata columns if <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a modified <code>GBreaks</code> object with a new <code>colinear</code> metadata
column indicating if an alignment is colinear with the next one.  If the
<code>details</code> option is set to <code>TRUE</code>, it will also output the relative position
of the following and previous alignments on the <em>target</em> and <em>query</em> genomes
(<code>tfoll</code>, <code>tprev</code>, <code>qfoll</code>, <code>qprev</code>), a partial colinearity flag for each
genome (<code>t_col</code> and <code>q_col</code>), and the distance to the next alignment on
each genome (<code>tdist</code> and <code>qdist</code>).</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Internally, <code>flagColinearAlignments()</code> uses the <code><a href="https://rdrr.io/pkg/GenomicRanges/man/nearest-methods.html" class="external-link">GenomicRanges::precede</a></code>
and <code><a href="https://rdrr.io/pkg/GenomicRanges/man/nearest-methods.html" class="external-link">GenomicRanges::follow</a></code> functions functions to determine what is the
<em>next</em>.  For a given range, these functions return the index position of the
range it precedes or follows, or <code>NA</code> as the first range follows nothing and
the last range precedes nothing.  See the examples for details.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>The flags are only valid as long as the Genomic Breaks object is not
modified by removing alignments or sorting them in a different order.</p>
<p>Pay attention that if the <code>mindwidth</code> option is passed, some intervals
are discarded from the returned object.  This parameter might be removed in
the future if too confusing or useless.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Flagging functions:
<code><a href="flagAll.html">flagAll</a>()</code>,
<code><a href="flagDoubleInversions.html">flagDoubleInversions</a>()</code>,
<code><a href="flagInversions.html">flagInversions</a>()</code>,
<code><a href="flagLongShort.html">flagLongShort</a>()</code>,
<code><a href="flagPairs.html">flagPairs</a>()</code>,
<code><a href="flagTranslocations.html">flagTranslocations</a>()</code></p>
<p>Other Colinearity functions:
<code><a href="GOC.html">GOC</a>()</code>,
<code><a href="bridgeRegions.html">bridgeRegions</a>()</code>,
<code><a href="chain_contigs.html">chain_contigs</a>()</code>,
<code><a href="coalesce_contigs.html">coalesce_contigs</a>()</code>,
<code><a href="dist2next.html">dist2next</a>()</code>,
<code><a href="filterColinearRegions.html">filterColinearRegions</a>()</code>,
<code><a href="flagPairs.html">flagPairs</a>()</code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Charlotte West</p>
<p>Charles Plessy</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">flagColinearAlignments</span><span class="op">(</span><span class="va">exampleColinear3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query  colinear</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrB:201-300      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagColinearAlignments</span><span class="op">(</span><span class="va">exampleColinear3</span>, details <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 10 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query     tfoll     qfoll     tprev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200         1         1      &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrB:201-300         1         1        -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400      &lt;NA&gt;      &lt;NA&gt;        -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           qprev     t_col     q_col     tdist     qdist  colinear</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;integer&gt; &lt;logical&gt; &lt;logical&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;logical&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]      &lt;NA&gt;      TRUE      TRUE         1         1      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]        -1      TRUE      TRUE         1         1      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]        -1      &lt;NA&gt;      &lt;NA&gt;       Inf       Inf     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Target range [1] precedes target range [2]</span></span></span>
<span class="r-in"><span><span class="fu">precede</span><span class="op">(</span><span class="va">exampleColinear3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  2  3 NA</span>
<span class="r-in"><span><span class="co"># Query range [1] precedes query range [2]</span></span></span>
<span class="r-in"><span><span class="fu">precede</span><span class="op">(</span><span class="va">exampleColinear3</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  2  3 NA</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Ranges on the minus strand</span></span></span>
<span class="r-in"><span><span class="va">gb2</span> <span class="op">&lt;-</span> <span class="va">exampleColinear3</span> <span class="op">|&gt;</span> <span class="fu"><a href="reverse.html">reverse</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">flagColinearAlignments</span><span class="op">(</span><span class="va">gb2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query  colinear</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   201-300      - | chrB:301-400      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   301-400      - | chrB:201-300      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   401-501      - | chrB:100-200     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Target range [1] follows target range [2]</span></span></span>
<span class="r-in"><span><span class="fu">follow</span><span class="op">(</span><span class="va">gb2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  2  3 NA</span>
<span class="r-in"><span><span class="co"># Or, ignoring strand, target range [1] precedes target range [2]</span></span></span>
<span class="r-in"><span><span class="fu">precede</span><span class="op">(</span><span class="va">gb2</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  2  3 NA</span>
<span class="r-in"><span><span class="co"># Query range [1] follows query range [2]</span></span></span>
<span class="r-in"><span><span class="fu">follow</span><span class="op">(</span><span class="va">gb2</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  2  3 NA</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Colinearity check on strandless objects</span></span></span>
<span class="r-in"><span><span class="va">gb3</span> <span class="op">&lt;-</span> <span class="va">exampleColinear3</span></span></span>
<span class="r-in"><span><span class="va">gb4</span> <span class="op">&lt;-</span> <span class="va">gb2</span></span></span>
<span class="r-in"><span><span class="fu">strand</span><span class="op">(</span><span class="va">gb4</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">strand</span><span class="op">(</span><span class="va">gb3</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"*"</span></span></span>
<span class="r-in"><span><span class="fu">flagColinearAlignments</span><span class="op">(</span><span class="va">gb3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query  colinear</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      * | chrB:100-200      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      * | chrB:201-300      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      * | chrB:301-400     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="fu">flagColinearAlignments</span><span class="op">(</span><span class="va">gb4</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query  colinear</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   201-300      * | chrB:301-400      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   301-400      * | chrB:201-300      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   401-501      * | chrB:100-200     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Ranges that should not coalesce because they are not</span></span></span>
<span class="r-in"><span><span class="co"># ordered properly</span></span></span>
<span class="r-in"><span><span class="fu">flagColinearAlignments</span><span class="op">(</span><span class="va">exampleNotColinear</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 2 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query  colinear</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-150      + | chrB:201-251     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   251-300      + |  chrB:50-100     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

