<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>The GenomicBreaks class — GBreaks-class • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The GenomicBreaks class — GBreaks-class"><meta name="description" content="The GBreaks class is a simple wrapper to the GenomicRanges::GRanges class."><meta property="og:description" content="The GBreaks class is a simple wrapper to the GenomicRanges::GRanges class."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.21.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>The GenomicBreaks class</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/GenomicBreaks.R" class="external-link"><code>R/GenomicBreaks.R</code></a></small>
      <div class="d-none name"><code>GBreaks-class.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The <code>GBreaks</code> class is a simple wrapper to the <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GenomicRanges::GRanges</a></code> class.</p>
    </div>


    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Aligned sequences of the <em>target</em> genome are represented as the main
ranges of the <code>GRanges</code> object, and their counterparts in the <em>query</em> genome
are represented as a <code>GRanges</code> object sored in the the metadata column
<code>query</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Bioconductor API functions:
<code><a href="getSeq.html">getSeq()</a></code>,
<code><a href="pairwiseAlignment.html">pairwiseAlignment()</a></code>,
<code><a href="range.html">range_GBreaks()</a></code>,
<code><a href="reverse.html">reverse()</a></code>,
<code><a href="subsetByOverlaps.html">subsetByOverlaps_GBreaks()</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="load_genomic_breaks.html">load_genomic_breaks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/SacCer3__SacPar.gff3.gz"</span>, package <span class="op">=</span> <span class="st">"GenomicBreaks"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gb</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 505 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames        ranges strand |     score                       query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;                   &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chrM       191-813      + |       984     NC_018044.1:61854-62304</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chrM     6313-9737      - |      9007     NC_018044.1:62854-65728</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chrM   12828-13986      + |      2733     NC_018044.1:22453-23578</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chrM   16435-18988      + |     12413     NC_018044.1:23579-26133</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chrM   20505-20650      + |       774     NC_018044.1:26134-26279</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...           ...    ... .       ...                         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [501]  chrXIII 769669-808526      + |    152331   NC_047499.1:762339-801278</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [502]  chrXIII 809285-837102      + |    115538   NC_047499.1:801658-829480</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [503]  chrXIII 837914-912582      + |    303512   NC_047499.1:830467-905221</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [504]  chrXIII 912605-917242      + |     13091   NC_047499.1:905399-910082</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [505]  chrXIII 919705-922192      + |      6580 NC_047501.1:1054871-1057371</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 17 sequences from an unspecified genome; no seqlengths</span>
<span class="r-in"><span><span class="va">gb</span><span class="op">$</span><span class="va">query</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 505 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            seqnames          ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1] NC_018044.1     61854-62304      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2] NC_018044.1     62854-65728      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3] NC_018044.1     22453-23578      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4] NC_018044.1     23579-26133      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5] NC_018044.1     26134-26279      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...         ...             ...    ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [501] NC_047499.1   762339-801278      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [502] NC_047499.1   801658-829480      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [503] NC_047499.1   830467-905221      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [504] NC_047499.1   905399-910082      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [505] NC_047501.1 1054871-1057371      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 17 sequences from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

