<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ GenomicBreaks</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.14.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="plotting-functions">Plotting functions<a class="anchor" aria-label="anchor" href="#plotting-functions"></a></h2>
      
      <p class="section-desc"></p><p>Graphical display of pairwise alignments</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="bp_heatmap.html">bp_heatmap()</a></code> 
        </dt>
        <dd>Breakpoint Associated Heatmaps</dd>
      </dl><dl><dt>
          
          <code><a href="bp_pair_analysis.html">bp_pair_analysis()</a></code> 
        </dt>
        <dd>Breakpoint Pairwise Comparison using Reference Genome as Coordinate System</dd>
      </dl><dl><dt>
          
          <code><a href="feature_coverage.html">feature_coverage()</a></code> 
        </dt>
        <dd>Feature coverage heatmap</dd>
      </dl><dl><dt>
          
          <code><a href="makeOxfordPlots.html">makeOxfordPlots()</a></code> 
        </dt>
        <dd>Make Oxford Plots</dd>
      </dl><dl><dt>
          
          <code><a href="plotApairOfChrs.html">plotApairOfChrs()</a></code> 
        </dt>
        <dd>Plots a pair of chromosomes from a <code>GBreaks</code> object.</dd>
      </dl></div><div class="section level2">
      <h2 id="heatmap-functions">Heatmap Functions<a class="anchor" aria-label="anchor" href="#heatmap-functions"></a></h2>
      
      <p class="section-desc"></p><p>Functions that return a <code>Heatmap</code> object from the <code>heatmaps</code> package.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="bp_heatmap.html">bp_heatmap()</a></code> 
        </dt>
        <dd>Breakpoint Associated Heatmaps</dd>
      </dl><dl><dt>
          
          <code><a href="bp_pair_analysis.html">bp_pair_analysis()</a></code> 
        </dt>
        <dd>Breakpoint Pairwise Comparison using Reference Genome as Coordinate System</dd>
      </dl><dl><dt>
          
          <code><a href="feature_coverage.html">feature_coverage()</a></code> 
        </dt>
        <dd>Feature coverage heatmap</dd>
      </dl></div><div class="section level2">
      <h2 id="genoplotr-functions">genoPlotR functions<a class="anchor" aria-label="anchor" href="#genoplotr-functions"></a></h2>
      
      <p class="section-desc"></p><p>Functions related to the <code>genoPlotR</code> package that we use for pair chromosome plots.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="gb2comp.html">gb2comp()</a></code> 
        </dt>
        <dd>Convert <code>GBreaks</code> object to a <code>comparison</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="gb2dna_seg.html">gb2dna_seg()</a></code> 
        </dt>
        <dd>Convert <code>GBreaks</code> object to a list of <code>dna_seg</code> objects</dd>
      </dl><dl><dt>
          
          <code><a href="gb2xlim.html">gb2xlim()</a></code> 
        </dt>
        <dd>GBreaks to xlim</dd>
      </dl><dl><dt>
          
          <code><a href="gr2dna_seg.html">gr2dna_seg()</a></code> 
        </dt>
        <dd>Convert <code>GRanges</code> object to <code>dna_seg</code> format</dd>
      </dl><dl><dt>
          
          <code><a href="plotApairOfChrs.html">plotApairOfChrs()</a></code> 
        </dt>
        <dd>Plots a pair of chromosomes from a <code>GBreaks</code> object.</dd>
      </dl></div><div class="section level2">
      <h2 id="distance-functions">Distance functions<a class="anchor" aria-label="anchor" href="#distance-functions"></a></h2>
      
      <p class="section-desc"></p><p>Various ways to estimate the similarity between two genomes.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="GOC.html">GOC()</a></code> 
        </dt>
        <dd>Gene Order Conservation</dd>
      </dl><dl><dt>
          
          <code><a href="correlation_index.html">correlation_index()</a></code> 
        </dt>
        <dd>Index representing how correlated syntenic regions are</dd>
      </dl><dl><dt>
          
          <code><a href="strand_randomisation_index.html">strand_randomisation_index()</a></code> 
        </dt>
        <dd>Index measuring how 'random' the alignment strand has become.</dd>
      </dl><dl><dt>
          
          <code><a href="synteny_index.html">synteny_index()</a></code> 
        </dt>
        <dd>Index representing how 'syntenic' two genomes are</dd>
      </dl></div><div class="section level2">
      <h2 id="colinearity-functions">Colinearity functions<a class="anchor" aria-label="anchor" href="#colinearity-functions"></a></h2>
      
      <p class="section-desc"></p><p>Functions to analyse colinearity and synteny.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="GOC.html">GOC()</a></code> 
        </dt>
        <dd>Gene Order Conservation</dd>
      </dl><dl><dt>
          
          <code><a href="bridgeRegions.html">bridgeRegions()</a></code> 
        </dt>
        <dd>Bridge regions</dd>
      </dl><dl><dt>
          
          <code><a href="chain_contigs.html">chain_contigs()</a></code> 
        </dt>
        <dd>Chain contigs in pairwise alignments</dd>
      </dl><dl><dt>
          
          <code><a href="coalesce_contigs.html">coalesce_contigs()</a></code> 
        </dt>
        <dd>Coalesce Pairwise Alignments</dd>
      </dl><dl><dt>
          
          <code><a href="dist2next.html">dist2next()</a></code> 
        </dt>
        <dd>Distances to next ranges</dd>
      </dl><dl><dt>
          
          <code><a href="filterColinearRegions.html">filterColinearRegions()</a></code> 
        </dt>
        <dd>Filter colinear regions</dd>
      </dl><dl><dt>
          
          <code><a href="flagColinearAlignments.html">flagColinearAlignments()</a></code> 
        </dt>
        <dd>Flag colinear alignments</dd>
      </dl><dl><dt>
          
          <code><a href="flagPairs.html">flagPairs()</a></code> 
        </dt>
        <dd>Flag successive entries of a <code>GBreaks</code> object</dd>
      </dl></div><div class="section level2">
      <h2 id="flags-and-filters">Flags and filters<a class="anchor" aria-label="anchor" href="#flags-and-filters"></a></h2>
      
      <p class="section-desc"></p><p>Functions to flag patterns of alignments and filter them.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="flagAll.html">flagAll()</a></code> 
        </dt>
        <dd>Compute all flags</dd>
      </dl><dl><dt>
          
          <code><a href="flagColinearAlignments.html">flagColinearAlignments()</a></code> 
        </dt>
        <dd>Flag colinear alignments</dd>
      </dl><dl><dt>
          
          <code><a href="flagDoubleInversions.html">flagDoubleInversions()</a></code> 
        </dt>
        <dd>Flag double inversions</dd>
      </dl><dl><dt>
          
          <code><a href="flagInversions.html">flagInversions()</a></code> 
        </dt>
        <dd>Flag inversions</dd>
      </dl><dl><dt>
          
          <code><a href="flagLongShort.html">flagLongShort()</a></code> 
        </dt>
        <dd>Flag long and short arms</dd>
      </dl><dl><dt>
          
          <code><a href="flagPairs.html">flagPairs()</a></code> 
        </dt>
        <dd>Flag successive entries of a <code>GBreaks</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="flagTranslocations.html">flagTranslocations()</a></code> 
        </dt>
        <dd>Flag translocations</dd>
      </dl><dl><dt>
          
          <code><a href="filterColinearRegions.html">filterColinearRegions()</a></code> 
        </dt>
        <dd>Filter colinear regions</dd>
      </dl><dl><dt>
          
          <code><a href="filterInversions.html">filterInversions()</a></code> 
        </dt>
        <dd>Extract central blocks in inversions</dd>
      </dl><dl><dt>
          
          <code><a href="filterTranslocations.html">filterTranslocations()</a></code> 
        </dt>
        <dd>Extract central blocks in translocations</dd>
      </dl><dl><dt>
          
          <code><a href="showInversions.html">showInversions()</a></code> 
        </dt>
        <dd>Show inversions and their flanking blocks.</dd>
      </dl><dl><dt>
          
          <code><a href="showTranslocations.html">showTranslocations()</a></code> 
        </dt>
        <dd>Show translocations and their flanking blocks.</dd>
      </dl></div><div class="section level2">
      <h2 id="bioconductor-api-functions">Bioconductor API functions<a class="anchor" aria-label="anchor" href="#bioconductor-api-functions"></a></h2>
      
      <p class="section-desc"></p><p>Functions that provide compatibility with GRanges, etc.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="GBreaks-class.html">GBreaks-class</a></code> <code><a href="GBreaks-class.html">GBreaks</a></code> 
        </dt>
        <dd>The GenomicBreaks class</dd>
      </dl><dl><dt>
          
          <code><a href="as.html">as</a></code> 
        </dt>
        <dd>Conversion from <code><a href="https://rdrr.io/pkg/CNEr/man/Axt-class.html" class="external-link">CNEr::Axt</a></code> objects</dd>
      </dl><dl><dt>
          
          <code><a href="getSeq.html">getSeq()</a></code> 
        </dt>
        <dd>Get genomic sequences</dd>
      </dl><dl><dt>
          
          <code><a href="pairwiseAlignment.html">pairwiseAlignment()</a></code> 
        </dt>
        <dd>Pairwise alignment of genomic ranges</dd>
      </dl><dl><dt>
          
          <code><a href="range.html">range_GBreaks()</a></code> <code><a href="range.html">range(<i>&lt;GBreaks&gt;</i>)</a></code> 
        </dt>
        <dd>Range method for <code>GBreaks</code> objects</dd>
      </dl><dl><dt>
          
          <code><a href="reverse.html">reverse()</a></code> 
        </dt>
        <dd>Reverse genomic ranges</dd>
      </dl><dl><dt>
          
          <code><a href="subsetByOverlaps.html">subsetByOverlaps_GBreaks()</a></code> <code><a href="subsetByOverlaps.html">subsetByOverlaps(<i>&lt;GBreaks&gt;</i>,<i>&lt;GBreaks&gt;</i>)</a></code> 
        </dt>
        <dd>subsetByOverlaps method for <code>GBreaks</code> objects</dd>
      </dl></div><div class="section level2">
      <h2 id="motifier-functions">Motifier functions<a class="anchor" aria-label="anchor" href="#motifier-functions"></a></h2>
      
      <p class="section-desc"></p><p>Functions that take <code>GBreaks</code> objects and return them modified.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="bridgeRegions.html">bridgeRegions()</a></code> 
        </dt>
        <dd>Bridge regions</dd>
      </dl><dl><dt>
          
          <code><a href="coalesce_contigs.html">coalesce_contigs()</a></code> 
        </dt>
        <dd>Coalesce Pairwise Alignments</dd>
      </dl><dl><dt>
          
          <code><a href="forceSeqLengths.html">forceSeqLengths()</a></code> 
        </dt>
        <dd>Set sequence lengths if there were none</dd>
      </dl><dl><dt>
          
          <code><a href="guessSeqLengths.html">guessSeqLengths()</a></code> 
        </dt>
        <dd>Guesstimate seqlevel lengths</dd>
      </dl><dl><dt>
          
          <code><a href="mergeSeqLevels.html">mergeSeqLevels()</a></code> 
        </dt>
        <dd>Merge seqlevels in a larger one</dd>
      </dl><dl><dt>
          
          <code><a href="reverse.html">reverse()</a></code> 
        </dt>
        <dd>Reverse genomic ranges</dd>
      </dl><dl><dt>
          
          <code><a href="swap.html">swap()</a></code> 
        </dt>
        <dd>Swap target and query genomes</dd>
      </dl></div><div class="section level2">
      <h2 id="reducing-functions">Reducing functions<a class="anchor" aria-label="anchor" href="#reducing-functions"></a></h2>
      
      <p class="section-desc"></p><p>Functions that take <code>GBreaks</code> objects and return simpler <code>GRanges</code> objects.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="cleanGaps.html">cleanGaps()</a></code> 
        </dt>
        <dd>Gaps between ranges</dd>
      </dl><dl><dt>
          
          <code><a href="get_bps.html">get_bps()</a></code> 
        </dt>
        <dd>Get breakpoints</dd>
      </dl></div><div class="section level2">
      <h2 id="other-functions">Other functions<a class="anchor" aria-label="anchor" href="#other-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="StructuralVariants.html">StructuralVariants</a></code> <code><a href="StructuralVariants.html">exampleColinear</a></code> <code><a href="StructuralVariants.html">exampleNotColinear</a></code> <code><a href="StructuralVariants.html">exampleColinear3</a></code> <code><a href="StructuralVariants.html">exampleColinear5</a></code> <code><a href="StructuralVariants.html">exampleInversion</a></code> <code><a href="StructuralVariants.html">exampleDoubleInversion</a></code> <code><a href="StructuralVariants.html">exampleInversion5uncollapsed</a></code> <code><a href="StructuralVariants.html">exampleTranslocation</a></code> <code><a href="StructuralVariants.html">exampleTranslocation2</a></code> <code><a href="StructuralVariants.html">exampleInsertion</a></code> <code><a href="StructuralVariants.html">exampleDeletion</a></code> 
        </dt>
        <dd>Structural Variants</dd>
      </dl><dl><dt>
          
          <code><a href="align_scaf_plot_2.html">align_scaf_plot_2()</a></code> 
        </dt>
        <dd>Compare Alignments over two Reference scaffolds</dd>
      </dl><dl><dt>
          
          <code><a href="bp_coverage.html">bp_coverage()</a></code> 
        </dt>
        <dd>Coverage over Breakpoints</dd>
      </dl><dl><dt>
          
          <code><a href="feature_coverage.html">feature_coverage()</a></code> 
        </dt>
        <dd>Feature coverage heatmap</dd>
      </dl><dl><dt>
          
          <code><a href="flipStrandNames.html">flipStrandNames()</a></code> 
        </dt>
        <dd>Reverse order and flip strand</dd>
      </dl><dl><dt>
          
          <code><a href="gr2dna_seg.html">gr2dna_seg()</a></code> 
        </dt>
        <dd>Convert <code>GRanges</code> object to <code>dna_seg</code> format</dd>
      </dl><dl><dt>
          
          <code><a href="int_frag.html">int_frag()</a></code> 
        </dt>
        <dd>Integrated intersection of both query and reference subsets on a continuous alignment</dd>
      </dl><dl><dt>
          
          <code><a href="inv2UCSCData.html">inv2UCSCData()</a></code> 
        </dt>
        <dd>Represent inversions as UCSCData objects</dd>
      </dl><dl><dt>
          
          <code><a href="isSorted.html">isSorted()</a></code> 
        </dt>
        <dd>Test if a <code>GBreaks</code> object is sorted</dd>
      </dl><dl><dt>
          
          <code><a href="leftInversionGaps.html">leftInversionGaps()</a></code> 
        </dt>
        <dd>Isolate the left-side gaps in inversions</dd>
      </dl><dl><dt>
          
          <code><a href="load_genomic_breaks.html">load_genomic_breaks()</a></code> 
        </dt>
        <dd>Load pairwise genome alignments</dd>
      </dl><dl><dt>
          
          <code><a href="longestMatchesInTarget.html">longestMatchesInTarget()</a></code> 
        </dt>
        <dd>Find longest matches from target to query genome.</dd>
      </dl><dl><dt>
          
          <code><a href="makeInversion.chr.html">makeInversion.chr()</a></code> 
        </dt>
        <dd>Make an inversion</dd>
      </dl><dl><dt>
          
          <code><a href="master_bp_analysis.html">master_bp_analysis()</a></code> 
        </dt>
        <dd>Master Breakpoint Analysis</dd>
      </dl><dl><dt>
          
          <code><a href="orderQuerySeqLevels.html">orderQuerySeqLevels()</a></code> 
        </dt>
        <dd>Ordering permutation for <em>query</em> relative to <em>target</em></dd>
      </dl><dl><dt>
          
          <code><a href="readMAF.html">readMAF()</a></code> 
        </dt>
        <dd>Read a MAF file</dd>
      </dl><dl><dt>
          
          <code><a href="scaf_align_plot.html">scaf_align_plot()</a></code> 
        </dt>
        <dd>Plot Alignment of Single Reference Scaffold</dd>
      </dl><dl><dt>
          
          <code><a href="strandNames.html">strandNames()</a></code> 
        </dt>
        <dd>Sequence name with strand information</dd>
      </dl><dl><dt>
          
          <code><a href="tan_bp.html">tan_bp()</a></code> 
        </dt>
        <dd>Breakpoint proximity to Tandem Repeats</dd>
      </dl><dl><dt>
          
          <code><a href="zipWithNext.html">zipWithNext()</a></code> 
        </dt>
        <dd>Zip elements with the next ones</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

