<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ GenomicBreaks</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicBreaks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.13.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/GenomicBreaks.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/StructuralVariants.html">Structural Variants</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/oist/GenomicBreaks/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="plotting-functions">Plotting functions <a href="#plotting-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Graphical display of pairwise alignments</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="bp_heatmap.html">bp_heatmap()</a></code> </p>
        </td>
        <td><p>Breakpoint Associated Heatmaps</p></td>
      </tr><tr><td>
          <p><code><a href="bp_pair_analysis.html">bp_pair_analysis()</a></code> </p>
        </td>
        <td><p>Breakpoint Pairwise Comparison using Reference Genome as Coordinate System</p></td>
      </tr><tr><td>
          <p><code><a href="feature_coverage.html">feature_coverage()</a></code> </p>
        </td>
        <td><p>Feature coverage heatmap</p></td>
      </tr><tr><td>
          <p><code><a href="makeOxfordPlots.html">makeOxfordPlots()</a></code> </p>
        </td>
        <td><p>Make Oxford Plots</p></td>
      </tr><tr><td>
          <p><code><a href="plotApairOfChrs.html">plotApairOfChrs()</a></code> </p>
        </td>
        <td><p>Plots a pair of chromosomes from a <code>GBreaks</code> object.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="heatmap-functions">Heatmap Functions <a href="#heatmap-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that return a <code>Heatmap</code> object from the <code>heatmaps</code> package.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="bp_heatmap.html">bp_heatmap()</a></code> </p>
        </td>
        <td><p>Breakpoint Associated Heatmaps</p></td>
      </tr><tr><td>
          <p><code><a href="bp_pair_analysis.html">bp_pair_analysis()</a></code> </p>
        </td>
        <td><p>Breakpoint Pairwise Comparison using Reference Genome as Coordinate System</p></td>
      </tr><tr><td>
          <p><code><a href="feature_coverage.html">feature_coverage()</a></code> </p>
        </td>
        <td><p>Feature coverage heatmap</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="genoplotr-functions">genoPlotR functions <a href="#genoplotr-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions related to the <code>genoPlotR</code> package that we use for pair chromosome plots.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="gb2comp.html">gb2comp()</a></code> </p>
        </td>
        <td><p>Convert <code>GBreaks</code> object to a <code>comparison</code> object</p></td>
      </tr><tr><td>
          <p><code><a href="gb2dna_seg.html">gb2dna_seg()</a></code> </p>
        </td>
        <td><p>Convert <code>GBreaks</code> object to a list of <code>dna_seg</code> objects</p></td>
      </tr><tr><td>
          <p><code><a href="gb2xlim.html">gb2xlim()</a></code> </p>
        </td>
        <td><p>GBreaks to xlim</p></td>
      </tr><tr><td>
          <p><code><a href="gr2dna_seg.html">gr2dna_seg()</a></code> </p>
        </td>
        <td><p>Convert <code>GRanges</code> object to <code>dna_seg</code> format</p></td>
      </tr><tr><td>
          <p><code><a href="plotApairOfChrs.html">plotApairOfChrs()</a></code> </p>
        </td>
        <td><p>Plots a pair of chromosomes from a <code>GBreaks</code> object.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="distance-functions">Distance functions <a href="#distance-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Various ways to estimate the similarity between two genomes.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="GOC.html">GOC()</a></code> </p>
        </td>
        <td><p>Gene Order Conservation</p></td>
      </tr><tr><td>
          <p><code><a href="correlation_index.html">correlation_index()</a></code> </p>
        </td>
        <td><p>Index representing how correlated syntenic regions are</p></td>
      </tr><tr><td>
          <p><code><a href="strand_randomisation_index.html">strand_randomisation_index()</a></code> </p>
        </td>
        <td><p>Index measuring how 'random' the alignment strand has become.</p></td>
      </tr><tr><td>
          <p><code><a href="synteny_index.html">synteny_index()</a></code> </p>
        </td>
        <td><p>Index representing how 'syntenic' two genomes are</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="colinearity-functions">Colinearity functions <a href="#colinearity-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions to analyse colinearity and synteny.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="GOC.html">GOC()</a></code> </p>
        </td>
        <td><p>Gene Order Conservation</p></td>
      </tr><tr><td>
          <p><code><a href="bridgeRegions.html">bridgeRegions()</a></code> </p>
        </td>
        <td><p>Bridge regions</p></td>
      </tr><tr><td>
          <p><code><a href="chain_contigs.html">chain_contigs()</a></code> </p>
        </td>
        <td><p>Chain contigs in pairwise alignments</p></td>
      </tr><tr><td>
          <p><code><a href="coalesce_contigs.html">coalesce_contigs()</a></code> </p>
        </td>
        <td><p>Coalesce Pairwise Alignments</p></td>
      </tr><tr><td>
          <p><code><a href="dist2next.html">dist2next()</a></code> </p>
        </td>
        <td><p>Distances to next ranges</p></td>
      </tr><tr><td>
          <p><code><a href="filterColinearRegions.html">filterColinearRegions()</a></code> </p>
        </td>
        <td><p>Filter colinear regions</p></td>
      </tr><tr><td>
          <p><code><a href="flagColinearAlignments.html">flagColinearAlignments()</a></code> </p>
        </td>
        <td><p>Flag colinear alignments</p></td>
      </tr><tr><td>
          <p><code><a href="flagPairs.html">flagPairs()</a></code> </p>
        </td>
        <td><p>Flag successive entries of a <code>GBreaks</code> object</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="flags-and-filters">Flags and filters <a href="#flags-and-filters" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions to flag patterns of alignments and filter them.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="flagAll.html">flagAll()</a></code> </p>
        </td>
        <td><p>Compute all flags</p></td>
      </tr><tr><td>
          <p><code><a href="flagColinearAlignments.html">flagColinearAlignments()</a></code> </p>
        </td>
        <td><p>Flag colinear alignments</p></td>
      </tr><tr><td>
          <p><code><a href="flagDoubleInversions.html">flagDoubleInversions()</a></code> </p>
        </td>
        <td><p>Flag double inversions</p></td>
      </tr><tr><td>
          <p><code><a href="flagInversions.html">flagInversions()</a></code> </p>
        </td>
        <td><p>Flag inversions</p></td>
      </tr><tr><td>
          <p><code><a href="flagLongShort.html">flagLongShort()</a></code> </p>
        </td>
        <td><p>Flag long and short arms</p></td>
      </tr><tr><td>
          <p><code><a href="flagPairs.html">flagPairs()</a></code> </p>
        </td>
        <td><p>Flag successive entries of a <code>GBreaks</code> object</p></td>
      </tr><tr><td>
          <p><code><a href="flagTranslocations.html">flagTranslocations()</a></code> </p>
        </td>
        <td><p>Flag translocations</p></td>
      </tr><tr><td>
          <p><code><a href="filterColinearRegions.html">filterColinearRegions()</a></code> </p>
        </td>
        <td><p>Filter colinear regions</p></td>
      </tr><tr><td>
          <p><code><a href="filterInversions.html">filterInversions()</a></code> </p>
        </td>
        <td><p>Extract central blocks in inversions</p></td>
      </tr><tr><td>
          <p><code><a href="filterTranslocations.html">filterTranslocations()</a></code> </p>
        </td>
        <td><p>Extract central blocks in translocations</p></td>
      </tr><tr><td>
          <p><code><a href="showInversions.html">showInversions()</a></code> </p>
        </td>
        <td><p>Show inversions and their flanking blocks.</p></td>
      </tr><tr><td>
          <p><code><a href="showTranslocations.html">showTranslocations()</a></code> </p>
        </td>
        <td><p>Show translocations and their flanking blocks.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="bioconductor-api-functions">Bioconductor API functions <a href="#bioconductor-api-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that provide compatibility with GRanges, etc.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="GBreaks-class.html">GBreaks-class</a></code> </p>
        </td>
        <td><p>The GenomicBreaks class</p></td>
      </tr><tr><td>
          <p><code><a href="as.html">as</a></code> </p>
        </td>
        <td><p>Conversion from <code><a href="https://rdrr.io/pkg/CNEr/man/Axt-class.html" class="external-link">CNEr::Axt</a></code> objects</p></td>
      </tr><tr><td>
          <p><code><a href="getSeq.html">getSeq()</a></code> </p>
        </td>
        <td><p>Get genomic sequences</p></td>
      </tr><tr><td>
          <p><code><a href="pairwiseAlignment.html">pairwiseAlignment()</a></code> </p>
        </td>
        <td><p>Pairwise alignment of genomic ranges</p></td>
      </tr><tr><td>
          <p><code><a href="range.html">range_GBreaks()</a></code> <code><a href="range.html">range(<i>&lt;GBreaks&gt;</i>)</a></code> </p>
        </td>
        <td><p>Range method for <code>GBreaks</code> objects</p></td>
      </tr><tr><td>
          <p><code><a href="reverse.html">reverse()</a></code> </p>
        </td>
        <td><p>Reverse genomic ranges</p></td>
      </tr><tr><td>
          <p><code><a href="subsetByOverlaps.html">subsetByOverlaps_GBreaks()</a></code> <code><a href="subsetByOverlaps.html">subsetByOverlaps(<i>&lt;GBreaks&gt;</i>,<i>&lt;GBreaks&gt;</i>)</a></code> </p>
        </td>
        <td><p>subsetByOverlaps method for <code>GBreaks</code> objects</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="motifier-functions">Motifier functions <a href="#motifier-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that take <code>GBreaks</code> objects and return them modified.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="bridgeRegions.html">bridgeRegions()</a></code> </p>
        </td>
        <td><p>Bridge regions</p></td>
      </tr><tr><td>
          <p><code><a href="coalesce_contigs.html">coalesce_contigs()</a></code> </p>
        </td>
        <td><p>Coalesce Pairwise Alignments</p></td>
      </tr><tr><td>
          <p><code><a href="forceSeqLengths.html">forceSeqLengths()</a></code> </p>
        </td>
        <td><p>Set sequence lengths if there were none</p></td>
      </tr><tr><td>
          <p><code><a href="guessSeqLengths.html">guessSeqLengths()</a></code> </p>
        </td>
        <td><p>Guesstimate seqlevel lengths</p></td>
      </tr><tr><td>
          <p><code><a href="mergeSeqLevels.html">mergeSeqLevels()</a></code> </p>
        </td>
        <td><p>Merge seqlevels in a larger one</p></td>
      </tr><tr><td>
          <p><code><a href="reverse.html">reverse()</a></code> </p>
        </td>
        <td><p>Reverse genomic ranges</p></td>
      </tr><tr><td>
          <p><code><a href="swap.html">swap()</a></code> </p>
        </td>
        <td><p>Swap target and query genomes</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="reducing-functions">Reducing functions <a href="#reducing-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that take <code>GBreaks</code> objects and return simpler <code>GRanges</code> objects.</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="cleanGaps.html">cleanGaps()</a></code> </p>
        </td>
        <td><p>Gaps between ranges</p></td>
      </tr><tr><td>
          <p><code><a href="get_bps.html">get_bps()</a></code> </p>
        </td>
        <td><p>Get breakpoints</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="other-functions">Other functions <a href="#other-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="StructuralVariants.html">StructuralVariants</a></code> </p>
        </td>
        <td><p>Structural Variants</p></td>
      </tr><tr><td>
          <p><code><a href="align_scaf_plot_2.html">align_scaf_plot_2()</a></code> </p>
        </td>
        <td><p>Compare Alignments over two Reference scaffolds</p></td>
      </tr><tr><td>
          <p><code><a href="bp_coverage.html">bp_coverage()</a></code> </p>
        </td>
        <td><p>Coverage over Breakpoints</p></td>
      </tr><tr><td>
          <p><code><a href="feature_coverage.html">feature_coverage()</a></code> </p>
        </td>
        <td><p>Feature coverage heatmap</p></td>
      </tr><tr><td>
          <p><code><a href="flipStrandNames.html">flipStrandNames()</a></code> </p>
        </td>
        <td><p>Reverse order and flip strand</p></td>
      </tr><tr><td>
          <p><code><a href="gr2dna_seg.html">gr2dna_seg()</a></code> </p>
        </td>
        <td><p>Convert <code>GRanges</code> object to <code>dna_seg</code> format</p></td>
      </tr><tr><td>
          <p><code><a href="int_frag.html">int_frag()</a></code> </p>
        </td>
        <td><p>Integrated intersection of both query and reference subsets on a continuous alignment</p></td>
      </tr><tr><td>
          <p><code><a href="inv2UCSCData.html">inv2UCSCData()</a></code> </p>
        </td>
        <td><p>Represent inversions as UCSCData objects</p></td>
      </tr><tr><td>
          <p><code><a href="isSorted.html">isSorted()</a></code> </p>
        </td>
        <td><p>Test if a <code>GBreaks</code> object is sorted</p></td>
      </tr><tr><td>
          <p><code><a href="leftInversionGaps.html">leftInversionGaps()</a></code> </p>
        </td>
        <td><p>Isolate the left-side gaps in inversions</p></td>
      </tr><tr><td>
          <p><code><a href="load_genomic_breaks.html">load_genomic_breaks()</a></code> </p>
        </td>
        <td><p>Load pairwise genome alignments</p></td>
      </tr><tr><td>
          <p><code><a href="longestMatchesInTarget.html">longestMatchesInTarget()</a></code> </p>
        </td>
        <td><p>Find longest matches from target to query genome.</p></td>
      </tr><tr><td>
          <p><code><a href="makeInversion.chr.html">makeInversion.chr()</a></code> </p>
        </td>
        <td><p>Make an inversion</p></td>
      </tr><tr><td>
          <p><code><a href="master_bp_analysis.html">master_bp_analysis()</a></code> </p>
        </td>
        <td><p>Master Breakpoint Analysis</p></td>
      </tr><tr><td>
          <p><code><a href="orderQuerySeqLevels.html">orderQuerySeqLevels()</a></code> </p>
        </td>
        <td><p>Ordering permutation for <em>query</em> relative to <em>target</em></p></td>
      </tr><tr><td>
          <p><code><a href="scaf_align_plot.html">scaf_align_plot()</a></code> </p>
        </td>
        <td><p>Plot Alignment of Single Reference Scaffold</p></td>
      </tr><tr><td>
          <p><code><a href="strandNames.html">strandNames()</a></code> </p>
        </td>
        <td><p>Sequence name with strand information</p></td>
      </tr><tr><td>
          <p><code><a href="tan_bp.html">tan_bp()</a></code> </p>
        </td>
        <td><p>Breakpoint proximity to Tandem Repeats</p></td>
      </tr><tr><td>
          <p><code><a href="zipWithNext.html">zipWithNext()</a></code> </p>
        </td>
        <td><p>Zip elements with the next ones</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

