<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.18.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="data-loading-functions">Data loading functions<a class="anchor" aria-label="anchor" href="#data-loading-functions"></a></h2>

      <div class="section-desc"><p>Functions to load data</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="load_genomic_breaks.html">load_genomic_breaks()</a></code>

        </dt>
        <dd>Load pairwise genome alignments</dd>
      </dl><dl><dt>

          <code><a href="readMAF.html">readMAF()</a></code>

        </dt>
        <dd>Read a MAF file</dd>
      </dl><dl><dt>

          <code><a href="readTrainFile.html">readTrainFile()</a></code>

        </dt>
        <dd>Extract values from <code>last-train</code> parameter file</dd>
      </dl></div><div class="section level2">
      <h2 id="plotting-functions">Plotting functions<a class="anchor" aria-label="anchor" href="#plotting-functions"></a></h2>

      <div class="section-desc"><p>Graphical display of pairwise alignments</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="bp_heatmap.html">bp_heatmap()</a></code>

        </dt>
        <dd>Breakpoint Associated Heatmaps</dd>
      </dl><dl><dt>

          <code><a href="bp_pair_analysis.html">bp_pair_analysis()</a></code>

        </dt>
        <dd>Breakpoint Pairwise Comparison using Reference Genome as Coordinate System</dd>
      </dl><dl><dt>

          <code><a href="feature_coverage.html">feature_coverage()</a></code>

        </dt>
        <dd>Feature coverage heatmap</dd>
      </dl><dl><dt>

          <code><a href="makeOxfordPlots.html">makeOxfordPlots()</a></code>

        </dt>
        <dd>Make Oxford Plots</dd>
      </dl><dl><dt>

          <code><a href="plotApairOfChrs.html">plotApairOfChrs()</a></code>

        </dt>
        <dd>Plots a pair of chromosomes from a <code>GBreaks</code> object.</dd>
      </dl></div><div class="section level2">
      <h2 id="heatmap-functions">Heatmap Functions<a class="anchor" aria-label="anchor" href="#heatmap-functions"></a></h2>

      <div class="section-desc"><p>Functions that return a <code>Heatmap</code> object from the <code>heatmaps</code> package.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="bp_heatmap.html">bp_heatmap()</a></code>

        </dt>
        <dd>Breakpoint Associated Heatmaps</dd>
      </dl><dl><dt>

          <code><a href="bp_pair_analysis.html">bp_pair_analysis()</a></code>

        </dt>
        <dd>Breakpoint Pairwise Comparison using Reference Genome as Coordinate System</dd>
      </dl><dl><dt>

          <code><a href="feature_coverage.html">feature_coverage()</a></code>

        </dt>
        <dd>Feature coverage heatmap</dd>
      </dl></div><div class="section level2">
      <h2 id="genoplotr-functions">genoPlotR functions<a class="anchor" aria-label="anchor" href="#genoplotr-functions"></a></h2>

      <div class="section-desc"><p>Functions related to the <code>genoPlotR</code> package that we use for pair chromosome plots.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="gb2comp.html">gb2comp()</a></code>

        </dt>
        <dd>Convert <code>GBreaks</code> object to a <code>comparison</code> object</dd>
      </dl><dl><dt>

          <code><a href="gb2dna_seg.html">gb2dna_seg()</a></code>

        </dt>
        <dd>Convert <code>GBreaks</code> object to a list of <code>dna_seg</code> objects</dd>
      </dl><dl><dt>

          <code><a href="gb2xlim.html">gb2xlim()</a></code>

        </dt>
        <dd>GBreaks to xlim</dd>
      </dl><dl><dt>

          <code><a href="gr2dna_seg.html">gr2dna_seg()</a></code>

        </dt>
        <dd>Convert <code>GRanges</code> object to <code>dna_seg</code> format</dd>
      </dl><dl><dt>

          <code><a href="plotApairOfChrs.html">plotApairOfChrs()</a></code>

        </dt>
        <dd>Plots a pair of chromosomes from a <code>GBreaks</code> object.</dd>
      </dl></div><div class="section level2">
      <h2 id="distance-functions">Distance functions<a class="anchor" aria-label="anchor" href="#distance-functions"></a></h2>

      <div class="section-desc"><p>Various ways to estimate the similarity between two genomes.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="F81_distance.html">F81_distance()</a></code>

        </dt>
        <dd>Felsenstein's 1981 Distance</dd>
      </dl><dl><dt>

          <code><a href="GOC.html">GOC()</a></code>

        </dt>
        <dd>Gene Order Conservation</dd>
      </dl><dl><dt>

          <code><a href="JC69_distance.html">JC69_distance()</a></code>

        </dt>
        <dd>Jukes-Cantor 1969 distance</dd>
      </dl><dl><dt>

          <code><a href="K80_distance.html">K80_distance()</a></code>

        </dt>
        <dd>Kimura's 2-parameter distance</dd>
      </dl><dl><dt>

          <code><a href="P_distance.html">P_distance()</a></code> <code><a href="P_distance.html">P_error()</a></code>

        </dt>
        <dd>Percent difference Distance</dd>
      </dl><dl><dt>

          <code><a href="T92_distance.html">T92_distance()</a></code>

        </dt>
        <dd>Tamura 1992 distance</dd>
      </dl><dl><dt>

          <code><a href="correlation_index.html">correlation_index()</a></code>

        </dt>
        <dd>Index representing how correlated syntenic regions are</dd>
      </dl><dl><dt>

          <code><a href="karyotype_index.html">karyotype_index()</a></code>

        </dt>
        <dd>Index representing how the karyotype changes between chromosomes of two genomes</dd>
      </dl><dl><dt>

          <code><a href="slidingWindow.html">slidingWindow()</a></code>

        </dt>
        <dd>Sliding window</dd>
      </dl><dl><dt>

          <code><a href="strand_randomisation_index.html">strand_randomisation_index()</a></code>

        </dt>
        <dd>Index measuring how 'random' the alignment strand has become.</dd>
      </dl><dl><dt>

          <code><a href="synteny_index.html">synteny_index()</a></code>

        </dt>
        <dd>Index representing how 'syntenic' two genomes are</dd>
      </dl><dl><dt>

          <code><a href="tau_index.html">tau_index()</a></code>

        </dt>
        <dd>Tau index</dd>
      </dl><dl><dt>

          <code><a href="GCequilibrium.html">GCequilibrium()</a></code>

        </dt>
        <dd>Equilibrium GC fraction (GC*)</dd>
      </dl><dl><dt>

          <code><a href="GCpressure.html">GCpressure()</a></code>

        </dt>
        <dd>GC pressure</dd>
      </dl><dl><dt>

          <code><a href="GCproportion.html">GCproportion()</a></code>

        </dt>
        <dd>Proportion of GC in alignment</dd>
      </dl><dl><dt>

          <code><a href="exampleSubstitutionMatrix.html">exampleSubstitutionMatrix</a></code>

        </dt>
        <dd>Example Substitution Matrices</dd>
      </dl><dl><dt>

          <code><a href="gapProportion.html">gapProportion()</a></code>

        </dt>
        <dd>Proportion of gaps in alignment</dd>
      </dl></div><div class="section level2">
      <h2 id="colinearity-functions">Colinearity functions<a class="anchor" aria-label="anchor" href="#colinearity-functions"></a></h2>

      <div class="section-desc"><p>Functions to analyse colinearity and synteny.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="GOC.html">GOC()</a></code>

        </dt>
        <dd>Gene Order Conservation</dd>
      </dl><dl><dt>

          <code><a href="bridgeRegions.html">bridgeRegions()</a></code>

        </dt>
        <dd>Bridge regions</dd>
      </dl><dl><dt>

          <code><a href="chain_contigs.html">chain_contigs()</a></code>

        </dt>
        <dd>Chain contigs in pairwise alignments</dd>
      </dl><dl><dt>

          <code><a href="coalesce_contigs.html">coalesce_contigs()</a></code>

        </dt>
        <dd>Coalesce Pairwise Alignments</dd>
      </dl><dl><dt>

          <code><a href="dist2next.html">dist2next()</a></code>

        </dt>
        <dd>Distances to next ranges</dd>
      </dl><dl><dt>

          <code><a href="filterColinearRegions.html">filterColinearRegions()</a></code>

        </dt>
        <dd>Filter colinear regions</dd>
      </dl><dl><dt>

          <code><a href="flagColinearAlignments.html">flagColinearAlignments()</a></code>

        </dt>
        <dd>Flag colinear alignments</dd>
      </dl><dl><dt>

          <code><a href="flagPairs.html">flagPairs()</a></code>

        </dt>
        <dd>Flag successive entries of a <code>GBreaks</code> object</dd>
      </dl></div><div class="section level2">
      <h2 id="flags-and-filters">Flags and filters<a class="anchor" aria-label="anchor" href="#flags-and-filters"></a></h2>

      <div class="section-desc"><p>Functions to flag patterns of alignments and filter them.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="flagAll.html">flagAll()</a></code>

        </dt>
        <dd>Compute all flags</dd>
      </dl><dl><dt>

          <code><a href="flagColinearAlignments.html">flagColinearAlignments()</a></code>

        </dt>
        <dd>Flag colinear alignments</dd>
      </dl><dl><dt>

          <code><a href="flagDoubleInversions.html">flagDoubleInversions()</a></code>

        </dt>
        <dd>Flag double inversions</dd>
      </dl><dl><dt>

          <code><a href="flagInversions.html">flagInversions()</a></code>

        </dt>
        <dd>Flag inversions</dd>
      </dl><dl><dt>

          <code><a href="flagLongShort.html">flagLongShort()</a></code>

        </dt>
        <dd>Flag long and short arms</dd>
      </dl><dl><dt>

          <code><a href="flagPairs.html">flagPairs()</a></code>

        </dt>
        <dd>Flag successive entries of a <code>GBreaks</code> object</dd>
      </dl><dl><dt>

          <code><a href="flagTranslocations.html">flagTranslocations()</a></code>

        </dt>
        <dd>Flag translocations</dd>
      </dl><dl><dt>

          <code><a href="filterColinearRegions.html">filterColinearRegions()</a></code>

        </dt>
        <dd>Filter colinear regions</dd>
      </dl><dl><dt>

          <code><a href="filterDoubleInversions.html">filterDoubleInversions()</a></code>

        </dt>
        <dd>Extract central blocks in double inversions</dd>
      </dl><dl><dt>

          <code><a href="filterInversions.html">filterInversions()</a></code>

        </dt>
        <dd>Extract central blocks in inversions</dd>
      </dl><dl><dt>

          <code><a href="filterTranslocations.html">filterTranslocations()</a></code>

        </dt>
        <dd>Extract central blocks in translocations</dd>
      </dl><dl><dt>

          <code><a href="showInversions.html">showInversions()</a></code>

        </dt>
        <dd>Show inversions and their flanking blocks.</dd>
      </dl><dl><dt>

          <code><a href="showTranslocations.html">showTranslocations()</a></code>

        </dt>
        <dd>Show translocations and their flanking blocks.</dd>
      </dl></div><div class="section level2">
      <h2 id="bioconductor-api-functions">Bioconductor API functions<a class="anchor" aria-label="anchor" href="#bioconductor-api-functions"></a></h2>

      <div class="section-desc"><p>Functions that provide compatibility with GRanges, etc.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="GBreaks-class.html">GBreaks-class</a></code> <code><a href="GBreaks-class.html">GBreaks</a></code>

        </dt>
        <dd>The GenomicBreaks class</dd>
      </dl><dl><dt>

          <code><a href="getSeq.html">getSeq()</a></code>

        </dt>
        <dd>Get genomic sequences</dd>
      </dl><dl><dt>

          <code><a href="pairwiseAlignment.html">pairwiseAlignment()</a></code>

        </dt>
        <dd>Pairwise alignment of genomic ranges</dd>
      </dl><dl><dt>

          <code><a href="range.html">range_GBreaks()</a></code> <code><a href="range.html">range(<i>&lt;GBreaks&gt;</i>)</a></code>

        </dt>
        <dd>Range method for <code>GBreaks</code> objects</dd>
      </dl><dl><dt>

          <code><a href="reverse.html">reverse()</a></code>

        </dt>
        <dd>Reverse genomic ranges</dd>
      </dl><dl><dt>

          <code><a href="subsetByOverlaps.html">subsetByOverlaps_GBreaks()</a></code> <code><a href="subsetByOverlaps.html">subsetByOverlaps(<i>&lt;GBreaks&gt;</i>,<i>&lt;GBreaks&gt;</i>)</a></code>

        </dt>
        <dd>subsetByOverlaps method for <code>GBreaks</code> objects</dd>
      </dl></div><div class="section level2">
      <h2 id="motifier-functions">Motifier functions<a class="anchor" aria-label="anchor" href="#motifier-functions"></a></h2>

      <div class="section-desc"><p>Functions that take <code>GBreaks</code> objects and return them modified.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="bridgeRegions.html">bridgeRegions()</a></code>

        </dt>
        <dd>Bridge regions</dd>
      </dl><dl><dt>

          <code><a href="coalesce_contigs.html">coalesce_contigs()</a></code>

        </dt>
        <dd>Coalesce Pairwise Alignments</dd>
      </dl><dl><dt>

          <code><a href="flipInversions.html">flipInversions()</a></code>

        </dt>
        <dd>Flip strand of inversions</dd>
      </dl><dl><dt>

          <code><a href="forceSeqLengths.html">forceSeqLengths()</a></code>

        </dt>
        <dd>Set sequence lengths if there were none</dd>
      </dl><dl><dt>

          <code><a href="guessSeqLengths.html">guessSeqLengths()</a></code>

        </dt>
        <dd>Guesstimate seqlevel lengths</dd>
      </dl><dl><dt>

          <code><a href="keepLongestPair.html">keepLongestPair()</a></code>

        </dt>
        <dd>Keep the longest pair of features</dd>
      </dl><dl><dt>

          <code><a href="matchPairs.html">matchPairs()</a></code>

        </dt>
        <dd>Match pairs of sequences</dd>
      </dl><dl><dt>

          <code><a href="mergeSeqLevels.html">mergeSeqLevels()</a></code> <code><a href="mergeSeqLevels.html">mergeSeqLevels_to_DF()</a></code>

        </dt>
        <dd>Merge seqlevels in a larger one</dd>
      </dl><dl><dt>

          <code><a href="removeDoubleInversions.html">removeDoubleInversions()</a></code>

        </dt>
        <dd>Remove double inversions</dd>
      </dl><dl><dt>

          <code><a href="removeInversions.html">removeInversions()</a></code>

        </dt>
        <dd>Remove inversions</dd>
      </dl><dl><dt>

          <code><a href="removeTranslocations.html">removeTranslocations()</a></code>

        </dt>
        <dd>Remove translocations</dd>
      </dl><dl><dt>

          <code><a href="reverse.html">reverse()</a></code>

        </dt>
        <dd>Reverse genomic ranges</dd>
      </dl><dl><dt>

          <code><a href="splitSeqLevel.html">splitSeqLevel()</a></code>

        </dt>
        <dd>Split a seqlevel in two pieces</dd>
      </dl><dl><dt>

          <code><a href="swap.html">swap()</a></code>

        </dt>
        <dd>Swap target and query genomes</dd>
      </dl></div><div class="section level2">
      <h2 id="reducing-functions">Reducing functions<a class="anchor" aria-label="anchor" href="#reducing-functions"></a></h2>

      <div class="section-desc"><p>Functions that take <code>GBreaks</code> objects and return simpler <code>GRanges</code> objects.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="cleanGaps.html">cleanGaps()</a></code>

        </dt>
        <dd>Gaps between ranges</dd>
      </dl><dl><dt>

          <code><a href="get_bps.html">get_bps()</a></code>

        </dt>
        <dd>Get breakpoints</dd>
      </dl><dl><dt>

          <code><a href="wholeGenomeClassification.html">wholeGenomeClassification()</a></code>

        </dt>
        <dd>Whole-genome classification object</dd>
      </dl></div><div class="section level2">
      <h2 id="converter-functions">Converter functions<a class="anchor" aria-label="anchor" href="#converter-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="GBreaksToMatrix.html">GBreaksToMatrix()</a></code>

        </dt>
        <dd>Convert to binned matrix of hits</dd>
      </dl><dl><dt>

          <code><a href="gb2comp.html">gb2comp()</a></code>

        </dt>
        <dd>Convert <code>GBreaks</code> object to a <code>comparison</code> object</dd>
      </dl><dl><dt>

          <code><a href="gb2dna_seg.html">gb2dna_seg()</a></code>

        </dt>
        <dd>Convert <code>GBreaks</code> object to a list of <code>dna_seg</code> objects</dd>
      </dl><dl><dt>

          <code><a href="gb2xlim.html">gb2xlim()</a></code>

        </dt>
        <dd>GBreaks to xlim</dd>
      </dl><dl><dt>

          <code><a href="gr2dna_seg.html">gr2dna_seg()</a></code>

        </dt>
        <dd>Convert <code>GRanges</code> object to <code>dna_seg</code> format</dd>
      </dl></div><div class="section level2">
      <h2 id="scaffolding-functions">Scaffolding functions<a class="anchor" aria-label="anchor" href="#scaffolding-functions"></a></h2>

      <div class="section-desc"><p>Functions to aid trivial scaffolding</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="flipStrandNames.html">flipStrandNames()</a></code>

        </dt>
        <dd>Reverse order and flip strand</dd>
      </dl><dl><dt>

          <code><a href="longestMatchesInTarget.html">longestMatchesInTarget()</a></code>

        </dt>
        <dd>Find longest matches from target to query genome.</dd>
      </dl><dl><dt>

          <code><a href="mergeSeqLevels.html">mergeSeqLevels()</a></code> <code><a href="mergeSeqLevels.html">mergeSeqLevels_to_DF()</a></code>

        </dt>
        <dd>Merge seqlevels in a larger one</dd>
      </dl><dl><dt>

          <code><a href="scaffoldByFlipAndMerge.html">scaffoldByFlipAndMerge()</a></code>

        </dt>
        <dd>Guided scaffolding</dd>
      </dl><dl><dt>

          <code><a href="splitSeqLevel.html">splitSeqLevel()</a></code>

        </dt>
        <dd>Split a seqlevel in two pieces</dd>
      </dl><dl><dt>

          <code><a href="strandNames.html">strandNames()</a></code>

        </dt>
        <dd>Sequence name with strand information</dd>
      </dl></div><div class="section level2">
      <h2 id="other-functions">Other functions<a class="anchor" aria-label="anchor" href="#other-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="StructuralVariants.html">StructuralVariants</a></code> <code><a href="StructuralVariants.html">exampleColinear</a></code> <code><a href="StructuralVariants.html">exampleNotColinear</a></code> <code><a href="StructuralVariants.html">exampleColinear3</a></code> <code><a href="StructuralVariants.html">exampleColinear5</a></code> <code><a href="StructuralVariants.html">exampleInversion</a></code> <code><a href="StructuralVariants.html">exampleNestedInversions</a></code> <code><a href="StructuralVariants.html">exampleInversion5uncollapsed</a></code> <code><a href="StructuralVariants.html">exampleDoubleInversion1</a></code> <code><a href="StructuralVariants.html">exampleDoubleInversion2</a></code> <code><a href="StructuralVariants.html">exampleDoubleInversion3</a></code> <code><a href="StructuralVariants.html">exampleDoubleInversion4</a></code> <code><a href="StructuralVariants.html">exampleTranslocation</a></code> <code><a href="StructuralVariants.html">exampleTranslocation2</a></code> <code><a href="StructuralVariants.html">exampleInsertion</a></code> <code><a href="StructuralVariants.html">exampleDeletion</a></code>

        </dt>
        <dd>Structural Variants</dd>
      </dl><dl><dt>

          <code><a href="align_scaf_plot_2.html">align_scaf_plot_2()</a></code>

        </dt>
        <dd>Compare Alignments over two Reference scaffolds</dd>
      </dl><dl><dt>

          <code><a href="feature_coverage.html">feature_coverage()</a></code>

        </dt>
        <dd>Feature coverage heatmap</dd>
      </dl><dl><dt>

          <code><a href="int_frag.html">int_frag()</a></code>

        </dt>
        <dd>Integrated intersection of both query and reference subsets on a continuous alignment</dd>
      </dl><dl><dt>

          <code><a href="inv2UCSCData.html">inv2UCSCData()</a></code>

        </dt>
        <dd>Represent inversions as UCSCData objects</dd>
      </dl><dl><dt>

          <code><a href="isSorted.html">isSorted()</a></code>

        </dt>
        <dd>Test if a <code>GBreaks</code> object is sorted</dd>
      </dl><dl><dt>

          <code><a href="leftInversionGaps.html">leftInversionGaps()</a></code>

        </dt>
        <dd>Isolate the left-side gaps in inversions</dd>
      </dl><dl><dt>

          <code><a href="makeInversion.chr.html">makeInversion.chr()</a></code>

        </dt>
        <dd>Make an inversion</dd>
      </dl><dl><dt>

          <code><a href="orderQuerySeqLevels.html">orderQuerySeqLevels()</a></code> <code><a href="orderQuerySeqLevels.html">orderQuerySeqLevels_DF_GR()</a></code>

        </dt>
        <dd>Ordering permutation for <em>query</em> relative to <em>target</em></dd>
      </dl><dl><dt>

          <code><a href="scaf_align_plot.html">scaf_align_plot()</a></code>

        </dt>
        <dd>Plot Alignment of Single Reference Scaffold</dd>
      </dl><dl><dt>

          <code><a href="tan_bp.html">tan_bp()</a></code>

        </dt>
        <dd>Breakpoint proximity to Tandem Repeats</dd>
      </dl><dl><dt>

          <code><a href="zipWithNext.html">zipWithNext()</a></code>

        </dt>
        <dd>Zip elements with the next ones</dd>
      </dl><dl><dt>

          <code><a href="inversionDistance.html">inversionDistance()</a></code>

        </dt>
        <dd>Inversion Distance</dd>
      </dl><dl><dt>

          <code><a href="permutationVector.html">permutationVector()</a></code>

        </dt>
        <dd>Permutation vector</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

