<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Load pairwise genome alignments — load_genomic_breaks • GenomicBreaks</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Load pairwise genome alignments — load_genomic_breaks"><meta property="og:description" content="Loads alignments of a query genome to a target genome from a text file in
GFF3 format.  By convention, the target genome is the one that was indexed
by the aligner."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicBreaks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.12.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/GenomicBreaks.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/StructuralVariants.html">Structural Variants</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/oist/GenomicBreaks/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Load pairwise genome alignments</h1>
    <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/HEAD/R/load_genomic_breaks.R" class="external-link"><code>R/load_genomic_breaks.R</code></a></small>
    <div class="hidden name"><code>load_genomic_breaks.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Loads alignments of a <em>query</em> genome to a <em>target</em> genome from a text file in
GFF3 format.  By convention, the <em>target</em> genome is the one that was indexed
by the aligner.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">load_genomic_breaks</span><span class="op">(</span>
  <span class="va">file</span>,
  target_bsgenome <span class="op">=</span> <span class="cn">NULL</span>,
  query_bsgenome <span class="op">=</span> <span class="cn">NULL</span>,
  sort <span class="op">=</span> <span class="cn">TRUE</span>,
  type <span class="op">=</span> <span class="st">"match_part"</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>file</dt>
<dd><p>Path to a file in GFF3 format.</p></dd>
<dt>target_bsgenome</dt>
<dd><p>A <code>BSgenome</code> object representing the target genome.</p></dd>
<dt>query_bsgenome</dt>
<dd><p>A <code>BSgenome</code> object representing the query genome.</p></dd>
<dt>sort</dt>
<dd><p>Returns the object sorted, ignoring strand information.</p></dd>
<dt>type</dt>
<dd><p>Sequence ontology term representing an alignment block
(default: <code>match_part</code>).</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Returns a <code><a href="GBreaks-class.html">GBreaks</a></code> object where each element represents a pairwise
alignment block.  The <code>granges</code> part of the object contains the coordinates
on the <em>target</em> genome, and the <code>query</code> metadata column contains the <em>query</em>coordinates in <code>GRanges</code> format.  The <code>seqinfo</code> of each <code>BSgenome</code> object
passed as parameters are copied to the <code>GRanges</code> objects accordingly.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function expects the pairwise alignment to be represented in GFF3 format
in the following way:</p><ul><li><p>Alignments blocks are represented by entries in specific sequence ontology
term in the <em>type</em> column.  Other entries will be discarded.  The default
type is <code>match_part</code>.</p></li>
<li><p>The coordinate system of the file is the one of the <em>target</em> genome.</p></li>
<li><p>The <code>Target</code> tag in the <em>attribute</em> column contains the coordinates of
the match in the <em>query</em> genome.  (Sorry that it is confusing…)</p></li>
<li><p>Stand information is set so that <em>query</em> genome coordinates are always on
the <em>plus</em> strand.</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BSgenome.Odioica.local.OSKA2016"</span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BSgenome.Odioica.local.OKI2018.I69"</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">load_genomic_breaks</span><span class="op">(</span></span>
<span class="r-in">  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/OSKA2016__I69-5.gff3.gz"</span>, package<span class="op">=</span><span class="st">"GenomicBreaks"</span><span class="op">)</span>,</span>
<span class="r-in">  target <span class="op">=</span> <span class="va">OSKA2016</span>,</span>
<span class="r-in">  query <span class="op">=</span> <span class="va">OKI2018_I69</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

