<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Loads alignments of a query genome to a target genome from a text file in
General Feature Format 3 (GFF3) or Multiple Alignemnt Format (MAF).  By
convention, the target genome is the one that was indexed by the aligner."><title>Load pairwise genome alignments — load_genomic_breaks • GenomicBreaks</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Load pairwise genome alignments — load_genomic_breaks"><meta property="og:description" content="Loads alignments of a query genome to a target genome from a text file in
General Feature Format 3 (GFF3) or Multiple Alignemnt Format (MAF).  By
convention, the target genome is the one that was indexed by the aligner."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.14.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Load pairwise genome alignments</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/HEAD/R/load_genomic_breaks.R" class="external-link"><code>R/load_genomic_breaks.R</code></a></small>
      <div class="d-none name"><code>load_genomic_breaks.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Loads alignments of a <em>query</em> genome to a <em>target</em> genome from a text file in
<em>General Feature Format 3</em> (GFF3) or <em>Multiple Alignemnt Format</em> (MAF).  By
convention, the <em>target</em> genome is the one that was indexed by the aligner.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">load_genomic_breaks</span><span class="op">(</span></span>
<span>  <span class="va">file</span>,</span>
<span>  target_bsgenome <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  query_bsgenome <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sort <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"match_part"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>file</dt>
<dd><p>Path to a file in GFF3 or MAF format.  The file can be compressed
with <em>gzip</em>.</p></dd>


<dt>target_bsgenome</dt>
<dd><p>A <code>BSgenome</code> object representing the <em>target</em> genome.</p></dd>


<dt>query_bsgenome</dt>
<dd><p>A <code>BSgenome</code> object representing the <em>query</em> genome.</p></dd>


<dt>sort</dt>
<dd><p>Returns the object sorted, ignoring strand information.</p></dd>


<dt>type</dt>
<dd><p>In GFF3 files, <em>Sequence Ontology</em> term representing an alignment
block (default: <code>match_part</code>).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Returns a <code><a href="GBreaks-class.html">GBreaks</a></code> object where each element represents a pairwise
alignment block.  The <code>granges</code> part of the object contains the coordinates
on the <em>target</em> genome, and the <code>query</code> metadata column contains the <em>query</em></p>


<p>coordinates in <code>GRanges</code> format.  The <code>seqinfo</code> of each <code>BSgenome</code> object
passed as parameters are copied to the <code>GRanges</code> objects accordingly.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>When the input is in GFF3 files, this function expects the pairwise alignment
to be represented in in the following way:</p><ul><li><p>Alignments blocks are represented by entries in specific sequence ontology
term in the <em>type</em> column.  Other entries will be discarded.  The default
type is <code>match_part</code>.</p></li>
<li><p>The coordinate system of the file is the one of the <em>target</em> genome.</p></li>
<li><p>The <code>Target</code> tag in the <em>attribute</em> column contains the coordinates of
the match in the <em>query</em> genome.  (Sorry that it is confusing…)</p></li>
<li><p>Stand information is set so that <em>query</em> genome coordinates are always on
the <em>plus</em> strand.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>The <a href="http://www.genome.ucsc.edu/FAQ/FAQformat.html#format5" class="external-link">MAF format</a>
documentation on the UCSC genome browser website, and the
<a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md" class="external-link">GFF3 specification</a>
from the <em>Sequence Ontology</em> group.</p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">load_genomic_breaks</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/contigs.genome.maf.gz"</span>, package <span class="op">=</span> <span class="st">"GenomicBreaks"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 2 ranges and 4 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames     ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;  &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1] MT192765.1    25-8666      + |     52990</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2] MT192765.1 8882-29829      - |    128566</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                             query   aLength   matches</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                         &lt;GRanges&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]    NODE_2_length_8774_cov_178.827802:6-8647      8642      8642</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2] NODE_1_length_20973_cov_191.628754:26-20973     20948     20945</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BSgenome.Scerevisiae.UCSC.sacCer3"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">load_genomic_breaks</span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/SacCer3__SacPar.gff3.gz"</span>, package <span class="op">=</span> <span class="st">"GenomicBreaks"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  target <span class="op">=</span> <span class="va">Scerevisiae</span>,</span></span>
<span class="r-in"><span>  query <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

