<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Percent difference Distance — P_distance • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Percent difference Distance — P_distance"><meta name="description" content="The P-distance is simply the percentage of the aligned base pairs where the
two bases are different.  See Details for a discussion on the many ways to
compute percent identity (and therefore difference.)
The P_error function reports the standard error given the alignment
length \(n\) and percent difference \(p\) computed from the matrix:
\(\sqrt{p(1 - p) / n}\)."><meta property="og:description" content="The P-distance is simply the percentage of the aligned base pairs where the
two bases are different.  See Details for a discussion on the many ways to
compute percent identity (and therefore difference.)
The P_error function reports the standard error given the alignment
length \(n\) and percent difference \(p\) computed from the matrix:
\(\sqrt{p(1 - p) / n}\)."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.19.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Percent difference Distance</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/P_distance.R" class="external-link"><code>R/P_distance.R</code></a></small>
      <div class="d-none name"><code>P_distance.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The P-distance is simply the percentage of the aligned base pairs where the
two bases are different.  See Details for a discussion on the many ways to
compute percent identity (and therefore difference.)</p>
<p>The <code>P_error</code> function reports the standard error given the alignment
length \(n\) and percent difference \(p\) computed from the matrix:
\(\sqrt{p(1 - p) / n}\).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">P_distance</span><span class="op">(</span><span class="va">m</span>, denominator <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L3"</span>, <span class="st">"L1"</span>, <span class="st">"L2"</span>, <span class="st">"L4"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">P_error</span><span class="op">(</span><span class="va">m</span>, denominator <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L3"</span>, <span class="st">"L1"</span>, <span class="st">"L2"</span>, <span class="st">"L4"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-m">m<a class="anchor" aria-label="anchor" href="#arg-m"></a></dt>
<dd><p>A matrix of counts (<code>P_distance</code>, <code>P_variance</code>) or probabilities
(<code>P_distance</code> only) for bases of the <em>target</em> genome to be aligned to bases
on the <em>query</em> genome.  As a convenience it can also receive a list produced
by the <code><a href="readTrainFile.html">readTrainFile()</a></code> function, containing this matrix.</p></dd>


<dt id="arg-denominator">denominator<a class="anchor" aria-label="anchor" href="#arg-denominator"></a></dt>
<dd><p>Denominator according to the nomenclature of May (2004).
Default is <code>L3</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A numeric value representing the evolutionary distance between two
genomes.  The larger the value, the more divergent the genomes.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The P-distance is computed as 1 minus the percent sequence identity
according to the definitions of May (2004), which is the ratio between the
number of matching bases and four possible denominators in use at that time.</p><ul><li><p>L1: Length of the shorter sequence.</p></li>
<li><p>L2: Number of aligned positions, i.e., alignment length (includes gaps, if any).</p></li>
<li><p>L3: Number of aligned residue pairs, i.e., identities and nonidentities (excludes gaps, if any).</p></li>
<li><p>L4: Arithmetic mean sequence length.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>May, A. C. W. (2004). Percent Sequence Identity: The Need to Be Explicit.
<em>Structure</em>, 12(5), 737–738. <a href="https://doi.org/10.1016/j.str.2004.04.001" class="external-link">doi:10.1016/j.str.2004.04.001</a></p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Alignment statistics:
<code><a href="F81_distance.html">F81_distance</a>()</code>,
<code><a href="GCequilibrium.html">GCequilibrium</a>()</code>,
<code><a href="GCpressure.html">GCpressure</a>()</code>,
<code><a href="GCproportion.html">GCproportion</a>()</code>,
<code><a href="JC69_distance.html">JC69_distance</a>()</code>,
<code><a href="K80_distance.html">K80_distance</a>()</code>,
<code><a href="T92_distance.html">T92_distance</a>()</code>,
<code><a href="exampleSubstitutionMatrix.html">exampleSubstitutionMatrix</a></code>,
<code><a href="gapProportion.html">gapProportion</a>()</code></p>
<p>Other Similarity indexes:
<code><a href="F81_distance.html">F81_distance</a>()</code>,
<code><a href="GOC.html">GOC</a>()</code>,
<code><a href="JC69_distance.html">JC69_distance</a>()</code>,
<code><a href="K80_distance.html">K80_distance</a>()</code>,
<code><a href="T92_distance.html">T92_distance</a>()</code>,
<code><a href="correlation_index.html">correlation_index</a>()</code>,
<code><a href="karyotype_index.html">karyotype_index</a>()</code>,
<code><a href="slidingWindow.html">slidingWindow</a>()</code>,
<code><a href="strand_randomisation_index.html">strand_randomisation_index</a>()</code>,
<code><a href="synteny_index.html">synteny_index</a>()</code>,
<code><a href="tau_index.html">tau_index</a>()</code></p>
<p>Other Alignment statistics:
<code><a href="F81_distance.html">F81_distance</a>()</code>,
<code><a href="GCequilibrium.html">GCequilibrium</a>()</code>,
<code><a href="GCpressure.html">GCpressure</a>()</code>,
<code><a href="GCproportion.html">GCproportion</a>()</code>,
<code><a href="JC69_distance.html">JC69_distance</a>()</code>,
<code><a href="K80_distance.html">K80_distance</a>()</code>,
<code><a href="T92_distance.html">T92_distance</a>()</code>,
<code><a href="exampleSubstitutionMatrix.html">exampleSubstitutionMatrix</a></code>,
<code><a href="gapProportion.html">gapProportion</a>()</code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Charles Plessy</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">P_distance</span><span class="op">(</span><span class="va">exampleSubstitutionMatrix</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.2312837</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># example code</span></span></span>
<span class="r-in"><span><span class="fu">P_error</span><span class="op">(</span><span class="va">exampleSubstitutionMatrix</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 7.701531e-05</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

