<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Pairwise alignment of genomic ranges — pairwiseAlignment • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pairwise alignment of genomic ranges — pairwiseAlignment"><meta name="description" content="Retrieves DNA sequence from GBreaks or GenomicRanges::GRanges objects that are
properly related to a BSgenome package, and aligns them with the
pwalign::pairwiseAlignment() function."><meta property="og:description" content="Retrieves DNA sequence from GBreaks or GenomicRanges::GRanges objects that are
properly related to a BSgenome package, and aligns them with the
pwalign::pairwiseAlignment() function."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.20.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Pairwise alignment of genomic ranges</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/pairwiseAlignment.R" class="external-link"><code>R/pairwiseAlignment.R</code></a></small>
      <div class="d-none name"><code>pairwiseAlignment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Retrieves DNA sequence from <code><a href="GBreaks-class.html">GBreaks</a></code> or <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GenomicRanges::GRanges</a></code> objects that are
properly related to a <code>BSgenome</code> package, and aligns them with the
<code>pwalign::pairwiseAlignment()</code> function.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">pattern</span>, <span class="va">subject</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'GRanges,GRanges'</span></span>
<span><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">pattern</span>, <span class="va">subject</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'GBreaks,ANY'</span></span>
<span><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">pattern</span>, <span class="va">subject</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-pattern">pattern<a class="anchor" aria-label="anchor" href="#arg-pattern"></a></dt>
<dd><p>A <code>GBreaks</code> or a <code>GRanges</code> object</p></dd>


<dt id="arg-subject">subject<a class="anchor" aria-label="anchor" href="#arg-subject"></a></dt>
<dd><p>A <code>GBreaks</code> or a <code>GRanges</code> object</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments passed to <code>pairwiseAlignment</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a <code><a href="https://rdrr.io/pkg/pwalign/man/PairwiseAlignments-class.html" class="external-link">pwalign::PairwiseAlignments</a></code> object.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Bioconductor API functions:
<code><a href="GBreaks-class.html">GBreaks-class</a></code>,
<code><a href="getSeq.html">getSeq()</a></code>,
<code><a href="range.html">range_GBreaks()</a></code>,
<code><a href="reverse.html">reverse()</a></code>,
<code><a href="subsetByOverlaps.html">subsetByOverlaps_GBreaks()</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Charles Plessy</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">Scerevisiae</span> <span class="op">&lt;-</span> <span class="fu">BSgenome.Scerevisiae.UCSC.sacCer3</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Scerevisiae.UCSC.sacCer3/man/package.html" class="external-link">Scerevisiae</a></span></span></span>
<span class="r-in"><span><span class="co"># Very arbitrary example</span></span></span>
<span class="r-in"><span><span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="GBreaks-class.html">GBreaks</a></span><span class="op">(</span> target <span class="op">=</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"chrI: 1-20"</span>, seqinfo <span class="op">=</span> <span class="fu">seqinfo</span><span class="op">(</span><span class="va">Scerevisiae</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>             , query  <span class="op">=</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"chrI:21-40"</span>, seqinfo <span class="op">=</span> <span class="fu">seqinfo</span><span class="op">(</span><span class="va">Scerevisiae</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in .call_fun_in_pwalign("pairwiseAlignment", ...):</span> pairwiseAlignment() has moved from Biostrings to the pwalign package, and is</span>
<span class="r-err co"><span class="r-pr">#&gt;</span>   formally defunct in Biostrings &gt;= 2.77.1. Please call</span>
<span class="r-err co"><span class="r-pr">#&gt;</span>   pwalign::pairwiseAlignment() to get rid of this error.</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

