<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plots a pair of chromosomes from a GBreaks object. — plotApairOfChrs • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plots a pair of chromosomes from a GBreaks object. — plotApairOfChrs"><meta name="description" content="Plots the mapping between a pair of chromosomes using the
genoPlotR::plot_gene_map function.  One sequence feature (usually a
chromosome) is selected from the target genome.  The corresponding sequence
feature on the query genome is either chosen automatically (being the one
with the largest fraction of the mappings) or given as a parameter."><meta property="og:description" content="Plots the mapping between a pair of chromosomes using the
genoPlotR::plot_gene_map function.  One sequence feature (usually a
chromosome) is selected from the target genome.  The corresponding sequence
feature on the query genome is either chosen automatically (being the one
with the largest fraction of the mappings) or given as a parameter."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.17.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Plots a pair of chromosomes from a <code>GBreaks</code> object.</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/genoPlotR.R" class="external-link"><code>R/genoPlotR.R</code></a></small>
      <div class="d-none name"><code>plotApairOfChrs.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plots the mapping between a pair of chromosomes using the
<code><a href="https://rdrr.io/pkg/genoPlotR/man/plot_gene_map.html" class="external-link">genoPlotR::plot_gene_map</a></code> function.  One sequence feature (usually a
chromosome) is selected from the <em>target</em> genome.  The corresponding sequence
feature on the <em>query</em> genome is either chosen automatically (being the one
with the largest fraction of the mappings) or given as a parameter.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plotApairOfChrs</span><span class="op">(</span></span>
<span>  <span class="va">gb</span>,</span>
<span>  chrT <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  chrQ <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  dna_seg_scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  dna_seg_labels <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gb">gb<a class="anchor" aria-label="anchor" href="#arg-gb"></a></dt>
<dd><p>A <code><a href="GBreaks-class.html">GBreaks</a></code> object.</p></dd>


<dt id="arg-chrt">chrT<a class="anchor" aria-label="anchor" href="#arg-chrt"></a></dt>
<dd><p>A sequence name on the <em>target</em> genome.  Defaults to the first
<em>sequence level</em> of the <code>gb</code> object.</p></dd>


<dt id="arg-chrq">chrQ<a class="anchor" aria-label="anchor" href="#arg-chrq"></a></dt>
<dd><p>(Optional) A sequence name on the <em>query</em> genome.  Defaults to
the longest cumulative match on <code>chrT</code>.</p></dd>


<dt id="arg-dna-seg-scale">dna_seg_scale<a class="anchor" aria-label="anchor" href="#arg-dna-seg-scale"></a></dt>
<dd><p>Plot coordinats and a scale bar (defaults to <code>TRUE</code>).</p></dd>


<dt id="arg-dna-seg-labels">dna_seg_labels<a class="anchor" aria-label="anchor" href="#arg-dna-seg-labels"></a></dt>
<dd><p>A character vector to override default labels for
sequence names.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Further arguments are passed to <code>plot_gene_map</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Plots to the active device and returns and <code>oma::oma_layout</code> object
like the <code><a href="https://rdrr.io/pkg/genoPlotR/man/plot_gene_map.html" class="external-link">genoPlotR::plot_gene_map</a></code> function.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>In this kind of plot, the ranges from the selected sequence on the
<em>target</em> genome that are not mapped to the <em>query</em> sequence are not
displayed.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other genoPlotR functions:
<code><a href="gb2comp.html">gb2comp</a>()</code>,
<code><a href="gb2dna_seg.html">gb2dna_seg</a>()</code>,
<code><a href="gb2xlim.html">gb2xlim</a>()</code>,
<code><a href="gr2dna_seg.html">gr2dna_seg</a>()</code></p>
<p>Other plot functions:
<code><a href="bp_heatmap.html">bp_heatmap</a>()</code>,
<code><a href="bp_pair_analysis.html">bp_pair_analysis</a>()</code>,
<code><a href="feature_coverage.html">feature_coverage</a>()</code>,
<code><a href="makeOxfordPlots.html">makeOxfordPlots</a>()</code></p>
<p>Other Structural variants:
<code><a href="StructuralVariants.html">StructuralVariants</a></code>,
<code><a href="flagDoubleInversions.html">flagDoubleInversions</a>()</code>,
<code><a href="flagInversions.html">flagInversions</a>()</code>,
<code><a href="flagPairs.html">flagPairs</a>()</code>,
<code><a href="flagTranslocations.html">flagTranslocations</a>()</code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Charles Plessy</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">plotApairOfChrs</span><span class="op">(</span><span class="va">exampleInversion</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plotApairOfChrs-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">plotApairOfChrs</span><span class="op">(</span><span class="va">exampleDeletion</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plotApairOfChrs-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Labels can be overriden.</span></span></span>
<span class="r-in"><span><span class="fu">plotApairOfChrs</span><span class="op">(</span><span class="va">exampleDeletion</span>, dna_seg_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"over..."</span>, <span class="st">"...ridden"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plotApairOfChrs-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

