<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Structural Variants — StructuralVariants • GenomicBreaks</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Structural Variants — StructuralVariants"><meta name="description" content="Concept examples of structural variants for test and
demonstration purpose"><meta property="og:description" content="Concept examples of structural variants for test and
demonstration purpose"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.14.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Structural Variants</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/StructuralVariants.R" class="external-link"><code>R/StructuralVariants.R</code></a></small>
      <div class="d-none name"><code>StructuralVariants.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Concept examples of structural variants for test and
demonstration purpose</p>
    </div>


    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p><code><a href="GBreaks-class.html">GBreaks</a></code> objects described in more details in the
<em>Structural Variants</em> vignette, available with the command
<code><a href="../articles/StructuralVariants.html">vignette("StructuralVariants", package = "GenomicBreaks")</a></code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The variants represented here are the ones that can be represented
in a <code><a href="GBreaks-class.html">GBreaks</a></code> object.  Short ones that are entirely contained in an
aligned region, for instance as a gap, are not covered</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Structural variants:
<code><a href="flagDoubleInversions.html">flagDoubleInversions</a>()</code>,
<code><a href="flagInversions.html">flagInversions</a>()</code>,
<code><a href="flagPairs.html">flagPairs</a>()</code>,
<code><a href="flagTranslocations.html">flagTranslocations</a>()</code>,
<code><a href="plotApairOfChrs.html">plotApairOfChrs</a>()</code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Charles Plessy</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">exampleColinear</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 2 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-150      + | chrB:100-150</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   251-300      + | chrB:251-300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleNotColinear</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 2 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-150      + | chrB:201-251</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   251-300      + |  chrB:50-100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleColinear3</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrB:201-300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleColinear5</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-190      + | chrB:100-190</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   200-290      + | chrB:200-290</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   300-390      + | chrB:300-390</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chrA   400-490      + | chrB:400-490</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chrA   500-590      + | chrB:500-590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleInversion</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-190      + | chrB:100-190</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   210-291      - | chrB:210-291</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleInversion5uncollapsed</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-190      + | chrB:100-190</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   200-290      - | chrB:300-390</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   300-390      - | chrB:200-290</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chrA   400-490      + | chrB:400-490</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chrA   500-590      + | chrB:500-590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleNestedInversions</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-190      + | chrA:100-190</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   200-290      - | chrA:400-490</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   300-390      + | chrA:300-390</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chrA   400-490      - | chrA:200-290</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chrA   500-590      + | chrA:500-590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleDoubleInversion1</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-190      + | chrB:100-190</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   200-290      + | chrB:400-490</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   300-390      - | chrB:200-290</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chrA   400-490      - | chrB:300-390</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chrA   500-590      + | chrB:500-590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleTranslocation</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      + | chrC:201-300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span><span class="va">exampleTranslocation2</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GBreaks object with 3 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        query</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrA   100-200      + | chrB:100-200</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrA   201-300      - | chrC:201-300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chrA   301-400      + | chrB:301-400</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

