<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Index representing how 'syntenic' two genomes are — synteny_index • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Index representing how 'syntenic' two genomes are — synteny_index"><meta name="description" content="A sequence feature on the target genome will be considered &amp;#8216;syntenic&amp;#8217;
if most of its alignments map to the same feature of the query genome, and
the two genomes are considered syntenic if most of their features are
syntenic."><meta property="og:description" content="A sequence feature on the target genome will be considered &amp;#8216;syntenic&amp;#8217;
if most of its alignments map to the same feature of the query genome, and
the two genomes are considered syntenic if most of their features are
syntenic."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.19.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Index representing how 'syntenic' two genomes are</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/synteny_index.R" class="external-link"><code>R/synteny_index.R</code></a></small>
      <div class="d-none name"><code>synteny_index.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>A sequence feature on the <em>target</em> genome will be considered ‘syntenic’
if most of its alignments map to the same feature of the <em>query</em> genome, and
the two genomes are considered syntenic if most of their features are
syntenic.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">synteny_index</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gb">gb<a class="anchor" aria-label="anchor" href="#arg-gb"></a></dt>
<dd><p>A <code><a href="GBreaks-class.html">GBreaks</a></code> object.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a numeric value between 0 and 1.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>For a given feature of the <em>target</em> genome the number of bases aligned on
each of the features of the <em>query</em> genome are computed.  The largest number
is taken.  To make features comparable, this number is then normalised by
total number of aligned bases.  Then, the computed numbers for all the
<em>target</em>'s features are averaged with a weighted mean, where the weight is
the total number of aligned bases of that feature.  Thus, a number close to
1 is expected for closely related genomes.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Here, the definition of ‘synteny’ is <em>sequences on the same
DNA strands</em>, or in technical terms: sequences that are on the same feature,
regardless of strand orientation.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Similarity indexes:
<code><a href="F81_distance.html">F81_distance</a>()</code>,
<code><a href="GOC.html">GOC</a>()</code>,
<code><a href="JC69_distance.html">JC69_distance</a>()</code>,
<code><a href="K80_distance.html">K80_distance</a>()</code>,
<code><a href="P_distance.html">P_distance</a>()</code>,
<code><a href="T92_distance.html">T92_distance</a>()</code>,
<code><a href="correlation_index.html">correlation_index</a>()</code>,
<code><a href="karyotype_index.html">karyotype_index</a>()</code>,
<code><a href="slidingWindow.html">slidingWindow</a>()</code>,
<code><a href="strand_randomisation_index.html">strand_randomisation_index</a>()</code>,
<code><a href="tau_index.html">tau_index</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">gb</span>       <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ref:100-200:+"</span>,   <span class="st">"Ref:400-500:+"</span>,    <span class="st">"Ref:600-700:+"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gb</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Que:1100-1200:+"</span>, <span class="st">"Que2:1700-1800:+"</span>, <span class="st">"Que:1500-1600:+"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">synteny_index</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.6666667</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

