<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Whole-genome classification object — wholeGenomeClassification • GenomicBreaks</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Whole-genome classification object — wholeGenomeClassification"><meta name="description" content="Classifies aligned genomic regions as isolated or collinear, and
unaligned regions as breakpoint or bridge regions.  The bridge regions
are always flanked by collinear alignments and the isolated alignments
are always flanked by breakpoint regions."><meta property="og:description" content="Classifies aligned genomic regions as isolated or collinear, and
unaligned regions as breakpoint or bridge regions.  The bridge regions
are always flanked by collinear alignments and the isolated alignments
are always flanked by breakpoint regions."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.15.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Whole-genome classification object</h1>
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/main/R/wholeGenomeClassification.R" class="external-link"><code>R/wholeGenomeClassification.R</code></a></small>
      <div class="d-none name"><code>wholeGenomeClassification.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Classifies aligned genomic regions as <em>isolated</em> or <em>collinear</em>, and
unaligned regions as <em>breakpoint</em> or <em>bridge</em> regions.  The <em>bridge</em> regions
are always flanked by <em>collinear alignments</em> and the <em>isolated alignments</em>
are always flanked by <em>breakpoint regions</em>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">wholeGenomeClassification</span><span class="op">(</span><span class="va">gb</span>, coa <span class="op">=</span> <span class="fu"><a href="coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>, ends <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gb">gb<a class="anchor" aria-label="anchor" href="#arg-gb"></a></dt>
<dd><p>A <code><a href="GBreaks-class.html">GBreaks</a></code> object representing a one-to-one whole genome
alignment.</p></dd>


<dt id="arg-coa">coa<a class="anchor" aria-label="anchor" href="#arg-coa"></a></dt>
<dd><p>The coalesced one-to-one alignment.  If not provided, it will be\
computed on-the-fly with the <code><a href="coalesce_contigs.html">coalesce_contigs</a></code> function.</p></dd>


<dt id="arg-ends">ends<a class="anchor" aria-label="anchor" href="#arg-ends"></a></dt>
<dd><p>Add a a <em>end region</em> type for the extremities of the <em>sequence
features</em> not covered by the original alignment.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GenomicRanges::GRanges</a></code> object representing the <em>target</em>
genome, inheriting its <em>sequence information</em> (<code><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">GenomeInfoDb::Seqinfo</a></code>).
The class of each region is indicated by a factor in the <code>type</code> metadata
column.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The strand of <em>bridge regions</em> is the one of their flanking <em>collinear
alignment</em> regions.  <em>Breakpoint</em> and <em>end</em> regions are unstranded.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Reducing functions:
<code><a href="cleanGaps.html">cleanGaps</a>()</code>,
<code><a href="get_bps.html">get_bps</a>()</code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Charles Plessy</p>
<p>Michael Mansfield</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">exampleColinear5</span> <span class="op">|&gt;</span> <span class="fu">wholeGenomeClassification</span><span class="op">(</span>ends <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 11 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |                type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;factor&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]     chrA      1-99      * | end region         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]     chrA   100-190      + | collinear alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]     chrA   191-199      + | bridge region      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]     chrA   200-290      + | collinear alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]     chrA   291-299      + | bridge region      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [6]     chrA   300-390      + | collinear alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [7]     chrA   391-399      + | bridge region      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [8]     chrA   400-490      + | collinear alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [9]     chrA   491-499      + | bridge region      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [10]     chrA   500-590      + | collinear alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [11]     chrA   591-600      * | end region         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

