<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to GenomicBreaks • GenomicBreaks</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to GenomicBreaks">
<meta property="og:description" content="GenomicBreaks">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicBreaks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GenomicBreaks_vignette.html">Introduction to GenomicBreaks</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/GenomicBreaks/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GenomicBreaks_vignette_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to GenomicBreaks</h1>
                        <h4 class="author">Charlotte West</h4>
                        <h4 class="author">Charles Plessy</h4>
            
            <h4 class="date">10 September, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/master/vignettes/GenomicBreaks_vignette.Rmd"><code>vignettes/GenomicBreaks_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicBreaks_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <em>GenomicBreaks</em> R package contains tools for visualising and analysing characteristics associated with <em>breakpoints</em> between pairs of <em>genomes</em>.</p>
<div id="load-pacakges" class="section level2">
<h2 class="hasAnchor">
<a href="#load-pacakges" class="anchor"></a>Load pacakges</h2>
<p>Core packages that provide functions we use a lot.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/GenomicBreaks">'GenomicBreaks'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/BSgenome">'BSgenome'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicFeatures">'GenomicFeatures'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicRanges">'GenomicRanges'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">'ggplot2'</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>We will use yeast data as an example, so let’s load the <code>Scerevisiae</code> BSgenome package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"BSgenome.Scerevisiae.UCSC.sacCer3"</span><span class="op">)</span></code></pre></div>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>This package contains an example data file representing the alignment of the <em>Saccharomyces paradoxus</em> genome version <code>ASM207905v1</code> to the <code>SacCer3</code> reference genome of <em>S. cerevisiae</em>. An alignment file was produced with a Nextflow <a href="https://github.com/oist/plessy_pairwiseGenomeComparison">pairwise alignment pipeline</a> and converted to <a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">GFF3</a> format with the <code>maf-convert gff -J 1000000</code> command of the <a href="https://gitlab.com/mcfrith/last/-/blob/main/doc/maf-convert.rst">LAST</a> package.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/SacCer3__SacPar.gff3.gz"</span>, package <span class="op">=</span> <span class="st">"GenomicBreaks"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="va">exdata</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 590 ranges and 8 metadata columns:
##         seqnames          ranges strand |      source       type     score
##            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;factor&gt;   &lt;factor&gt; &lt;numeric&gt;
##     [1]     chrM     58008-76168      + | maf-convert match             NA
##     [2]     chrM     58008-60967      + | maf-convert match_part     12243
##     [3]     chrM     61783-62447      + | maf-convert match_part      2557
##     [4]     chrM     63688-64744      + | maf-convert match_part      1520
##     [5]     chrM     66006-66345      + | maf-convert match_part      1159
##     ...      ...             ...    ... .         ...        ...       ...
##   [586]    chrIX   300905-301573      - | maf-convert match_part      1576
##   [587]    chrXV 1073474-1074872      - | maf-convert match             NA
##   [588]    chrXV 1073474-1074872      - | maf-convert match_part      3761
##   [589]   chrVII 1081381-1082672      - | maf-convert match             NA
##   [590]   chrVII 1081381-1082672      - | maf-convert match_part      4057
##             phase            ID                   Name                 Target
##         &lt;integer&gt;   &lt;character&gt;            &lt;character&gt;            &lt;character&gt;
##     [1]      &lt;NA&gt; NC_018044.1.1    NC_018044.1:1-16261                   &lt;NA&gt;
##     [2]      &lt;NA&gt;          &lt;NA&gt;     NC_018044.1:1-2820     NC_018044.1 1 2820
##     [3]      &lt;NA&gt;          &lt;NA&gt;  NC_018044.1:2834-3463  NC_018044.1 2834 3463
##     [4]      &lt;NA&gt;          &lt;NA&gt;  NC_018044.1:4786-5811  NC_018044.1 4786 5811
##     [5]      &lt;NA&gt;          &lt;NA&gt;  NC_018044.1:6743-7110  NC_018044.1 6743 7110
##     ...       ...           ...                    ...                    ...
##   [586]      &lt;NA&gt;          &lt;NA&gt; NC_047502.1:665227-6.. NC_047502.1 665227 6..
##   [587]      &lt;NA&gt; NC_047502.1.8  NC_047502.1:7842-9171                   &lt;NA&gt;
##   [588]      &lt;NA&gt;          &lt;NA&gt;  NC_047502.1:7842-9171  NC_047502.1 7842 9171
##   [589]      &lt;NA&gt; NC_047502.1.9  NC_047502.1:2001-3295                   &lt;NA&gt;
##   [590]      &lt;NA&gt;          &lt;NA&gt;  NC_047502.1:2001-3295  NC_047502.1 2001 3295
##                  Parent
##         &lt;CharacterList&gt;
##     [1]                
##     [2]   NC_018044.1.1
##     [3]   NC_018044.1.1
##     [4]   NC_018044.1.1
##     [5]   NC_018044.1.1
##     ...             ...
##   [586]   NC_047502.1.7
##   [587]                
##   [588]   NC_047502.1.8
##   [589]                
##   [590]   NC_047502.1.9
##   -------
##   seqinfo: 17 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div id="gbreaks-objects" class="section level1">
<h1 class="hasAnchor">
<a href="#gbreaks-objects" class="anchor"></a>GBreaks objects</h1>
<p>Pairwise genome alignments are loaded in <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> objects that wrap the <code>GRanges</code> class. By convention, we write that the <em>query</em> genome (the one that was provided as a FASTA file) is aligned to the <em>target</em> genome (the one that was indexed by the aligner). The <em>target</em> genome (on the left) is the main part of the object, and the <em>query</em> genome information (right) is contained in the metadata columns (<code>mcols</code>) of the structure. Information on contig or scaffold length is optionaly loaded from <code>BSgenome</code> objects, from which <code>seqinfo</code> data is extracted and passed to the <code>GRanges</code>. By convention, strand information of the alignment is held by the <em>target</em> <code>GRanges</code>, and the <em>query</em> <code>GRanges</code> are strandless. The objects are sorted by <code>seqname</code> first.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_genomic_breaks.html">load_genomic_breaks</a></span><span class="op">(</span><span class="va">exdata</span>, <span class="va">Scerevisiae</span><span class="op">)</span>
<span class="va">gb</span></code></pre></div>
<pre><code>## GBreaks object with 505 ranges and 2 metadata columns:
##         seqnames      ranges strand |     score                   query
##            &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;               &lt;GRanges&gt;
##     [1]     chrI  5860-10010      + |     10609  NC_047487.1:8723-12716
##     [2]     chrI 11157-11782      + |      1253 NC_047487.1:13035-13680
##     [3]     chrI 25371-26528      + |      3651 NC_047487.1:15380-16537
##     [4]     chrI 26849-29699      + |      6130 NC_047487.1:17263-20175
##     [5]     chrI 29937-30607      + |       767 NC_047487.1:20203-20866
##     ...      ...         ...    ... .       ...                     ...
##   [501]     chrM 67039-67870      + |      1441   NC_018044.1:7758-8439
##   [502]     chrM 68205-68580      + |       830   NC_018044.1:8783-9180
##   [503]     chrM 69178-76168      + |     14528  NC_018044.1:9650-16261
##   [504]     chrM 77356-80022      + |      8066 NC_018044.1:53206-55865
##   [505]     chrM 80919-85779      + |      6712 NC_018044.1:57885-61592
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">$</span><span class="va">query</span></code></pre></div>
<pre><code>## GRanges object with 505 ranges and 0 metadata columns:
##            seqnames      ranges strand
##               &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;
##     [1] NC_047487.1  8723-12716      *
##     [2] NC_047487.1 13035-13680      *
##     [3] NC_047487.1 15380-16537      *
##     [4] NC_047487.1 17263-20175      *
##     [5] NC_047487.1 20203-20866      *
##     ...         ...         ...    ...
##   [501] NC_018044.1   7758-8439      *
##   [502] NC_018044.1   8783-9180      *
##   [503] NC_018044.1  9650-16261      *
##   [504] NC_018044.1 53206-55865      *
##   [505] NC_018044.1 57885-61592      *
##   -------
##   seqinfo: 17 sequences from an unspecified genome; no seqlengths</code></pre>
<div id="breakpoints-and-alignment-stops" class="section level2">
<h2 class="hasAnchor">
<a href="#breakpoints-and-alignment-stops" class="anchor"></a>Breakpoints and alignment stops</h2>
<p>Let us define an <strong>alignment stop</strong> to be a position defined in either the <em>target</em> or <em>query</em> genome, where an alignment begins or ends. Looking at the first line in the <code>gb</code> object above, <code>ChrI</code> positions 5860 and 10010 of the <em>target</em> would be defined as alignment stops. Correspondingly, we have positions 8723 and 12176 in <code>NC_047487.1</code> of the <em>query</em> genome as alignment stops.</p>
<p>The function <code><a href="../reference/get_bps.html">get_bps()</a></code> takes in a <code>GBreaks</code> pairwise alignment or a simple <code>GRanges</code> object, and produces a <code>GRanges</code> object of it’s associated alignment stops. It has <code>direction</code>, <code>stranded</code> and <code>sorted</code> options to refind its output.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_bps.html">get_bps</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chrI      5860      *
##   [2]     chrI     10010      *
##   [3]     chrI     11157      *
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_bps.html">get_bps</a></span><span class="op">(</span><span class="va">gb</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chrI      5860      *
##   [2]     chrI     11157      *
##   [3]     chrI     25371      *
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_bps.html">get_bps</a></span><span class="op">(</span><span class="va">gb</span>,                      stranded <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chrI      5860      +
##   [2]     chrI     10010      -
##   [3]     chrI     11157      +
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_bps.html">get_bps</a></span><span class="op">(</span><span class="va">gb</span>,                      stranded <span class="op">=</span> <span class="cn">TRUE</span>, sorted <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chrI      5860      +
##   [2]     chrI     11157      +
##   [3]     chrI     25371      +
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_bps.html">get_bps</a></span><span class="op">(</span><span class="va">gb</span>, direction <span class="op">=</span> <span class="st">"right"</span>, stranded <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chrI     10010      -
##   [2]     chrI     11782      -
##   [3]     chrI     26528      -
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_bps.html">get_bps</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span>, sorted <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##          seqnames    ranges strand
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1] NC_047487.1      8723      *
##   [2] NC_047487.1     13035      *
##   [3] NC_047487.1     15380      *
##   -------
##   seqinfo: 17 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_bps.html">get_bps</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##          seqnames    ranges strand
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1] NC_018044.1         1      *
##   [2] NC_018044.1      2820      *
##   [3] NC_018044.1      2834      *
##   -------
##   seqinfo: 17 sequences from an unspecified genome; no seqlengths</code></pre>
<p>Now, let us define a <strong>genomic breakpoint</strong> (or simply breakpoint) to be a genomic structural mutation, arising from breakage and repair of the chromosome. Such structural events include insertion, deletion, inversion and translocation, and often arise during recombination.</p>
<p>This package seeks to determine, of the group of alignment stops, which are likely genomic breakpoints and analyse them. Thus, filtering out alignment stops that are a result of bioinformatic errors.</p>
</div>
<div id="arms" class="section level2">
<h2 class="hasAnchor">
<a href="#arms" class="anchor"></a>Arms</h2>
<p>If the positions of the centromeres are known, an annotation file can be crafted to indicate the coordinates of the short and long arms, and the function <code><a href="../reference/flagLongShort.html">flagLongShort()</a></code> can be used to transfer this annotation to the <code>GBreaks</code> object.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chrI:1-151465"</span>, <span class="st">"chrI:151582-230218"</span><span class="op">)</span><span class="op">)</span>
<span class="va">annot</span><span class="op">$</span><span class="va">Type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"short"</span>, <span class="st">"long"</span><span class="op">)</span>
<span class="fu"><a href="../reference/flagLongShort.html">flagLongShort</a></span><span class="op">(</span><span class="va">gb</span>, <span class="va">annot</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 505 ranges and 3 metadata columns:
##         seqnames      ranges strand |     score                   query
##            &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;               &lt;GRanges&gt;
##     [1]     chrI  5860-10010      + |     10609  NC_047487.1:8723-12716
##     [2]     chrI 11157-11782      + |      1253 NC_047487.1:13035-13680
##     [3]     chrI 25371-26528      + |      3651 NC_047487.1:15380-16537
##     [4]     chrI 26849-29699      + |      6130 NC_047487.1:17263-20175
##     [5]     chrI 29937-30607      + |       767 NC_047487.1:20203-20866
##     ...      ...         ...    ... .       ...                     ...
##   [501]     chrM 67039-67870      + |      1441   NC_018044.1:7758-8439
##   [502]     chrM 68205-68580      + |       830   NC_018044.1:8783-9180
##   [503]     chrM 69178-76168      + |     14528  NC_018044.1:9650-16261
##   [504]     chrM 77356-80022      + |      8066 NC_018044.1:53206-55865
##   [505]     chrM 80919-85779      + |      6712 NC_018044.1:57885-61592
##                 Arm
##         &lt;character&gt;
##     [1]       short
##     [2]       short
##     [3]       short
##     [4]       short
##     [5]       short
##     ...         ...
##   [501]        &lt;NA&gt;
##   [502]        &lt;NA&gt;
##   [503]        &lt;NA&gt;
##   [504]        &lt;NA&gt;
##   [505]        &lt;NA&gt;
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
</div>
<div id="measures-of-distance" class="section level2">
<h2 class="hasAnchor">
<a href="#measures-of-distance" class="anchor"></a>Measures of distance</h2>
<div id="synteny-index" class="section level3">
<h3 class="hasAnchor">
<a href="#synteny-index" class="anchor"></a>Synteny index</h3>
<p>Ad-hoc index measuring to what extent a scaffold of the <em>target</em> genome is mostly aligned to a single scaffold in the <em>query</em> genome. See <code><a href="../reference/synteny_index.html">?synteny_index</a></code> for details. One limitation to the use of this index is that it requires that at least the <em>query</em> genome is a complete chromosome assembly.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/synteny_index.html">synteny_index</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.9954872</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/synteny_index.html">synteny_index</a></span><span class="op">(</span><span class="fu"><a href="../reference/swap.html">swap</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.9954982</code></pre>
</div>
<div id="correlation" class="section level3">
<h3 class="hasAnchor">
<a href="#correlation" class="anchor"></a>Correlation</h3>
<p>Ad-hoc index measuring the correlation of the coordinates of the syntenic alignments in scaffolds of a <em>target</em> genome and their best match in the <em>query</em> genome. See <code><a href="../reference/correlation_index.html">?correlation_index</a></code> for details. This index is more robust to the presence of uncollapsed haplotypes in the <em>query</em> genome.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/correlation_index.html">correlation_index</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.9942222</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/correlation_index.html">correlation_index</a></span><span class="op">(</span><span class="fu"><a href="../reference/swap.html">swap</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.9946129</code></pre>
</div>
<div id="gene-order-conservation" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-order-conservation" class="anchor"></a>Gene Order Conservation</h3>
<p>The <code><a href="../reference/GOC.html">GOC()</a></code> function calcluates the Gene Order Conservation number defined by Rocha (2003, <a href="https://doi.org/10.1016/j.tig.2003.09.011" class="uri">https://doi.org/10.1016/j.tig.2003.09.011</a>) as “<em>the average number of orthologues for which the consecutive orthologue co-occurs close by in the other genome. It varies between 0 (no co-occurrence) and 1 (complete gene order conservation)</em>”.</p>
<p><em>Note that calculating GOC on whole-genome alignments is not expected to produce meaningful results, and the example below is just to show that the computation is possible. This function is more useful when comparing the position of orthologues, represented in a <code>GBreaks</code> object.</em></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GOC.html">GOC</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.9227557</code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GOC.html">GOC</a></span><span class="op">(</span><span class="fu"><a href="../reference/swap.html">swap</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.9123173</code></pre>
</div>
</div>
<div id="visualising-alignments" class="section level2">
<h2 class="hasAnchor">
<a href="#visualising-alignments" class="anchor"></a>Visualising alignments</h2>
<p>We can use the function <code>scaf_align_plot</code> to see the alignment of query scaffolds/chromosomes onto a specific target scaffold/chromosome of interest.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/scaf_align_plot.html">scaf_align_plot</a></span><span class="op">(</span><span class="va">gb</span>, scaf <span class="op">=</span> <span class="st">"chrI"</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicBreaks_vignette_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>Here we have plotted the alignments on the Okinawa genome (y-axis) onto the Osaka genome, scaffold 2 (x-axis). The coverage of the alignment is the bar along the top.</p>
<p>We can plot alignments for two of the target scaffolds side-by-side, using <code>align_scaf_plot-2</code>. This can be used in aid of superscaffolding.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/align_scaf_plot_2.html">align_scaf_plot_2</a></span><span class="op">(</span><span class="va">gb</span>, scaf <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chrI"</span>,<span class="st">"chrII"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicBreaks_vignette_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>As you can see, scaffold 2 of the query genome aligns to the majority of both target scaffolds. This could be used as evidence to support the superscaffolding of these two scaffolds - in an effort toward chromosomal assembly.</p>
</div>
<div id="coalescing-alignments" class="section level2">
<h2 class="hasAnchor">
<a href="#coalescing-alignments" class="anchor"></a>Coalescing alignments</h2>
<p>Large syntenic regions can often appear cluttered with alignment breaks, spanning just a few basepairs. They are either an artefact (for instance in case of incomplete purge of haplotypes) and or true breakpoint. The algorithm in <code>coalesce_contigs</code> is used to produce a new GRanges object with fewer alignment breaks by coalescing alignments separated by short (user specified) distances. This distance need be agreeable in both the target and query genome in order for the coalscion to happen.</p>
<p>For example, coalescing gaps of less than 500 basepairs in the <code>gb</code> alignment:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 505</code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">coa</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 88</code></pre>
<p>The resulting GRanges object has far fewer alignments and therefore far fewer alignment stops. The algorithm is an initial step in alignment stop filtering, with the goal of a reduced number of alignment stops that have a high probability of being breakpoints.</p>
<div id="width-summary-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#width-summary-plots" class="anchor"></a>Width summary plots</h3>
<p>The distribution of widths is not the same on long and short arms. For this reason, it would be unwise to filter by width on some operations such as coalescing.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># width2df &lt;- function(where, what, gbl) {</span>
<span class="co">#   gb &lt;- gbl[[where]]</span>
<span class="co">#   if (length(gb) == 0) return (NULL)</span>
<span class="co">#   data.frame(where = where, what = what, width = width(gb))</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># df &lt;- rbind(</span>
<span class="co">#   # do.call allows a list to replace "..." in a function call</span>
<span class="co">#   do.call(rbind, lapply(names(gbs), width2df, "aligned", gbs)),</span>
<span class="co">#   do.call(rbind, lapply(names(coa), width2df, "collapsed", coa))</span>
<span class="co"># )</span>
<span class="co"># </span>
<span class="co"># ggplot(df) +</span>
<span class="co">#   aes(width, col = what) +</span>
<span class="co">#   geom_freqpoly() +</span>
<span class="co">#   scale_x_log10() +</span>
<span class="co">#   facet_wrap(~where, scales = "free_y")</span>
<span class="co"># </span>
<span class="co"># gg_freq_poly &lt;- function(DF) {</span>
<span class="co">#     ggplot(DF |&gt; as("data.frame")) +</span>
<span class="co">#     aes(width) +</span>
<span class="co">#     geom_freqpoly() +</span>
<span class="co">#     scale_x_log10()</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># DF &lt;- DataFrame(width = width(gbs$Oki_Osa), Type = gbs$Oki_Osa$Type, chr = seqnames(gbs$Oki_Osa))</span>
<span class="co"># </span>
<span class="co"># gg_freq_poly(DF) +</span>
<span class="co">#   aes(col=Type) +</span>
<span class="co">#   facet_wrap(~chr, scales = "free_y")</span></code></pre></div>
<p>After coalescing, width distribution becomes more clearly bimodal, with a small peak under 500 bp, found on all arms except YSR. The second peak shows two different distributions for long and short arms.</p>
<p>Therefore, if the first peak represents events that we do not want to consider breaking synteny (either alignment artefacts or insertion/transpositions of short mobile elements), we can re-coalesce once after removing the shortest intervals.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DF &lt;- DataFrame(</span>
<span class="co">#   width = width(coa$Oki_Osa),</span>
<span class="co">#   Type = mcols(OKI_longShort)[findOverlaps(coa$Oki_Osa, OKI_longShort) |&gt; subjectHits(), "Type"],</span>
<span class="co">#   chr = seqnames(coa$Oki_Osa))</span>
<span class="co"># </span>
<span class="co"># gg_freq_poly(DF) +</span>
<span class="co">#   aes(col=Type) +</span>
<span class="co">#   facet_wrap(~chr, scales = "free_y")</span></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># coa2 &lt;- sapply(coa, coalesce_contigs, minwidth = 500) |&gt; SimpleList()</span>
<span class="co"># sapply(coa2, length)</span>
<span class="co"># </span>
<span class="co"># DF &lt;- DataFrame(</span>
<span class="co">#   width = width(coa2$Oki_Osa),</span>
<span class="co">#   Type = mcols(OKI_longShort)[findOverlaps(coa2$Oki_Osa, OKI_longShort) |&gt; subjectHits(), "Type"],</span>
<span class="co">#   chr = seqnames(coa2$Oki_Osa))</span>
<span class="co"># </span>
<span class="co"># gg_freq_poly(DF) +</span>
<span class="co">#   aes(col=Type) +</span>
<span class="co">#   facet_wrap(~chr, scales = "free_y")</span></code></pre></div>
</div>
</div>
<div id="genome-plots-with-genoplotr" class="section level2">
<h2 class="hasAnchor">
<a href="#genome-plots-with-genoplotr" class="anchor"></a>Genome plots with genoPlotR</h2>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library("genoPlotR")</span>
<span class="co"># </span>
<span class="co"># example("dna_seg")</span>
<span class="co"># </span>
<span class="co"># gr2dna_seg &lt;- function (gr) {</span>
<span class="co">#   if (is.null(names(gr))) names(gr) &lt;- as.character(gr)</span>
<span class="co">#   strand(gr[strand(gr) == "*"]) &lt;- "+"</span>
<span class="co">#   df &lt;- data.frame(</span>
<span class="co">#     name   = names(gr),</span>
<span class="co">#     start  = start(gr),</span>
<span class="co">#     end    = end(gr),</span>
<span class="co">#     strand = strand(gr),</span>
<span class="co">#     col    = "grey"</span>
<span class="co">#   )</span>
<span class="co">#   dna_seg(df)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># gbRef2dna_seq &lt;- function (gb) {</span>
<span class="co">#   if (is.null(names(gb))) names(gb) &lt;- as.character(gb)</span>
<span class="co">#   strand(gb) &lt;- "+"</span>
<span class="co">#   gr2dna_seg(gb)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># gbQuery2dna_seq &lt;- function (gb) {</span>
<span class="co">#   if (is.null(names(gb$query))) names(gb$query) &lt;- as.character(gb)</span>
<span class="co">#   strand(gb$query) &lt;- strand(gb)</span>
<span class="co">#   gr2dna_seg(gb$query)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># # Plot whole chromosome 1</span>
<span class="co"># ds &lt;- gr2dna_seg(coa$Oki_Osa[seqnames(coa$Oki_Osa) == "chr1"])</span>
<span class="co"># head(ds) ; tail(ds)</span>
<span class="co"># plot_gene_map(list(ds))</span>
<span class="co"># </span>
<span class="co"># gb2comp &lt;- function(gb) {</span>
<span class="co">#   ref &lt;- gr2dna_seg(gb)</span>
<span class="co">#   strand(gb$query) &lt;- "+"</span>
<span class="co">#   que &lt;- gr2dna_seg(gb$query)</span>
<span class="co">#   df &lt;- data.frame(</span>
<span class="co">#     start1 = ref$start,</span>
<span class="co">#     end1   = ref$end,</span>
<span class="co">#     start2 = que$start,</span>
<span class="co">#     end2   = que$end</span>
<span class="co">#   )</span>
<span class="co">#   as.comparison(df)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># plotApairOfChrs &lt;- function(gb, chr) {</span>
<span class="co">#   gb &lt;- gb[seqnames(gb) == chr]</span>
<span class="co">#     keepMainMatch &lt;- function(gb) {</span>
<span class="co">#     bestMatch &lt;- tapply(width(gb$query), seqnames(gb$query), sum) |&gt; sort() |&gt; tail(1) |&gt; names()</span>
<span class="co">#     gb[seqnames(gb$query) == bestMatch]</span>
<span class="co">#   }</span>
<span class="co">#   roi &lt;- keepMainMatch(gb)</span>
<span class="co">#   dsList &lt;-   list(</span>
<span class="co">#       roi |&gt; gbQuery2dna_seq(),</span>
<span class="co">#       roi |&gt; gbRef2dna_seq()</span>
<span class="co">#       )</span>
<span class="co">#   </span>
<span class="co">#   compList &lt;- list(</span>
<span class="co">#       roi |&gt; swap() |&gt; gb2comp()</span>
<span class="co">#     )</span>
<span class="co">#   plot_gene_map(dsList, compList)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># plotApairOfChrs(coa$Oki_Bar, "chr1")</span>
<span class="co"># plotApairOfChrs(coa$Oki_Osa, "chr1")</span>
<span class="co"># plotApairOfChrs(coa$Osa_Oki, "Chr1")</span>
<span class="co"># plotApairOfChrs(coa$Osa_Bar, "Chr1")</span>
<span class="co"># plotApairOfChrs(coa2$Osa_Bar, "Chr1")</span></code></pre></div>
<div id="plot-a-simple-region-" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-a-simple-region-" class="anchor"></a>Plot a simple region.</h3>
<p>This region can be plotted because there is a one-to-one correspondence between alignments to Osaka and alignments to Norway.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># roiNameO &lt;- seqlevelsInUse(roi_Oki_O$query)[1]</span>
<span class="co"># roiNameOki &lt;- seqlevelsInUse(roi)[1]</span>
<span class="co"># roiNameN &lt;- seqlevelsInUse(roi_Oki_N$query)[1]</span>
<span class="co"># </span>
<span class="co"># dsList &lt;-   list(</span>
<span class="co">#     roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |&gt; gbQuery2dna_seq(),</span>
<span class="co">#     roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |&gt; gbRef2dna_seq(),</span>
<span class="co">#     roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |&gt; gbRef2dna_seq(),</span>
<span class="co">#     roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |&gt; gbQuery2dna_seq()</span>
<span class="co">#     )</span>
<span class="co"># names(dsList) &lt;- c(roiNameO, roiNameOki, roiNameOki, roiNameN)</span>
<span class="co"># </span>
<span class="co"># tmpgb &lt;- roi_Oki_O</span>
<span class="co"># tmpgb$query &lt;- granges(roi_Oki_N)</span>
<span class="co"># </span>
<span class="co"># compList &lt;- list(</span>
<span class="co">#     roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |&gt; swap() |&gt; gb2comp(),</span>
<span class="co">#     tmpgb |&gt; gb2comp(),</span>
<span class="co">#     roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |&gt; gb2comp()</span>
<span class="co">#   )</span>
<span class="co"># </span>
<span class="co"># plot_gene_map(dsList, compList, dna_seg_scale=TRUE)</span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># dsList &lt;-   list(</span>
<span class="co">#     roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |&gt; gbRef2dna_seq(),</span>
<span class="co">#     roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |&gt; gbQuery2dna_seq(),</span>
<span class="co">#     roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |&gt; gbQuery2dna_seq()</span>
<span class="co">#     )</span>
<span class="co"># names(dsList) &lt;- c(roiNameOki, roiNameO, roiNameN)</span>
<span class="co"># </span>
<span class="co"># compList &lt;- list(</span>
<span class="co">#     roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |&gt; gb2comp(),</span>
<span class="co">#     roi_O_N[seqnames(roi_O_N$query) == roiNameN] |&gt; gb2comp()</span>
<span class="co">#   )</span>
<span class="co"># </span>
<span class="co"># plot_gene_map(dsList, compList, dna_seg_scale=TRUE, tree=ade4::newick2phylog(addRoot(treeNode(roiNameOki, treeNode(roiNameO, roiNameN)))))</span></code></pre></div>
</div>
</div>
<div id="trivial-inversions" class="section level2">
<h2 class="hasAnchor">
<a href="#trivial-inversions" class="anchor"></a>Trivial inversions</h2>
<div id="representation" class="section level3">
<h3 class="hasAnchor">
<a href="#representation" class="anchor"></a>Representation</h3>
<p>In the triplet of alignments below below, the middle block is on a different strand than its neighbors, and they would be colinear otherwise. This represents an inversion. <em>Without an outgroup it is not possible to know in which of the target or query genome the break happened.</em></p>
<p>Let’s call “gaps” the unaligned regions between the blocks.</p>
<pre><code>┌───────────────┬───────────────┬───────────────┐
│ XSR:101-200:+ │ XSR:201-300:- │ XSR:301-400:+ │  (OKI2018_I69)
└───────────────┴───────────────┴───────────────┘
┌───────────────┬───────────────┬───────────────┐
│  S1:101-200:+ │  S1:201-300:+ │  S1:301-400:+ │  (OSKA2016)
└───────────────┴───────────────┴───────────────┘</code></pre>
<p>It is represented as:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/example.html">example</a></span><span class="op">(</span><span class="st">"flagInversions"</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## flgInv&gt; inv       &lt;- GRanges(c("XSR:101-200:+", "XSR:201-300:-",  "XSR:301-400:+"))
## 
## flgInv&gt; inv$query &lt;- GRanges(c( "S1:101-200",    "S1:201-300",     "S1:301-400"))
## 
## flgInv&gt; flagInversions(inv)
## GRanges object with 3 ranges and 2 metadata columns:
##       seqnames    ranges strand |      query   inv
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;GRanges&gt; &lt;Rle&gt;
##   [1]      XSR   101-200      + | S1:101-200  TRUE
##   [2]      XSR   201-300      - | S1:201-300 FALSE
##   [3]      XSR   301-400      + | S1:301-400 FALSE
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/ivanek/Gviz">"Gviz"</a></span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required package: grid</code></pre>
<pre><code>## 
## Attaching package: 'grid'</code></pre>
<pre><code>## The following object is masked from 'package:Biostrings':
## 
##     pattern</code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>ucscChromosomeNames<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">gen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/GenomeAxisTrack-class.html">GenomeAxisTrack</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"genome"</span><span class="op">)</span>
<span class="va">trk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span><span class="va">inv</span>, name <span class="op">=</span> <span class="st">"inv"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">gen</span>, <span class="va">trk</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicBreaks_vignette_files/figure-html/ideal_breakpoints-1.png" width="700"></p>
</div>
<div id="number-of-trivial-inversions" class="section level3">
<h3 class="hasAnchor">
<a href="#number-of-trivial-inversions" class="anchor"></a>Number of trivial inversions</h3>
<p>More inversions are found after coalescing colinear blocks because of situations where <code>+ - +</code> was <code>+ - - +</code> before collapsing.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 505 ranges and 3 metadata columns:
##         seqnames      ranges strand |     score                   query   inv
##            &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;               &lt;GRanges&gt; &lt;Rle&gt;
##     [1]     chrI  5860-10010      + |     10609  NC_047487.1:8723-12716 FALSE
##     [2]     chrI 11157-11782      + |      1253 NC_047487.1:13035-13680 FALSE
##     [3]     chrI 25371-26528      + |      3651 NC_047487.1:15380-16537 FALSE
##     [4]     chrI 26849-29699      + |      6130 NC_047487.1:17263-20175 FALSE
##     [5]     chrI 29937-30607      + |       767 NC_047487.1:20203-20866 FALSE
##     ...      ...         ...    ... .       ...                     ...   ...
##   [501]     chrM 67039-67870      + |      1441   NC_018044.1:7758-8439 FALSE
##   [502]     chrM 68205-68580      + |       830   NC_018044.1:8783-9180 FALSE
##   [503]     chrM 69178-76168      + |     14528  NC_018044.1:9650-16261 FALSE
##   [504]     chrM 77356-80022      + |      8066 NC_018044.1:53206-55865 FALSE
##   [505]     chrM 80919-85779      + |      6712 NC_018044.1:57885-61592 FALSE
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/showInversions.html">showInversions</a></span><span class="op">(</span><span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 6 ranges and 3 metadata columns:
##       seqnames        ranges strand |     score                     query   inv
##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;                 &lt;GRanges&gt; &lt;Rle&gt;
##   213   chrVII 794407-811237      + |     66296 NC_047493.1:774599-791335  TRUE
##   214   chrVII 823466-828795      - |     22822 NC_047493.1:792155-797486 FALSE
##   215   chrVII 828909-845164      + |     66264 NC_047493.1:799044-815349 FALSE
##   276     chrX 204622-354004      + |    606899 NC_047496.1:201590-351096  TRUE
##   277     chrX 355632-374591      - |     77414 NC_047496.1:353062-371883 FALSE
##   278     chrX 374938-377431      + |      8310 NC_047496.1:373079-375567 FALSE
##   -------
##   seqinfo: 17 sequences (1 circular) from sacCer3 genome</code></pre>
</div>
<div id="isolate-the-left-side-gaps-in-inversions" class="section level3">
<h3 class="hasAnchor">
<a href="#isolate-the-left-side-gaps-in-inversions" class="anchor"></a>Isolate the left-side gaps in inversions</h3>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># lgaps &lt;- function(reduced_gr) {</span>
<span class="co">#   # Flag inversions.</span>
<span class="co">#   #  WHAT FOLLOWS ASSUMES THAT THE FLAG IS ON THE LEFT-SIDE BLOCK OF THE TRIPLE</span>
<span class="co">#   lgaps &lt;- flagInversions(reduced_gr)</span>
<span class="co">#   # Extract inversions</span>
<span class="co">#   invs &lt;- lgaps[lgaps$inv]</span>
<span class="co">#   # Extend inversion of 1 bp so that they overlap with their neighbor gap</span>
<span class="co">#   invs &lt;- shift(invs, 1)</span>
<span class="co">#   # Remove strand information</span>
<span class="co">#   strand(lgaps) &lt;- "*"</span>
<span class="co">#   # Then extract gap positions ignoring strand</span>
<span class="co">#   lgaps &lt;- gaps(lgaps)</span>
<span class="co">#   # Remove bogus gaps on + and - strands</span>
<span class="co">#   lgaps &lt;- lgaps[strand(lgaps) == "*"] </span>
<span class="co">#   # Return the gaps overlapping with the flagged inversions</span>
<span class="co">#   subsetByOverlaps(lgaps, invs)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># # Using coa and not gbs for reason on better detection explained above.</span>
<span class="co"># inv.lgaps &lt;- sapply(coa, lgaps) |&gt; SimpleList()</span>
<span class="co"># </span>
<span class="co"># data.frame(width=width(inv.lgaps$Oki_Osa)) |&gt;</span>
<span class="co">#   ggplot(aes(width)) + geom_histogram() +  scale_x_log10()</span>
<span class="co"># </span>
<span class="co"># inv.lgaps.Seq &lt;- SimpleList()</span>
<span class="co"># </span>
<span class="co"># inv.lgaps.Seq$Oki_Osa &lt;- BSgenome::getSeq(OKI2018_I69,  inv.lgaps$Oki_Osa)</span>
<span class="co"># inv.lgaps.Seq$Oki_Kum &lt;- BSgenome::getSeq(OKI2018_I69,  inv.lgaps$Oki_Kum)</span>
<span class="co"># inv.lgaps.Seq$Osa_Oki &lt;- BSgenome::getSeq(OSKA2016v1.9, inv.lgaps$Osa_Oki)</span>
<span class="co"># inv.lgaps.Seq$Osa_Aom &lt;- BSgenome::getSeq(OSKA2016v1.9, inv.lgaps$Osa_Aom)</span>
<span class="co"># inv.lgaps.Seq$Oki_Bar &lt;- BSgenome::getSeq(OKI2018_I69,  inv.lgaps$Oki_Bar)</span>
<span class="co"># inv.lgaps.Seq$Bar_Oki &lt;- BSgenome::getSeq(Bar2_p4,      inv.lgaps$Bar_Oki)</span></code></pre></div>
</div>
<div id="at-richness" class="section level3">
<h3 class="hasAnchor">
<a href="#at-richness" class="anchor"></a>AT richness</h3>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ## Inverted regions are AT-rich as usual.</span>
<span class="co"># invSeq &lt;- SimpleList()</span>
<span class="co"># invSeq$Oki_Osa &lt;- BSgenome::getSeq(OKI2018_I69,  filterInversions(flagInversions(coa$Oki_Osa)))</span>
<span class="co"># invSeq$Oki_Bar &lt;- BSgenome::getSeq(OKI2018_I69,  filterInversions(flagInversions(coa$Oki_Bar)))</span>
<span class="co"># invSeq$Oki_Kum &lt;- BSgenome::getSeq(OKI2018_I69,  filterInversions(flagInversions(coa$Oki_Kum)))</span>
<span class="co"># invSeq$Kum_Oki &lt;- BSgenome::getSeq(KUM_M3,  swap(filterInversions(flagInversions(coa$Oki_Kum))))</span>
<span class="co"># invSeq$Osa_Oki &lt;- BSgenome::getSeq(OSKA2016v1.9, filterInversions(flagInversions(coa$Osa_Oki)))</span>
<span class="co"># invSeq$Osa_Bar &lt;- BSgenome::getSeq(OSKA2016v1.9, filterInversions(flagInversions(coa$Osa_Bar)))</span>
<span class="co"># invSeq$Osa_Kum &lt;- BSgenome::getSeq(OSKA2016v1.9, filterInversions(flagInversions(coa$Osa_Kum)))</span>
<span class="co"># invSeq$Osa_Aom &lt;- BSgenome::getSeq(OSKA2016v1.9, filterInversions(flagInversions(coa$Osa_Aom)))</span>
<span class="co"># invSeq$Bar_Osa &lt;- BSgenome::getSeq(Bar2_p4,      filterInversions(flagInversions(coa$Bar_Osa)))</span>
<span class="co"># invSeq$Bar_Oki &lt;- BSgenome::getSeq(Bar2_p4,      filterInversions(flagInversions(coa$Bar_Oki)))</span>
<span class="co"># invSeq$Bar_Kum &lt;- BSgenome::getSeq(Bar2_p4,      filterInversions(flagInversions(coa$Bar_Kum)))</span>
<span class="co"># </span>
<span class="co"># sapply(invSeq, \(seq) letterFrequency(seq, "AT", as.prob = TRUE) |&gt; summary())</span>
<span class="co"># genomes.AT</span>
<span class="co"># </span>
<span class="co"># ## But gaps are a bit more AT-rich.</span>
<span class="co"># sapply(inv.lgaps.Seq, \(seq) letterFrequency(seq, "AT", as.prob = TRUE) |&gt; summary())</span>
<span class="co"># </span>
<span class="co"># ## This difference is significant.</span>
<span class="co"># t.test(letterFrequency(invSeq$Oki_Osa, "AT", as.prob = TRUE), letterFrequency(inv.lgaps.Seq$Oki_Osa, "AT", as.prob = TRUE))</span>
<span class="co"># t.test(letterFrequency(invSeq$Osa_Oki, "AT", as.prob = TRUE), letterFrequency(inv.lgaps.Seq$Osa_Oki, "AT", as.prob = TRUE))</span>
<span class="co"># t.test(letterFrequency(invSeq$Oki_Bar, "AT", as.prob = TRUE), letterFrequency(inv.lgaps.Seq$Oki_Bar, "AT", as.prob = TRUE))</span>
<span class="co"># t.test(letterFrequency(invSeq$Oki_Kum, "AT", as.prob = TRUE), letterFrequency(inv.lgaps.Seq$Oki_Kum, "AT", as.prob = TRUE))</span></code></pre></div>
<p>I was tempted to explain this difference with A/T homopolymer stetches. But de novo motif prediction with rGADEM finds AT-rich motifs in any set of Oik sequences that I have tried.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library("rGADEM")</span>
<span class="co"># # Slow !</span>
<span class="co"># # Note that GADEM crashes on sequences longer than 50 kbp.</span>
<span class="co"># gadems &lt;- SimpleList()</span>
<span class="co"># gadems$Oki_Osa &lt;- GADEM(inv.lgaps.Seq$Oki_Osa[width(inv.lgaps.Seq$Oki_Osa) &lt; 50000], verbose=1, genome=OKI2018_I69)</span>
<span class="co"># gadems$Oki_Kum &lt;- GADEM(inv.lgaps.Seq$Oki_Kum[width(inv.lgaps.Seq$Oki_Kum) &lt; 50000], verbose=1, genome=OKI2018_I69)</span>
<span class="co"># </span>
<span class="co"># gadems$Osa_Oki &lt;- GADEM(inv.lgaps.Seq$Osa_Oki[width(inv.lgaps.Seq$Osa_Oki) &lt; 50000], verbose=1, genome=OSKA2016v1.9)</span>
<span class="co"># gadems$Osa_Aom &lt;- GADEM(inv.lgaps.Seq$Osa_Aom[width(inv.lgaps.Seq$Osa_Aom) &lt; 50000], verbose=1, genome=OSKA2016v1.9)</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># gadems$Kum_Oki &lt;- GADEM(invSeq$Kum_Oki[width(invSeq$Kum_Oki) &lt; 50000], verbose=1, genome=KUM_M3)</span>
<span class="co"># gadems$Osa_Oki &lt;- GADEM(invSeq$Osa_Oki[width(invSeq$Osa_Oki) &lt; 50000], verbose=1, genome=OSKA2016v1.9)</span>
<span class="co"># # gadem &lt;- GADEM(invSeq_Oki_O[width(invSeq_Oki_O) &lt; 50000], verbose=1, genome=OKI2018_I69)</span>
<span class="co"># # consensus(gadem)</span>
<span class="co"># # [1] "nAAAAwwnnnrAAAAn"</span>
<span class="co"># </span>
<span class="co"># # gadem &lt;- GADEM(lgapsSeq_Oki_O, verbose=1, genome=OKI2018_I69)</span>
<span class="co"># # consensus(gadem)</span>
<span class="co"># # [1] "rAAGCsGCwwmkCGrCTTyn" "nAAAAwwnnwnAAAAAn"    "nTTTsAAAAw"           "nCCTGAsTCAGGkTTTn"    "nTTTTTnnnnrAAAn"      "wTAAGGGTrTCm"</span>
<span class="co"># </span>
<span class="co"># #saveRDS(gadem, file = "lgapsSeq_Oki_O_gadem.rds")</span>
<span class="co"># gadem &lt;- readRDS("lgapsSeq_Oki_O_gadem.rds")</span>
<span class="co"># </span>
<span class="co"># for (n in seq_along(gadem@motifList)) seqLogo::seqLogo(rGADEM::getPWM(gadem)[[n]])</span>
<span class="co"># pwm &lt;- rGADEM::getPWM(gadem)[[1]]</span>
<span class="co"># </span>
<span class="co"># #RGadem positive control: check introns ?</span>
<span class="co"># # set.seed(1664)</span>
<span class="co"># # intr &lt;- sample(intronicParts(tx_OKI), 1000)</span>
<span class="co"># # intr &lt;- BSgenome::getSeq(OKI2018_I69, intr)</span>
<span class="co"># # gadem &lt;- GADEM(intr, verbose=1, genome=OKI2018_I69)</span>
<span class="co"># # &gt; consensus(gadem)</span>
<span class="co"># # [1] "nAAAAwnnwnAAAwn"</span>
<span class="co"># # Also found motifs like rryCAATTbwTkCGmAkyT</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># #RGadem positive control: check promoters ?</span>
<span class="co"># # set.seed(1664)</span>
<span class="co"># # prom &lt;- sample(promoters(tx_OKI), 1000)</span>
<span class="co"># # prom &lt;- BSgenome::getSeq(OKI2018_I69, prom)</span>
<span class="co"># # gadem &lt;- GADEM(prom, verbose=1, genome=OKI2018_I69)</span>
<span class="co"># # consensus(gadem)</span>
<span class="co"># # [1] "nAAAAwwnwnnAAAAwn" "nTTTTCTTyn"       </span>
<span class="co"># # Also found motifs like TTTmAAAA</span></code></pre></div>
</div>
<div id="representation-of-some-inversions" class="section level3">
<h3 class="hasAnchor">
<a href="#representation-of-some-inversions" class="anchor"></a>Representation of some inversions</h3>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># x &lt;- coa$Oki_Osa |&gt; flagInversions() |&gt; dist2next(ignore.strand = TRUE)</span>
<span class="co"># </span>
<span class="co"># pmpcoord &lt;- showInversions(x)</span>
<span class="co"># </span>
<span class="co"># seqinfo2gieStain &lt;- function(si) {</span>
<span class="co">#   data.frame(</span>
<span class="co">#     chrom      = seqlevels(si),</span>
<span class="co">#     chromStart = 0,</span>
<span class="co">#     chromEnd   = seqlengths(si),</span>
<span class="co">#     name       = seqlevels(si),</span>
<span class="co">#     gieStain   = "gneg"</span>
<span class="co">#   )</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># #r &lt;- 63:70</span>
<span class="co"># r &lt;- 94:98</span>
<span class="co"># #r &lt;- 316:319</span>
<span class="co"># options(ucscChromosomeNames=FALSE)</span>
<span class="co"># gen &lt;- GenomeAxisTrack(name="OKI2018_I69")</span>
<span class="co"># ide &lt;- IdeogramTrack(chromosome = seqlevelsInUse(x[r]), genome = "OKI2018_I69", bands = seqinfo2gieStain(OKI2018_I69))</span>
<span class="co"># trk &lt;- AnnotationTrack(x[r], name = "OKI2018_I69", id=LETTERS[seq_along(r)])</span>
<span class="co"># p1 &lt;- grid::grid.grabExpr(plotTracks(list(ide, gen, trk), featureAnnotation = "id"))</span>
<span class="co"># </span>
<span class="co"># gen2 &lt;- GenomeAxisTrack(name="OSKA2016v1.9")</span>
<span class="co"># ide2 &lt;- IdeogramTrack(chromosome = seqlevelsInUse(x$query[r]), genome = "OSKA2016v1.9", bands = seqinfo2gieStain(OSKA2016v1.9))</span>
<span class="co"># trk2 &lt;- AnnotationTrack(x[r]$query, name = "OSKA2016v1.9", id=LETTERS[seq_along(r)])</span>
<span class="co"># p2 &lt;- grid::grid.grabExpr(plotTracks(list(ide2,gen2,trk2), featureAnnotation = "id"))</span>
<span class="co"># </span>
<span class="co"># gridExtra::grid.arrange(p1, p2, nrow=2)</span></code></pre></div>
</div>
</div>
<div id="trivial-insersions" class="section level2">
<h2 class="hasAnchor">
<a href="#trivial-insersions" class="anchor"></a>Trivial insersions</h2>
<div id="representation-1" class="section level3">
<h3 class="hasAnchor">
<a href="#representation-1" class="anchor"></a>Representation</h3>
<p>The in<b>s</b>ersion below:</p>
<pre><code>┌───────────────┬───────────────┬───────────────┐
│ XSR:101-200:+ │ XSR:201-300:+ │ XSR:301-400:+ │  (OKI2018_I69)
└───────────────┴───────────────┴───────────────┘
┌───────────────┬───────────────┬───────────────┐
│  S1:101-200:+ │  S2:801-900:+ │  S1:201-300:+ │  (OSKA2016)
└───────────────┴───────────────┴───────────────┘</code></pre>
<p>Is represented as:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># example("flagInsersions")</span>
<span class="co"># </span>
<span class="co"># library("Gviz")</span>
<span class="co"># options(ucscChromosomeNames=FALSE)</span>
<span class="co"># gen &lt;- GenomeAxisTrack(name = "genome")</span>
<span class="co"># trk &lt;- AnnotationTrack(ins, name = "ins")</span>
<span class="co"># plotTracks(list(gen, trk))</span></code></pre></div>
<p><em>Without an outgroup it is not possible to know in which genome the break happened.</em></p>
</div>
<div id="detection" class="section level3">
<h3 class="hasAnchor">
<a href="#detection" class="anchor"></a>Detection</h3>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># sapply(gbs, function(gb) sum(flagInsersions(coalesce_contigs(gb))$ins))</span>
<span class="co"># </span>
<span class="co"># y &lt;- flagInsersions(reduced_gr_Oki_O)</span>
<span class="co"># </span>
<span class="co"># #r &lt;- 63:70</span>
<span class="co"># r &lt;- 23:30</span>
<span class="co"># #r &lt;- 316:319</span>
<span class="co"># options(ucscChromosomeNames=FALSE)</span>
<span class="co"># gen &lt;- GenomeAxisTrack(name="OKI2018_I69")</span>
<span class="co"># ide &lt;- IdeogramTrack(chromosome = seqlevelsInUse(x[r]), genome = "OKI2018_I69", bands = seqinfo2gieStain(OKI2018_I69))</span>
<span class="co"># trk &lt;- AnnotationTrack(x[r], name = "OKI2018_I69", id=LETTERS[seq_along(r)])</span>
<span class="co"># p1 &lt;- grid::grid.grabExpr(plotTracks(list(ide, gen, trk), featureAnnotation = "id"))</span>
<span class="co"># </span>
<span class="co"># gen2 &lt;- GenomeAxisTrack(name="OSKA2016")</span>
<span class="co"># ide2 &lt;- IdeogramTrack(chromosome = seqlevelsInUse(x$query[r])[1], genome = "OSKA2016", bands = seqinfo2gieStain(OSKA2016))</span>
<span class="co"># trk2 &lt;- AnnotationTrack(x[r]$query, name = "OSKA2016", id=LETTERS[seq_along(r)])</span>
<span class="co"># p2 &lt;- grid::grid.grabExpr(plotTracks(list(ide2,gen2,trk2), featureAnnotation = "id"))</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># gen3 &lt;- GenomeAxisTrack(name="OSKA2016")</span>
<span class="co"># ide3 &lt;- IdeogramTrack(chromosome = seqlevelsInUse(x$query[r])[2], genome = "OSKA2016", bands = seqinfo2gieStain(OSKA2016))</span>
<span class="co"># trk3 &lt;- AnnotationTrack(x[r]$query, name = "OSKA2016", id=LETTERS[seq_along(r)])</span>
<span class="co"># p3 &lt;- grid::grid.grabExpr(plotTracks(list(ide3,gen3,trk3), featureAnnotation = "id"))</span>
<span class="co"># </span>
<span class="co"># gridExtra::grid.arrange(p1, p2, p3, nrow=3)</span></code></pre></div>
</div>
</div>
<div id="tandem-repeats" class="section level2">
<h2 class="hasAnchor">
<a href="#tandem-repeats" class="anchor"></a>Tandem repeats</h2>
<p>Repeated regions of genomes are notoriously difficult to assemble and align. Thus, information about an alignment stop’s proximity to a tandem repeat is useful information in breakpoint classification. Given a user-defined tolerance, the function <code>tan_bp</code> will classify alignment stops as either far from, close to, or within tandem repeats (elements 1,2 and 3 of the produced list, respectively). The function requires the tandem repeat coverage of one of the genomes. By default, the function will expect the target genome tandem coverage. Setting <code>query_tf = TRUE</code> will switch the outcome to be query genome oriented.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># tan_O &lt;- rtracklayer::import.gff3(system.file("extdata/OSKA2016.tantanf4w100.gff3.gz",  package="GenomicBreaks"))</span>
<span class="co"># tan_O &lt;- GRanges(tan_O, seqinfo = seqinfo(OSKA2016))</span>
<span class="co"># tan_gr_O &lt;- tan_bp(gr_ob = gr_O_Oki, tan = tan_O, tol = 50)</span>
<span class="co"># tan_gr_O</span>
<span class="co"># </span>
<span class="co"># tan_Oki &lt;- rtracklayer::import.gff3(system.file("extdata/I69-5.tantanf4w100.gff3.gz",  package="GenomicBreaks"))</span>
<span class="co"># tan_Oki &lt;- GRanges(tan_Oki, seqinfo = seqinfo(OKI2018_I69))</span>
<span class="co"># tan_gr_Oki &lt;- tan_bp(gr_ob = gr_O_Oki, tan = tan_Oki, tol = 50, query_tf = TRUE)</span>
<span class="co"># tan_gr_Oki</span></code></pre></div>
<p>We can use another script to plot the density of tandem repeats around alignment stops (and breakpoints). In the following plots, breakpoints are aligned at the centre of the plots. Intuitively, this function can be generalised to see the density of any binary coverage characteristic around breakpoints, given a GRanges object with per-base coverage of a given characteristic. Here we stick with tandem repeat info:</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># hm_tan_O &lt;- gr_O_Oki %&gt;%</span>
<span class="co">#   feature_coverage(tan_O, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T&gt;%</span>
<span class="co">#   plotHeatmap</span>
<span class="co"># </span>
<span class="co"># hm_red_tan_O &lt;- reduced_gr_O_Oki %&gt;%</span>
<span class="co">#   feature_coverage(tan_O, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T&gt;%</span>
<span class="co">#   plotHeatmap</span>
<span class="co"># </span>
<span class="co"># hm_tan_Oki &lt;- gr_Oki_O %&gt;%</span>
<span class="co">#   feature_coverage(tan_Oki, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T&gt;%</span>
<span class="co">#   plotHeatmapList</span></code></pre></div>
</div>
<div id="coverage" class="section level2">
<h2 class="hasAnchor">
<a href="#coverage" class="anchor"></a>Coverage</h2>
<p>Another characteristic of alignment stops that we can examine is the coverage depth over said alignment stops. By aligning raw reads to assembled genomes, we can obtain per-base coverage depth for target or query genome. Low coverage directly on and surrounding an alignment stop may suggest unreliability in it being a true breakpoint. The function <code>bp_coverage</code> returns a GRanges object of the bps, with associated averaged and point converage in the metadatacolumns.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># cov_gr_O_Oki &lt;- bp_coverage(gr_ob = q_Oki, cov_gr = Oki_cov_pb, win = 50)</span>
<span class="co"># cov_gr_O_Oki</span></code></pre></div>
</div>
</div>
<div id="examples" class="section level1">
<h1 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h1>
<div id="nucleaic-acid-content-heatmaps" class="section level2">
<h2 class="hasAnchor">
<a href="#nucleaic-acid-content-heatmaps" class="anchor"></a>Nucleaic acid content heatmaps</h2>
<p>The <em>GenomicBreaks</em> function <code>bp_heatmap</code> plots nucleic acid content in windows centrered on alignment stops. Lets look at <code>GC</code> content before and after coalescing:</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library(heatmaps)</span>
<span class="co"># hm_GC1 &lt;- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "GC", direction = "left") |&gt; SimpleList()</span>
<span class="co"># hm_GC2 &lt;- sapply(coa, bp_heatmap, basep_range = 1000, pat = "GC", direction = "left") |&gt; SimpleList()</span>
<span class="co"># </span>
<span class="co"># maxHeight &lt;- 2000</span>
<span class="co"># sm_GC1 &lt;- sapply(hm_GC1, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># sm_GC2 &lt;- sapply(hm_GC2, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># sm_GC3 &lt;- sapply(hm_GC3, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># </span>
<span class="co"># plotHeatmapList(sm_GC1)</span>
<span class="co"># plotHeatmapList(sm_GC2)</span>
<span class="co"># plotHeatmapList(sm_GC3)</span></code></pre></div>
<p>The alignment stops are ordered in such a way that start (or left breaks) are centred on the top half of the plot, and end alignment stops are centred on the bottom. This is why we see different directionality in the plots. At the alignment stops, the GC content seems to be lower, but higher once within the aligned region.</p>
<p>Now looking at TATA content:</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># hm_GC1_TATA &lt;- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "TATA", direction = "left") |&gt; SimpleList()</span>
<span class="co"># hm_GC2_TATA &lt;- sapply(coa, bp_heatmap, basep_range = 1000, pat = "TATA", direction = "left") |&gt; SimpleList()</span>
<span class="co"># </span>
<span class="co"># maxHeight &lt;- 2000</span>
<span class="co"># sm_GC1_TATA &lt;- sapply(hm_GC1_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># sm_GC2_TATA &lt;- sapply(hm_GC2_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># sm_GC3_TATA &lt;- sapply(hm_GC3_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># </span>
<span class="co"># plotHeatmapList(sm_GC1_TATA)</span>
<span class="co"># plotHeatmapList(sm_GC2_TATA)</span>
<span class="co"># plotHeatmapList(sm_GC3_TATA)</span></code></pre></div>
<p>The directionaility here is consistent with the <code>GC</code> content analysis. Furthermore; “TATA box sequence can act as a basal promoter element not only for RNA polymerase II (RNAP II) transcription, but also for transcription by RNA polymerase III (RNAP III)” - Wang Y, Jensen RC, Stumph WE. Role of TATA box sequence and orientation in determining RNA polymerase II/III transcription specificity. Nucleic Acids Res. 1996;24(15):3100–3106. <a href="doi:10.1093/nar/24.15.3100" class="uri">doi:10.1093/nar/24.15.3100</a>. However, the areas are not particularly enriched for TATA boxes, so it does not necessarily show that breakpoints are occurring directly after promoter regions.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># hm_GC1_mot &lt;- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "AGCNGC", direction = "left") |&gt; SimpleList()</span>
<span class="co"># hm_GC2_mot &lt;- sapply(coa, bp_heatmap, basep_range = 1000, pat = "AGCNGC", direction = "left") |&gt; SimpleList()</span>
<span class="co"># </span>
<span class="co"># maxHeight &lt;- 2000</span>
<span class="co"># sm_GC1_mot &lt;- sapply(hm_GC1_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># sm_GC2_mot &lt;- sapply(hm_GC2_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># sm_GC3_mot &lt;- sapply(hm_GC3_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="co"># </span>
<span class="co"># plotHeatmapList(sm_GC1_mot)</span>
<span class="co"># plotHeatmapList(sm_GC2_mot)</span>
<span class="co"># plotHeatmapList(sm_GC3_mot)</span></code></pre></div>
</div>
<div id="cluster-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#cluster-analysis" class="anchor"></a>Cluster analysis</h2>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># # Accessory functions</span>
<span class="co"># clusterHeatmap &lt;- function(hm, k = 2) kmeans(image(hm), k)</span>
<span class="co"># </span>
<span class="co"># orderHeatmap &lt;- function(hm, cl) {</span>
<span class="co">#   image(hm) &lt;- image(hm)[order(cl$cluster),]</span>
<span class="co">#   hm</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># subsetHeatmap &lt;- function(hm, bool) {</span>
<span class="co">#   image(hm) &lt;- image(hm)[bool,]</span>
<span class="co">#   hm</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># # DAta</span>
<span class="co"># hm_GC   &lt;- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "GC",     direction = "left")</span>
<span class="co"># hm_TATA &lt;- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "TATA",   direction = "left")</span>
<span class="co"># hm_tan  &lt;- feature_coverage(gr_Oki_O, tan_Oki, win = 1000, lab = "Tandem", direction = "left")</span>
<span class="co"># </span>
<span class="co"># # Define 3 K-mean clusters</span>
<span class="co"># clusterHeatmap(hm_GC, 3) -&gt; hm_cl_k4</span>
<span class="co"># </span>
<span class="co"># # Order and smooth</span>
<span class="co"># shm_GC &lt;- orderHeatmap(hm_GC, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># shm_TATA &lt;- orderHeatmap(hm_TATA, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># shm_tan &lt;- orderHeatmap(hm_tan, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># # Plot the ordered data</span>
<span class="co"># plotHeatmapList(list(shm_GC, shm_TATA, shm_tan))</span>
<span class="co"># par(mfrow = c(1,1)) # plotHeatmapList messes with mrfrow...</span>
<span class="co"># plotHeatmapMeta(list(hm_GC, hm_TATA, hm_tan))</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(subsetHeatmap(hm_GC, hm_cl_k4$cluster == 1),</span>
<span class="co">#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 2),</span>
<span class="co">#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 3)))</span>
<span class="co"># </span>
<span class="co"># kmeanOrderedPlot &lt;- function (hm, k=3, seed = 1) {</span>
<span class="co">#   set.seed(seed)</span>
<span class="co">#   </span>
<span class="co">#   cl &lt;- kmeans(image(hm), k)</span>
<span class="co">#   mat &lt;- image(hm)[order(cl$cluster),]</span>
<span class="co">#   </span>
<span class="co">#   kmhm = Heatmap(</span>
<span class="co">#     mat,</span>
<span class="co">#     coords=c(-500,500),</span>
<span class="co">#     label="kmeans",</span>
<span class="co">#     scale=range(mat))</span>
<span class="co">#   </span>
<span class="co">#   plotHeatmapList(kmhm,</span>
<span class="co">#                   cex.label=1.5,</span>
<span class="co">#                   partition=as.vector(table(cl$cluster)),</span>
<span class="co">#                   partition.legend=TRUE,</span>
<span class="co">#                   partition.lines=TRUE,</span>
<span class="co">#                   legend=TRUE,</span>
<span class="co">#                   legend.pos="r",</span>
<span class="co">#                   legend.width=0.3)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># kmeanOrderedPlot(shm_GC)</span></code></pre></div>
</div>
<div id="evidence-for-breakpoint-hotspots" class="section level2">
<h2 class="hasAnchor">
<a href="#evidence-for-breakpoint-hotspots" class="anchor"></a>Evidence for breakpoint hotspots</h2>
<p>The function <code>bp_pair_analysis</code> takes two pairwise alignments with the same target genome, and plots the alignment stops on to the centred alignment stops of the other. In the plot below, Osaka is there consistent target genome, and as such acts as a coordinate system to relate alignment stops across genomes. Lets produce two of these 3-way analyses; one between Osaka-Okinawa-Norway, and one between Osaka-Okinawa-Aomori</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># hm1 &lt;- bp_pair_analysis(gr_ref_q1 = gr_O_Oki, gr_ref_q2 = gr_O_N, win = 1000, lab = "Oki~Nor")</span>
<span class="co"># plotHeatmapMeta(hm1)</span>
<span class="co"># </span>
<span class="co"># hm2 &lt;- bp_pair_analysis(gr_ref_q1 = gr_O_Oki, gr_ref_q2 = gr_O_A, win = 1000, lab = "Oki~Aom")</span>
<span class="co"># plotHeatmapMeta(hm2)</span></code></pre></div>
<p>The accumulation of alignment breaks of one pairwise alignment onto another suggests the existence of breakpoint hotspots; regions where breaks are far more likely to occur. Intuitively, this would lead to the presumption of synteny blocks, also. The pattern is consistent after coalescing, too;</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># reduced_gr_O_N &lt;- coalesce_contigs(gr_ob = gr_O_N, tol = 500)</span>
<span class="co"># </span>
<span class="co"># hm3 &lt;- bp_pair_analysis(gr_ref_q1 = reduced_gr_O_Oki, gr_ref_q2 = reduced_gr_O_N, win = 1000, lab = "Oki~Nor")</span>
<span class="co"># plotHeatmapMeta(hm3)</span></code></pre></div>
</div>
<div id="tandem-repeat-coverage-around-alignment-stops-breakpoints" class="section level2">
<h2 class="hasAnchor">
<a href="#tandem-repeat-coverage-around-alignment-stops-breakpoints" class="anchor"></a>Tandem repeat coverage around alignment stops (breakpoints)</h2>
<p>The function <code>feature_coverage</code> will plot the coverage of tandem repeats around centred alignment stops.</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># hm_tan1 &lt;- feature_coverage(gr_ob = gr_O_Oki, feat = tan_O, win = 4000, lab = "Osk ~ Oki", direction = "left")</span>
<span class="co"># plotHeatmapMeta(hm_tan1)</span>
<span class="co">#  </span>
<span class="co"># hm_tan2 &lt;- feature_coverage(gr_ob = reduced_gr_O_Oki, feat = tan_O, win = 4000, lab = "OSK~Oki (coalesced)", direction = "left")</span>
<span class="co"># plotHeatmapMeta(hm_tan2)</span>
<span class="co"># </span>
<span class="co"># # should I also add comparison of proportions of tandem proximities before and after coalescion?</span></code></pre></div>
<p>There appears to be less tandem repeats around alignment stops after coalescion, inferring that perhaps that some are artefacts of difficult alignment near to tandem repeats.</p>
</div>
<div id="gene-feature-coverage-around-alignment-stops-breakpoints" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-feature-coverage-around-alignment-stops-breakpoints" class="anchor"></a>Gene feature coverage around alignment stops (breakpoints)</h2>
<p>The function <code>feature_coverage</code> will plot the coverage of tandem repeats around centred alignment stops.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># #current_GB &lt;- gr_Oki_O # To ease recomputing with another object</span>
<span class="co"># current_GB &lt;- reduced_gr_Oki_O</span>
<span class="co"># hmList_OKI &lt;- list()</span>
<span class="co"># </span>
<span class="co"># hmList_OKI$genes      &lt;- feature_coverage(current_GB, genes(tx_OKI),                         win = 2000, lab = "Oki genes", direction = "left")</span>
<span class="co"># hmList_OKI$promoters  &lt;- feature_coverage(current_GB, promoters(tx_OKI),                     win = 2000, lab = "Oki promoters", direction = "left")</span>
<span class="co"># hmList_OKI$exons      &lt;- feature_coverage(current_GB, exonicParts(tx_OKI),                   win = 2000, lab = "Oki exons", direction = "left")</span>
<span class="co"># hmList_OKI$introns    &lt;- feature_coverage(current_GB, intronicParts(tx_OKI),                 win = 2000, lab = "Oki introns", direction = "left")</span>
<span class="co"># hmList_OKI$cds        &lt;- feature_coverage(current_GB, cds(tx_OKI),                           win = 2000, lab = "Oki cds", direction = "left")</span>
<span class="co"># hmList_OKI$utrs5      &lt;- feature_coverage(current_GB, unlist(fiveUTRsByTranscript(tx_OKI)),  win = 2000, lab = "Oki 5′ UTRs", direction = "left")</span>
<span class="co"># hmList_OKI$utrs3      &lt;- feature_coverage(current_GB, unlist(threeUTRsByTranscript(tx_OKI)), win = 2000, lab = "Oki 3′ UTRs", direction = "left")</span>
<span class="co"># hmList_OKI$pwm        &lt;- feature_coverage(current_GB, pwmHits                              , win = 2000, lab = "PWM", direction = "left")</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(hmList_OKI)</span>
<span class="co"># plotHeatmapMeta(hmList_OKI$genes)</span>
<span class="co"># plotHeatmapMeta(hmList_OKI[c("exons", "cds")])</span>
<span class="co"># plotHeatmapMeta(hmList_OKI[c("exons", "introns", "utrs5", "utrs3", "pwm")])</span>
<span class="co"># plotHeatmapMeta(hmList_OKI[c("introns", "utrs5", "utrs3", "pwm")])</span>
<span class="co"># plotHeatmapMeta(hmList_OKI[c("utrs5", "utrs3", "pwm")])</span></code></pre></div>
<p>Aligned regions are enriched in exons and depleted in introns. Is it because exons are more alignable or is it because it is deleterous to break genes in exons ?</p>
</div>
<div id="gene-feature-coverage-around-pwm-hits" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-feature-coverage-around-pwm-hits" class="anchor"></a>Gene feature coverage around PWM hits</h2>
<p>Let’s look at coverage around PWM hits of the AAGCsGCwwmkCGrCTTyn motif</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># current_GB &lt;- pwmHits</span>
<span class="co"># </span>
<span class="co"># hmList_PWM_OKI &lt;- list()</span>
<span class="co"># hmList_PWM_OKI$genes     &lt;- feature_coverage(current_GB, genes(tx_OKI),                         win = 2000, lab = "Oki genes", direction = "left")</span>
<span class="co"># hmList_PWM_OKI$promoters &lt;- feature_coverage(current_GB, promoters(tx_OKI),                     win = 2000, lab = "Oki promoters", direction = "left")</span>
<span class="co"># hmList_PWM_OKI$exons     &lt;- feature_coverage(current_GB, exonicParts(tx_OKI),                   win = 2000, lab = "Oki exons", direction = "left")</span>
<span class="co"># hmList_PWM_OKI$introns   &lt;- feature_coverage(current_GB, intronicParts(tx_OKI),                 win = 2000, lab = "Oki introns", direction = "left")</span>
<span class="co"># hmList_PWM_OKI$cds       &lt;- feature_coverage(current_GB, cds(tx_OKI),                           win = 2000, lab = "Oki cds", direction = "left")</span>
<span class="co"># hmList_PWM_OKI$utrs5     &lt;- feature_coverage(current_GB, unlist(fiveUTRsByTranscript(tx_OKI)),  win = 2000, lab = "Oki 5′ UTRs", direction = "left")</span>
<span class="co"># hmList_PWM_OKI$utrs3     &lt;- feature_coverage(current_GB, unlist(threeUTRsByTranscript(tx_OKI)), win = 2000, lab = "Oki 3′ UTRs", direction = "left")</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(hmList_PWM_OKI)</span>
<span class="co"># plotHeatmapMeta(hmList_PWM_OKI[c("exons", "introns")])</span>
<span class="co"># plotHeatmapMeta(hmList_PWM_OKI[c("exons", "utrs5", "utrs3")])</span>
<span class="co"># plotHeatmapMeta(hmList_PWM_OKI[c("utrs5", "utrs3")])</span></code></pre></div>
<p>They are enriched in introns…</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># # Sandbox to make sanity checks.</span>
<span class="co"># </span>
<span class="co"># a &lt;- reduced_gr_Oki_O</span>
<span class="co"># b &lt;- unlist(fiveUTRsByTranscript(tx_OKI))</span>
<span class="co"># </span>
<span class="co"># hm_a &lt;- feature_coverage(a, b, win = 2000, lab = "All ranges", direction = "left")</span>
<span class="co"># hm_b &lt;- feature_coverage(subsetByOverlaps(a,  b), b, win = 2000, lab = "Subsetted ranges", direction = "left")</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(hm_a))</span>
<span class="co"># plotHeatmapMeta(list(hm_b))</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
