<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Structural Variants â€¢ GenomicBreaks</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Structural Variants">
<meta property="og:description" content="GenomicBreaks">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicBreaks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.12.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/GenomicBreaks.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/StructuralVariants.html">Structural Variants</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/GenomicBreaks/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="StructuralVariants_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Structural Variants</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">13 December, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/HEAD/vignettes/StructuralVariants.Rmd" class="external-link"><code>vignettes/StructuralVariants.Rmd</code></a></small>
      <div class="hidden name"><code>StructuralVariants.Rmd</code></div>

    </div>

    
    
<div class="section level1">
<h1 id="structural-variants">Structural Variants<a class="anchor" aria-label="anchor" href="#structural-variants"></a>
</h1>
<p>The <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> objects here are provided as <em>lazy data</em> and by convention their name always start with <code>example</code>.</p>
<div class="section level2">
<h2 id="colinearity">Colinearity<a class="anchor" aria-label="anchor" href="#colinearity"></a>
</h2>
<div class="section level3">
<h3 id="definition">Definition<a class="anchor" aria-label="anchor" href="#definition"></a>
</h3>
<p>We define <em>structural variants</em> as patterns of interruption of the colinearity between genomes.</p>
<p>Colinearity is a property of pairs of mapped regions.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleColinear</span>, main <span class="op">=</span> <span class="st">"Two colinear regions"</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/colinearRegions_pair-1.png" width="700"></p>
<p>Regions that mapped on reverse strands <strong>and</strong> in reverse order are also colinear.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="fu"><a href="../reference/reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleColinear</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Two colinear regions on different strands"</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/colinearRegions_pair_minus_strand-1.png" width="700"></p>
<p>Regions that are only in reverse order, or only mapped on opposite strands <strong>are not colinear</strong>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleNotColinear</span>, main <span class="op">=</span> <span class="st">"Same strand but not colinear"</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/notColinearRegions_pair-1.png" width="700"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="fu"><a href="../reference/reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleNotColinear</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Both on opposite strand but not colinear"</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/notColinearRegions_pair-2.png" width="700"></p>
<p>In this example, the syntenic chromosomes are called <code>chrA</code> and <code>chrB</code>, but the names do not matter.</p>
</div>
<div class="section level3">
<h3 id="technical-details">Technical details<a class="anchor" aria-label="anchor" href="#technical-details"></a>
</h3>
<p>Remember that the <code><a href="../reference/flagColinearAlignments.html">flagColinearAlignments()</a></code> function puts the flag on the first member of the pair, ordered by the sequence on the <em>target</em> genome.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="va">exampleColinear</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     chrA   100-150      + | chrB:100-150      TRUE</span>
<span class="co">##   [2]     chrA   251-300      + | chrB:251-300     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="fu"><a href="../reference/reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleColinear</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     chrA      1-50      - | chrB:251-300      TRUE</span>
<span class="co">##   [2]     chrA   151-201      - | chrB:100-150     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Note the tolerance window</span>
<span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="va">exampleColinear</span>, tol <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     chrA   100-150      + | chrB:100-150     FALSE</span>
<span class="co">##   [2]     chrA   251-300      + | chrB:251-300     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Strandless mappings can also be colinear</span>
<span class="va">exampleColinear</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">"*"</span><span class="op">)</span> |&gt; <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     chrA   100-150      * | chrB:100-150      TRUE</span>
<span class="co">##   [2]     chrA   251-300      * | chrB:251-300     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="va">exampleNotColinear</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     chrA   100-150      + | chrB:201-251     FALSE</span>
<span class="co">##   [2]     chrA   251-300      + |  chrB:50-100     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<p>To make some sanity checks against the other flagging functions there is also an object containing three colinear mappings.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="va">exampleColinear3</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     chrA   100-200      + | chrB:100-200      TRUE</span>
<span class="co">##   [2]     chrA   201-300      + | chrB:201-300      TRUE</span>
<span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
</div>
<div class="section level3">
<h3 id="limitation">Limitation<a class="anchor" aria-label="anchor" href="#limitation"></a>
</h3>
<p>At the moment we can not study the structural variants that are entirely contained in an aligned region, for instance an indel represented as an alignment gap.</p>
</div>
</div>
<div class="section level2">
<h2 id="inversions">Inversions<a class="anchor" aria-label="anchor" href="#inversions"></a>
</h2>
<div class="section level3">
<h3 id="trivial-inversions">Trivial inversions<a class="anchor" aria-label="anchor" href="#trivial-inversions"></a>
</h3>
<p>Trivial inversions are triplets of mapped regions that would be colinear if the strand of the second one were reversed. Note that, like for colinear patterns, the flag is put on the first mapping of the triplet.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleInversion</span> |&gt; <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 3 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear   inv</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt; &lt;Rle&gt;</span>
<span class="co">##   [1]     chrA   100-190      + | chrB:100-190     FALSE  TRUE</span>
<span class="co">##   [2]     chrA   210-291      - | chrB:210-291     FALSE FALSE</span>
<span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleInversion</span>, main <span class="op">=</span> <span class="st">"An inverted region flanked by colinear regions"</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/inversions-1.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="fu"><a href="../reference/reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleInversion</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"An inverted region flanked by colinear regions (opposite strands)"</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/inversions-2.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Removing the inversion restores colinearity</span>
<span class="va">exampleInversion</span><span class="op">[</span><span class="op">-</span><span class="fl">2</span><span class="op">]</span> |&gt; <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     chrA   100-190      + | chrB:100-190      TRUE</span>
<span class="co">##   [2]     chrA   301-400      + | chrB:301-400     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Flipping the inversion restores colinearity</span>
<span class="va">exampleInversion</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">"+"</span><span class="op">)</span> |&gt; <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     chrA   100-190      + | chrB:100-190      TRUE</span>
<span class="co">##   [2]     chrA   210-291      + | chrB:210-291      TRUE</span>
<span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Note the tolerance window</span>
<span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="va">exampleInversion</span>, tol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query   inv</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span>
<span class="co">##   [1]     chrA   100-190      + | chrB:100-190 FALSE</span>
<span class="co">##   [2]     chrA   210-291      - | chrB:210-291 FALSE</span>
<span class="co">##   [3]     chrA   301-400      + | chrB:301-400 FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
</div>
<div class="section level3">
<h3 id="double-inversions">Double inversions<a class="anchor" aria-label="anchor" href="#double-inversions"></a>
</h3>
<p>Double inversions are the result of two overlapping trivial inversions, and must be detected with a different approach.</p>
<p>Here we show a triplet resulting from a double inversion, flanked by two regions that did not move.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Starting point: 5 colinear blocks</span>
<span class="va">exampleColinear5</span> |&gt; <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions-1.png" width="700"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Then, lets have an inversion that encompasses blocks 1 and 2.</span>
<span class="va">exampleInversion5uncollapsed</span> |&gt; <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions-2.png" width="700"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Finally another inversion takes place around the new blocks 3 and 4</span>
<span class="va">exampleDoubleInversion</span> |&gt; <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions-3.png" width="700"></p>
<p>Same story but showing coalesced alignments.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Starting point: 5 colinear blocks</span>
<span class="va">exampleColinear5</span> |&gt; <span class="fu"><a href="../reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions2-1.png" width="700"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Then, lets have an inversion that encompasses blocks 1 and 2.</span>
<span class="va">exampleInversion5uncollapsed</span> |&gt; <span class="fu"><a href="../reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions2-2.png" width="700"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Finally another inversion takes place around the new blocks 3 and 4</span>
<span class="va">exampleDoubleInversion</span> |&gt; <span class="fu"><a href="../reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions2-3.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="transolocations">Transolocations<a class="anchor" aria-label="anchor" href="#transolocations"></a>
</h2>
<p>If a region has moved, but is not an inversion, then it is a translocation.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleTranslocation</span> |&gt; <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 3 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear   inv</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt; &lt;Rle&gt;</span>
<span class="co">##   [1]     chrA   100-200      + | chrB:100-200     FALSE FALSE</span>
<span class="co">##   [2]     chrA   201-300      + | chrC:201-300     FALSE FALSE</span>
<span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleTranslocation</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/translocation-1.png" width="700"></p>
<p>It does not matter if the region moved on the minus strand</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleTranslocation2</span> |&gt; <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 3 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |        query  colinear   inv</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt; &lt;Rle&gt;</span>
<span class="co">##   [1]     chrA   100-200      + | chrB:100-200     FALSE FALSE</span>
<span class="co">##   [2]     chrA   201-300      - | chrC:201-300     FALSE FALSE</span>
<span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleTranslocation2</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/translocationOnMinus-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="insertions-deletions-indels">Insertions / Deletions (indels)<a class="anchor" aria-label="anchor" href="#insertions-deletions-indels"></a>
</h2>
<p>NOTE that searching for insertion/deletions make little sense in one-to-one alignment unless the maximal distance for colinearity is set to a meaningful value.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleDeletion</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 1 metadata column:</span>
<span class="co">##       seqnames    ranges strand |        query</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="co">##   [1]     chrA   100-200      + | chrB:100-200</span>
<span class="co">##   [2]     chrA   201-300      + | chrC:401-500</span>
<span class="co">##   [3]     chrA   301-400      + | chrB:201-300</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleDeletion</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/indels-1.png" width="700"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleInsertion</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 1 metadata column:</span>
<span class="co">##       seqnames    ranges strand |        query</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="co">##   [1]     chrA   100-200      + | chrB:100-200</span>
<span class="co">##   [2]     chrA   201-300      + | chrB:301-400</span>
<span class="co">##   [3]     chrC   401-500      + | chrB:201-300</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 2 sequences from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleInsertion</span><span class="op">)</span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/indels-2.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
