<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="GenomicBreaks">
<title>Structural Variants â€¢ GenomicBreaks</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Structural Variants">
<meta property="og:description" content="GenomicBreaks">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicBreaks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.14.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/GenomicBreaks.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/StructuralVariants.html">Structural Variants</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/oist/GenomicBreaks/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Structural Variants</h1>
                        <h4 data-toc-skip class="author">Charles
Plessy</h4>
            
            <h4 data-toc-skip class="date">22 May, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/GenomicBreaks/blob/HEAD/vignettes/StructuralVariants.Rmd" class="external-link"><code>vignettes/StructuralVariants.Rmd</code></a></small>
      <div class="d-none name"><code>StructuralVariants.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="structural-variants">Structural Variants<a class="anchor" aria-label="anchor" href="#structural-variants"></a>
</h2>
<p>The <code><a href="../reference/GBreaks-class.html">GBreaks()</a></code> objects here are provided as <em>lazy
data</em> and by convention their name always start with
<code>example</code>.</p>
<div class="section level3">
<h3 id="colinearity">Colinearity<a class="anchor" aria-label="anchor" href="#colinearity"></a>
</h3>
<div class="section level4">
<h4 id="definition">Definition<a class="anchor" aria-label="anchor" href="#definition"></a>
</h4>
<p>We define <em>structural variants</em> as patterns of interruption of
the colinearity between genomes.</p>
<p>Colinearity is a property of pairs of mapped regions.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleColinear</span>, main <span class="op">=</span> <span class="st">"Two colinear regions"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/colinearRegions_pair-1.png" width="700"></p>
<p>Regions that mapped on reverse strands <strong>and</strong> in
reverse order are also colinear.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="fu"><a href="../reference/reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleColinear</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Two colinear regions on different strands"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/colinearRegions_pair_minus_strand-1.png" width="700"></p>
<p>Regions that are only in reverse order, or only mapped on opposite
strands <strong>are not colinear</strong>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleNotColinear</span>, main <span class="op">=</span> <span class="st">"Same strand but not colinear"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/notColinearRegions_pair-1.png" width="700"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="fu"><a href="../reference/reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleNotColinear</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Both on opposite strand but not colinear"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/notColinearRegions_pair-2.png" width="700"></p>
<p>In this example, the syntenic chromosomes are called
<code>chrA</code> and <code>chrB</code>, but the names do not
matter.</p>
</div>
<div class="section level4">
<h4 id="technical-details">Technical details<a class="anchor" aria-label="anchor" href="#technical-details"></a>
</h4>
<p>Remember that the <code><a href="../reference/flagColinearAlignments.html">flagColinearAlignments()</a></code> function puts
the flag on the first member of the pair, ordered by the sequence on the
<em>target</em> genome.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="va">exampleColinear</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-150      + | chrB:100-150      TRUE</span></span>
<span><span class="co">##   [2]     chrA   251-300      + | chrB:251-300     FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="fu"><a href="../reference/reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleColinear</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     chrA      1-50      - | chrB:251-300      TRUE</span></span>
<span><span class="co">##   [2]     chrA   151-201      - | chrB:100-150     FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note the tolerance window</span></span>
<span><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="va">exampleColinear</span>, tol <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-150      + | chrB:100-150     FALSE</span></span>
<span><span class="co">##   [2]     chrA   251-300      + | chrB:251-300     FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Strandless mappings can also be colinear</span></span>
<span><span class="va">exampleColinear</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">"*"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-150      * | chrB:100-150      TRUE</span></span>
<span><span class="co">##   [2]     chrA   251-300      * | chrB:251-300     FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="va">exampleNotColinear</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-150      + | chrB:201-251     FALSE</span></span>
<span><span class="co">##   [2]     chrA   251-300      + |  chrB:50-100     FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<p>To make some sanity checks against the other flagging functions there
is also an object containing three colinear mappings.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="va">exampleColinear3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-200      + | chrB:100-200      TRUE</span></span>
<span><span class="co">##   [2]     chrA   201-300      + | chrB:201-300      TRUE</span></span>
<span><span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="limitation">Limitation<a class="anchor" aria-label="anchor" href="#limitation"></a>
</h4>
<p>At the moment we can not study the structural variants that are
entirely contained in an aligned region, for instance an indel
represented as an alignment gap.</p>
</div>
</div>
<div class="section level3">
<h3 id="inversions">Inversions<a class="anchor" aria-label="anchor" href="#inversions"></a>
</h3>
<div class="section level4">
<h4 id="trivial-inversions">Trivial inversions<a class="anchor" aria-label="anchor" href="#trivial-inversions"></a>
</h4>
<p>Trivial inversions are triplets of mapped regions that would be
colinear if the strand of the second one were reversed. Note that, like
for colinear patterns, the flag is put on the first mapping of the
triplet.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleInversion</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 3 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear   inv</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt; &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-190      + | chrB:100-190     FALSE  TRUE</span></span>
<span><span class="co">##   [2]     chrA   210-291      - | chrB:210-291     FALSE FALSE</span></span>
<span><span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleInversion</span>, main <span class="op">=</span> <span class="st">"An inverted region flanked by colinear regions"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/inversions-1.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="fu"><a href="../reference/reverse.html">reverse</a></span><span class="op">(</span><span class="va">exampleInversion</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"An inverted region flanked by colinear regions (opposite strands)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/inversions-2.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Removing the inversion restores colinearity</span></span>
<span><span class="va">exampleInversion</span><span class="op">[</span><span class="op">-</span><span class="fl">2</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 2 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-190      + | chrB:100-190      TRUE</span></span>
<span><span class="co">##   [2]     chrA   301-400      + | chrB:301-400     FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Flipping the inversion restores colinearity</span></span>
<span><span class="va">exampleInversion</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">"+"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-190      + | chrB:100-190      TRUE</span></span>
<span><span class="co">##   [2]     chrA   210-291      + | chrB:210-291      TRUE</span></span>
<span><span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note the tolerance window</span></span>
<span><span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="va">exampleInversion</span>, tol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query   inv</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-190      + | chrB:100-190 FALSE</span></span>
<span><span class="co">##   [2]     chrA   210-291      - | chrB:210-291 FALSE</span></span>
<span><span class="co">##   [3]     chrA   301-400      + | chrB:301-400 FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="double-inversions">Double inversions<a class="anchor" aria-label="anchor" href="#double-inversions"></a>
</h4>
<p>Double inversions are the result of two overlapping trivial
inversions, and are detected with a different approach, implemented in
the <code><a href="../reference/flagDoubleInversions.html">flagDoubleInversions()</a></code> function.</p>
<p>Here we construct a double inversion in 3 steps:</p>
<ol style="list-style-type: decimal">
<li>First, we start from 5 collinear ranges,</li>
<li>then we invert ranges 2 and 3,</li>
<li>then we invert ranges 3 an 4 (of step 2).</li>
</ol>
<p>show a triplet resulting from a double inversion, flanked by two
regions that did not move.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleColinear5</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span>main <span class="op">=</span> <span class="st">"5 collinear ranges"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions-1.png" width="700"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleInversion5uncollapsed</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span>main <span class="op">=</span> <span class="st">"Invert ranges 2 and 3"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions-2.png" width="700"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleDoubleInversion1</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span>main <span class="op">=</span> <span class="st">"Invert ranges 3 and 4 of the previous step"</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions-3.png" width="700"></p>
<p>Same story but showing coalesced alignments.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleColinear5</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions2-1.png" width="700"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleInversion5uncollapsed</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions2-2.png" width="700"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleDoubleInversion1</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/doubleInversions2-3.png" width="700"></p>
<p>Given the double inversion example above, but not its history, we can
deduce the coordinates of the inversions (pairs of breakpoints), but not
the order in which they took place.</p>
</div>
<div class="section level4">
<h4 id="nested-inversions">Nested inversions<a class="anchor" aria-label="anchor" href="#nested-inversions"></a>
</h4>
<p>The <code><a href="../reference/flagInversions.html">?flagInversions</a></code> algorithm only detects the simplest
cases. In order to detect nested inversions, the inner ones have to be
removed.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleNestedInversions</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/nestedInversions-1.png" width="700"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleNestedInversions</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 5 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query   inv</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-190      + | chrA:100-190 FALSE</span></span>
<span><span class="co">##   [2]     chrA   200-290      - | chrA:400-490  TRUE</span></span>
<span><span class="co">##   [3]     chrA   300-390      + | chrA:300-390 FALSE</span></span>
<span><span class="co">##   [4]     chrA   400-490      - | chrA:200-290 FALSE</span></span>
<span><span class="co">##   [5]     chrA   500-590      + | chrA:500-590 FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">flipped</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">exampleNestedInversions</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flipInversions.html">flipInversions</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 5 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-190      + | chrA:100-190</span></span>
<span><span class="co">##   [2]     chrA   200-290      - | chrA:400-490</span></span>
<span><span class="co">##   [3]     chrA   300-390      - | chrA:300-390</span></span>
<span><span class="co">##   [4]     chrA   400-490      - | chrA:200-290</span></span>
<span><span class="co">##   [5]     chrA   500-590      + | chrA:500-590</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">flipped</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/nestedInversions-2.png" width="700"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">flipped</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 3 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query     score   inv</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;integer&gt; &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-190      + | chrA:100-190        91  TRUE</span></span>
<span><span class="co">##   [2]     chrA   200-490      - | chrA:200-490       291 FALSE</span></span>
<span><span class="co">##   [3]     chrA   500-590      + | chrA:500-590        91 FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="transolocations">Transolocations<a class="anchor" aria-label="anchor" href="#transolocations"></a>
</h3>
<p>If a region has moved, but is not an inversion, then it is a
translocation.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleTranslocation</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 3 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear   inv</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt; &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-200      + | chrB:100-200     FALSE FALSE</span></span>
<span><span class="co">##   [2]     chrA   201-300      + | chrC:201-300     FALSE FALSE</span></span>
<span><span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleTranslocation</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/translocation-1.png" width="700"></p>
<p>It does not matter if the region moved on the minus strand</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleTranslocation2</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagColinearAlignments.html">flagColinearAlignments</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/flagInversions.html">flagInversions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 3 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query  colinear   inv</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt; &lt;logical&gt; &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-200      + | chrB:100-200     FALSE FALSE</span></span>
<span><span class="co">##   [2]     chrA   201-300      - | chrC:201-300     FALSE FALSE</span></span>
<span><span class="co">##   [3]     chrA   301-400      + | chrB:301-400     FALSE FALSE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleTranslocation2</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/translocationOnMinus-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="insertions-deletions-indels">Insertions / Deletions (indels)<a class="anchor" aria-label="anchor" href="#insertions-deletions-indels"></a>
</h3>
<p>NOTE that searching for insertion/deletions make little sense in
one-to-one alignment unless the maximal distance for colinearity is set
to a meaningful value.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleDeletion</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-200      + | chrB:100-200</span></span>
<span><span class="co">##   [2]     chrA   201-300      + | chrC:401-500</span></span>
<span><span class="co">##   [3]     chrA   301-400      + | chrB:201-300</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleDeletion</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/indels-1.png" width="700"></p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleInsertion</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 3 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames    ranges strand |        query</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span></span>
<span><span class="co">##   [1]     chrA   100-200      + | chrB:100-200</span></span>
<span><span class="co">##   [2]     chrA   201-300      + | chrB:301-400</span></span>
<span><span class="co">##   [3]     chrC   401-500      + | chrB:201-300</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">exampleInsertion</span><span class="op">)</span></span></code></pre></div>
<p><img src="StructuralVariants_files/figure-html/indels-2.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Charlotte West, Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
