# Generated by using Rcpp::compileAttributes() -> do not edit by hand
# Generator token: 10BE3573-1514-4C36-9D1C-5A225CD40393

#' Read a MAF file
#'
#' Reads a pairwise genome alignment in MAF format.  The file can be plain
#' text or compressed with `gzip`.
#'
#' Known limitations: Does not expand shell metacharacters.  Trusts blindly
#' file extension to determine compression.  Does not perform any validation on
#' the file format.  Assumes that the score comes first in the 'a' lines.
#'
#' @param inputFileName The name of the file to read
#' @return a `list` object with containg coordinates of the alignments in both
#' genomes and other information such as alignment width and number of matches.
#' @importFrom Rcpp evalCpp
#' @useDynLib GenomicBreaks, .registration = TRUE
readMAF <- function(inputFileName) {
    .Call(`_GenomicBreaks_readMAF`, inputFileName)
}

